<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="karltayeb" />

<meta name="date" content="2022-03-16" />

<title>Baboon Diet Example</title>

<script src="site_libs/header-attrs-2.9/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">logistic-susie-gsea</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/karltayeb/logistic-susie-gsea">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Baboon Diet Example</h1>
<h4 class="author">karltayeb</h4>
<h4 class="date">2022-03-16</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-05-18
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>logistic-susie-gsea/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220105code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20220105)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220105code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220105)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomkarltayeblogisticsusiegseatreeb15f3c3e91221abe01f8bfcdac6f967f9e2530bbtargetblankb15f3c3a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/karltayeb/logistic-susie-gsea/tree/b15f3c3e91221abe01f8bfcdac6f967f9e2530bb" target="_blank">b15f3c3</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomkarltayeblogisticsusiegseatreeb15f3c3e91221abe01f8bfcdac6f967f9e2530bbtargetblankb15f3c3a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/karltayeb/logistic-susie-gsea/tree/b15f3c3e91221abe01f8bfcdac6f967f9e2530bb" target="_blank">b15f3c3</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    _targets.R
    Ignored:    _targets.html
    Ignored:    _targets/objects/
    Ignored:    _targets/user/
    Ignored:    _targets/workspaces/
    Ignored:    _targets_r/
    Ignored:    cache/
    Ignored:    data/adipose_2yr_topsnp.txt
    Ignored:    data/chimp_human_eb_de/
    Ignored:    data/de-droplet-noshrink.RData
    Ignored:    data/deng/
    Ignored:    data/fetal_reference_cellid_gene_sets.RData
    Ignored:    data/human_chimp_eb/
    Ignored:    data/pbmc-purified/
    Ignored:    data/wenhe_baboon_diet/
    Ignored:    figure/
    Ignored:    library/

Untracked files:
    Untracked:  Rplots.pdf
    Untracked:  _targets/meta/
    Untracked:  clustrmq_slurm.sh
    Untracked:  l1.sim.aggregate.scores
    Untracked:  simulation_targets/

Unstaged changes:
    Modified:   _targets.Rmd
    Modified:   code/target_run_enrichment.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/baboon_diet.Rmd</code>) and HTML (<code>docs/baboon_diet.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/karltayeb/logistic-susie-gsea/blob/b15f3c3e91221abe01f8bfcdac6f967f9e2530bb/analysis/baboon_diet.Rmd" target="_blank">b15f3c3</a>
</td>
<td>
Karl Tayeb
</td>
<td>
2022-05-18
</td>
<td>
wflow_publish(“analysis/baboon_diet.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/karltayeb/logistic-susie-gsea/6e2224f57072d030f417a5ac270dac8667d6f8d0/docs/baboon_diet.html" target="_blank">6e2224f</a>
</td>
<td>
karltayeb
</td>
<td>
2022-04-12
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/karltayeb/logistic-susie-gsea/blob/5e3a4e3372da7da07eece5d21a34c754b8718ce0/analysis/baboon_diet.Rmd" target="_blank">5e3a4e3</a>
</td>
<td>
karltayeb
</td>
<td>
2022-04-12
</td>
<td>
wflow_publish(c(“analysis/index.Rmd”, “analysis/baboon_diet.Rmd”,
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/karltayeb/logistic-susie-gsea/26e6c281786276373a45be7b4d46bc630b3de030/docs/baboon_diet.html" target="_blank">26e6c28</a>
</td>
<td>
karltayeb
</td>
<td>
2022-03-30
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/karltayeb/logistic-susie-gsea/1ee6003b7116e90512f176dd0af9142910c6022e/docs/baboon_diet.html" target="_blank">1ee6003</a>
</td>
<td>
karltayeb
</td>
<td>
2022-03-30
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/karltayeb/logistic-susie-gsea/blob/6798b78ba72dd4b84a4c34ffd9eb115eb7a786a5/analysis/baboon_diet.Rmd" target="_blank">6798b78</a>
</td>
<td>
karltayeb
</td>
<td>
2022-03-30
</td>
<td>
wflow_publish(“analysis/baboon_diet.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/karltayeb/logistic-susie-gsea/7d1a82b7d32a76d375888b878163600f7df1bf51/docs/baboon_diet.html" target="_blank">7d1a82b</a>
</td>
<td>
karltayeb
</td>
<td>
2022-03-30
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/karltayeb/logistic-susie-gsea/blob/1e5cfb0f4b5db442503436075d4baed088ceacbd/analysis/baboon_diet.Rmd" target="_blank">1e5cfb0</a>
</td>
<td>
karltayeb
</td>
<td>
2022-03-30
</td>
<td>
wflow_publish(“analysis/baboon_diet.Rmd”)
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(gseasusie)
library(ggplot2)
library(tidyverse)
library(targets)

tar_load(baboon_diet_enrichment_summary)
results &lt;- baboon_diet_enrichment_summary</code></pre>
<pre class="r"><code>res &lt;- results$enrichment_summary[[2]]
res &lt;- res %&gt;%
  mutate(pval_marginal = 2*pnorm(-abs(effect_marginal/effect_se_marginal))) %&gt;%
  mutate(pval_residual = 2*pnorm(-abs(effect_residual/effect_se_residual)))

res %&gt;% gseasusie::enrichment_volcano2()
res %&gt;% gseasusie::residual_enrichment_histogram2()
res %&gt;% gseasusie:::static_table()</code></pre>
<pre class="r"><code>results &lt;- results %&gt;% arrange(experiment, ptop, db)
experiments &lt;- results$experiment %&gt;% unique()

for(this_experiment in experiments){
  cat(&#39;\n&#39;)
  cat(&#39;##&#39;, this_experiment)

  sub_res &lt;- results %&gt;% filter(experiment == this_experiment)

  N &lt;- dim(sub_res)[1]
  for (i in 1:N){
    this_db &lt;- sub_res$db[[i]]
    this_ptop &lt;- sub_res$ptop[[i]]
    res &lt;- sub_res$enrichment_summary[[i]]

    cat(&quot;\n&quot;) 
    cat(&quot;### prop genes = &quot;, this_ptop, &#39;, db = &#39;, this_db, &quot;\n&quot;) # Create second level headings with the names.
    volcano &lt;- res %&gt;% gseasusie::enrichment_volcano2()
    hist &lt;- res %&gt;% filter(overlap &gt; 0) %&gt;% residual_enrichment_histogram2() 
    print(volcano)
    print(hist)
    cat(&quot;\n\n&quot;)
    print(gseasusie:::static_table(res))
  }
}</code></pre>
<div id="adipose" class="section level2">
<h2>Adipose</h2>
<div id="prop-genes-0.05-db-all_msigdb" class="section level3">
<h3>prop genes = 0.05 , db = all_msigdb</h3>
<p><img src="figure/baboon_diet.Rmd/experiment.level.results-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-experiment.level.results-1">
Past versions of experiment.level.results-1.png
</button>
</p>
<div id="fig-experiment.level.results-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/karltayeb/logistic-susie-gsea/blob/6e2224f57072d030f417a5ac270dac8667d6f8d0/docs/figure/baboon_diet.Rmd/experiment.level.results-1.png" target="_blank">6e2224f</a>
</td>
<td>
karltayeb
</td>
<td>
2022-04-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/baboon_diet.Rmd/experiment.level.results-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-experiment.level.results-2">
Past versions of experiment.level.results-2.png
</button>
</p>
<div id="fig-experiment.level.results-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/karltayeb/logistic-susie-gsea/blob/6e2224f57072d030f417a5ac270dac8667d6f8d0/docs/figure/baboon_diet.Rmd/experiment.level.results-2.png" target="_blank">6e2224f</a>
</td>
<td>
karltayeb
</td>
<td>
2022-04-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
geneSet
</th>
<th style="text-align:left;">
description
</th>
<th style="text-align:left;">
geneSetSize
</th>
<th style="text-align:left;">
overlap
</th>
<th style="text-align:left;">
logOddsRatio
</th>
<th style="text-align:left;">
beta
</th>
<th style="text-align:left;">
alpha
</th>
<th style="text-align:left;">
pip
</th>
<th style="text-align:left;">
pFishersExact
</th>
<th style="text-align:left;">
fisherRank
</th>
</tr>
</thead>
<tbody>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L1</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M13283
</td>
<td style="text-align:left;">
Liver selective genes
</td>
<td style="text-align:left;">
203
</td>
<td style="text-align:left;">
104
</td>
<td style="text-align:left;">
1.37
</td>
<td style="text-align:left;">
1.82
</td>
<td style="text-align:left;color: green !important;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
2.24e-81
</td>
<td style="text-align:left;">
1
</td>
</tr>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L2</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M2194
</td>
<td style="text-align:left;">
Genes down-regulated in liver samples of liver-specific knockout of HNF4A [GeneID=3172].
</td>
<td style="text-align:left;">
138
</td>
<td style="text-align:left;">
63
</td>
<td style="text-align:left;">
1.24
</td>
<td style="text-align:left;">
2.06
</td>
<td style="text-align:left;color: green !important;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
3.25e-45
</td>
<td style="text-align:left;">
8
</td>
</tr>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L3</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M7054
</td>
<td style="text-align:left;">
Genes up-regulated specifically in human liver tissue.
</td>
<td style="text-align:left;">
44
</td>
<td style="text-align:left;">
35
</td>
<td style="text-align:left;">
1.89
</td>
<td style="text-align:left;">
2.96
</td>
<td style="text-align:left;color: green !important;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
6.08e-38
</td>
<td style="text-align:left;">
14
</td>
</tr>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L4</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M38993
</td>
<td style="text-align:left;">
Transcriptome of human HepaRG hepatocellular carcinoma liver progenitors in responses to a WNT3A-enriched microenvironment and dissection of pathways dependent on _-catenin and/or blocked by the SFRP-like Wnt inhibitor FZD8_CRD.
</td>
<td style="text-align:left;">
335
</td>
<td style="text-align:left;">
84
</td>
<td style="text-align:left;">
0.85
</td>
<td style="text-align:left;">
1.22
</td>
<td style="text-align:left;color: green !important;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
2.04e-36
</td>
<td style="text-align:left;">
16
</td>
</tr>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L5</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M14108
</td>
<td style="text-align:left;">
Neighborhood of HPX
</td>
<td style="text-align:left;">
108
</td>
<td style="text-align:left;">
73
</td>
<td style="text-align:left;">
1.64
</td>
<td style="text-align:left;">
1.95
</td>
<td style="text-align:left;color: green !important;">
0.998
</td>
<td style="text-align:left;">
0.998
</td>
<td style="text-align:left;">
1.61e-69
</td>
<td style="text-align:left;">
2
</td>
</tr>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L6</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M2396
</td>
<td style="text-align:left;">
Genes down-regulated in pancreatic islets upon knockout of HNF1A [GeneID=6927].
</td>
<td style="text-align:left;">
90
</td>
<td style="text-align:left;">
29
</td>
<td style="text-align:left;">
0.972
</td>
<td style="text-align:left;">
1.83
</td>
<td style="text-align:left;color: green !important;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
1.98e-16
</td>
<td style="text-align:left;">
38
</td>
</tr>
</tbody>
</table>
</div>
<div id="prop-genes-0.1-db-all_msigdb" class="section level3">
<h3>prop genes = 0.1 , db = all_msigdb</h3>
<p><img src="figure/baboon_diet.Rmd/experiment.level.results-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-experiment.level.results-3">
Past versions of experiment.level.results-3.png
</button>
</p>
<div id="fig-experiment.level.results-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/karltayeb/logistic-susie-gsea/blob/6e2224f57072d030f417a5ac270dac8667d6f8d0/docs/figure/baboon_diet.Rmd/experiment.level.results-3.png" target="_blank">6e2224f</a>
</td>
<td>
karltayeb
</td>
<td>
2022-04-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/baboon_diet.Rmd/experiment.level.results-4.png" width="672" style="display: block; margin: auto;" /></p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
geneSet
</th>
<th style="text-align:left;">
description
</th>
<th style="text-align:left;">
geneSetSize
</th>
<th style="text-align:left;">
overlap
</th>
<th style="text-align:left;">
logOddsRatio
</th>
<th style="text-align:left;">
beta
</th>
<th style="text-align:left;">
alpha
</th>
<th style="text-align:left;">
pip
</th>
<th style="text-align:left;">
pFishersExact
</th>
<th style="text-align:left;">
fisherRank
</th>
</tr>
</thead>
<tbody>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L1</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M13283
</td>
<td style="text-align:left;">
Liver selective genes
</td>
<td style="text-align:left;">
203
</td>
<td style="text-align:left;">
115
</td>
<td style="text-align:left;">
1.1
</td>
<td style="text-align:left;">
1.44
</td>
<td style="text-align:left;color: green !important;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
3.2e-62
</td>
<td style="text-align:left;">
1
</td>
</tr>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L2</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M2194
</td>
<td style="text-align:left;">
Genes down-regulated in liver samples of liver-specific knockout of HNF4A [GeneID=3172].
</td>
<td style="text-align:left;">
138
</td>
<td style="text-align:left;">
71
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
1.62
</td>
<td style="text-align:left;color: green !important;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
5.79e-35
</td>
<td style="text-align:left;">
7
</td>
</tr>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L3</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M14108
</td>
<td style="text-align:left;">
Neighborhood of HPX
</td>
<td style="text-align:left;">
108
</td>
<td style="text-align:left;">
76
</td>
<td style="text-align:left;">
1.35
</td>
<td style="text-align:left;">
1.58
</td>
<td style="text-align:left;color: green !important;">
0.982
</td>
<td style="text-align:left;">
0.982
</td>
<td style="text-align:left;">
1.76e-51
</td>
<td style="text-align:left;">
2
</td>
</tr>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L4</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M15429
</td>
<td style="text-align:left;">
Genes having at least one occurrence of the motif GGTTAATNWTTAMCN in the regions spanning 4 kb centered on their transcription starting sites [-2kb, +2kb]. This matches the TCF1 [GeneSymbol=TCF1] transcription factor binding site V$HNF1_01 (v7.4 TRANSFAC).
</td>
<td style="text-align:left;">
205
</td>
<td style="text-align:left;">
61
</td>
<td style="text-align:left;">
0.596
</td>
<td style="text-align:left;">
1.13
</td>
<td style="text-align:left;color: green !important;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
1.74e-15
</td>
<td style="text-align:left;">
32
</td>
</tr>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L5</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M38993
</td>
<td style="text-align:left;">
Transcriptome of human HepaRG hepatocellular carcinoma liver progenitors in responses to a WNT3A-enriched microenvironment and dissection of pathways dependent on _-catenin and/or blocked by the SFRP-like Wnt inhibitor FZD8_CRD.
</td>
<td style="text-align:left;">
335
</td>
<td style="text-align:left;">
106
</td>
<td style="text-align:left;">
0.646
</td>
<td style="text-align:left;">
0.942
</td>
<td style="text-align:left;color: green !important;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
1.52e-28
</td>
<td style="text-align:left;">
14
</td>
</tr>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L6</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M7054
</td>
<td style="text-align:left;">
Genes up-regulated specifically in human liver tissue.
</td>
<td style="text-align:left;">
44
</td>
<td style="text-align:left;">
35
</td>
<td style="text-align:left;">
1.55
</td>
<td style="text-align:left;">
2.34
</td>
<td style="text-align:left;color: green !important;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
1.97e-27
</td>
<td style="text-align:left;">
16
</td>
</tr>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L7</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M10712
</td>
<td style="text-align:left;">
A small subcellular vesicle, surrounded by a membrane, that is formed from the Golgi apparatus and contains a highly concentrated protein destined for secretion. Secretory granules move towards the periphery of the cell and upon stimulation, their membranes fuse with the cell membrane, and their protein load is exteriorized. Processing of the contained protein may take place in secretory granules. [GOC:mah, <a href="ISBN:0198506732" class="uri">ISBN:0198506732</a>]
</td>
<td style="text-align:left;">
668
</td>
<td style="text-align:left;">
130
</td>
<td style="text-align:left;">
0.361
</td>
<td style="text-align:left;">
0.677
</td>
<td style="text-align:left;color: green !important;">
0.983
</td>
<td style="text-align:left;">
0.983
</td>
<td style="text-align:left;">
4.85e-14
</td>
<td style="text-align:left;">
38
</td>
</tr>
</tbody>
</table>
</div>
<div id="prop-genes-0.2-db-all_msigdb" class="section level3">
<h3>prop genes = 0.2 , db = all_msigdb</h3>
<p><img src="figure/baboon_diet.Rmd/experiment.level.results-5.png" width="672" style="display: block; margin: auto;" /><img src="figure/baboon_diet.Rmd/experiment.level.results-6.png" width="672" style="display: block; margin: auto;" /></p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
geneSet
</th>
<th style="text-align:left;">
description
</th>
<th style="text-align:left;">
geneSetSize
</th>
<th style="text-align:left;">
overlap
</th>
<th style="text-align:left;">
logOddsRatio
</th>
<th style="text-align:left;">
beta
</th>
<th style="text-align:left;">
alpha
</th>
<th style="text-align:left;">
pip
</th>
<th style="text-align:left;">
pFishersExact
</th>
<th style="text-align:left;">
fisherRank
</th>
</tr>
</thead>
<tbody>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L1</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M13283
</td>
<td style="text-align:left;">
Liver selective genes
</td>
<td style="text-align:left;">
203
</td>
<td style="text-align:left;">
130
</td>
<td style="text-align:left;">
0.87
</td>
<td style="text-align:left;">
1.5
</td>
<td style="text-align:left;color: green !important;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
4.92e-43
</td>
<td style="text-align:left;">
1
</td>
</tr>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L2</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M2194
</td>
<td style="text-align:left;">
Genes down-regulated in liver samples of liver-specific knockout of HNF4A [GeneID=3172].
</td>
<td style="text-align:left;">
138
</td>
<td style="text-align:left;">
86
</td>
<td style="text-align:left;">
0.832
</td>
<td style="text-align:left;">
1.49
</td>
<td style="text-align:left;color: green !important;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
1.37e-27
</td>
<td style="text-align:left;">
6
</td>
</tr>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L3</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M11266
</td>
<td style="text-align:left;">
Lysosome
</td>
<td style="text-align:left;">
110
</td>
<td style="text-align:left;">
50
</td>
<td style="text-align:left;">
0.528
</td>
<td style="text-align:left;">
1.17
</td>
<td style="text-align:left;color: green !important;">
0.998
</td>
<td style="text-align:left;">
0.998
</td>
<td style="text-align:left;">
1.85e-09
</td>
<td style="text-align:left;">
56
</td>
</tr>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L4</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M7054
</td>
<td style="text-align:left;">
Genes up-regulated specifically in human liver tissue.
</td>
<td style="text-align:left;">
44
</td>
<td style="text-align:left;">
38
</td>
<td style="text-align:left;">
1.41
</td>
<td style="text-align:left;">
2.33
</td>
<td style="text-align:left;color: green !important;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
4.4e-21
</td>
<td style="text-align:left;">
11
</td>
</tr>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L5</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M13558
</td>
<td style="text-align:left;">
Any process that results in a change in state or activity of a cell or an organism (in terms of movement, secretion, enzyme production, gene expression, etc.) as a result of an oxygen-containing compound stimulus. [GOC:pr, GOC:TermGenie]
</td>
<td style="text-align:left;">
1310
</td>
<td style="text-align:left;">
369
</td>
<td style="text-align:left;">
0.217
</td>
<td style="text-align:left;">
0.379
</td>
<td style="text-align:left;color: green !important;">
0.999
</td>
<td style="text-align:left;">
0.999
</td>
<td style="text-align:left;">
1.49e-13
</td>
<td style="text-align:left;">
24
</td>
</tr>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L7</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M15429
</td>
<td style="text-align:left;">
Genes having at least one occurrence of the motif GGTTAATNWTTAMCN in the regions spanning 4 kb centered on their transcription starting sites [-2kb, +2kb]. This matches the TCF1 [GeneSymbol=TCF1] transcription factor binding site V$HNF1_01 (v7.4 TRANSFAC).
</td>
<td style="text-align:left;">
205
</td>
<td style="text-align:left;">
85
</td>
<td style="text-align:left;">
0.461
</td>
<td style="text-align:left;">
0.882
</td>
<td style="text-align:left;color: green !important;">
0.999
</td>
<td style="text-align:left;">
0.999
</td>
<td style="text-align:left;">
1.57e-12
</td>
<td style="text-align:left;">
27
</td>
</tr>
</tbody>
</table>
</div>
<div id="prop-genes-0.5-db-all_msigdb" class="section level3">
<h3>prop genes = 0.5 , db = all_msigdb</h3>
<p><img src="figure/baboon_diet.Rmd/experiment.level.results-7.png" width="672" style="display: block; margin: auto;" /><img src="figure/baboon_diet.Rmd/experiment.level.results-8.png" width="672" style="display: block; margin: auto;" /></p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
geneSet
</th>
<th style="text-align:left;">
description
</th>
<th style="text-align:left;">
geneSetSize
</th>
<th style="text-align:left;">
overlap
</th>
<th style="text-align:left;">
logOddsRatio
</th>
<th style="text-align:left;">
beta
</th>
<th style="text-align:left;">
alpha
</th>
<th style="text-align:left;">
pip
</th>
<th style="text-align:left;">
pFishersExact
</th>
<th style="text-align:left;">
fisherRank
</th>
</tr>
</thead>
<tbody>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L1</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M13283
</td>
<td style="text-align:left;">
Liver selective genes
</td>
<td style="text-align:left;">
203
</td>
<td style="text-align:left;">
160
</td>
<td style="text-align:left;">
0.578
</td>
<td style="text-align:left;">
1.15
</td>
<td style="text-align:left;color: green !important;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
2.94e-17
</td>
<td style="text-align:left;">
1
</td>
</tr>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L2</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M567
</td>
<td style="text-align:left;">
SRP-dependent cotranslational protein targeting to membrane
</td>
<td style="text-align:left;">
100
</td>
<td style="text-align:left;">
16
</td>
<td style="text-align:left;">
-0.724
</td>
<td style="text-align:left;">
-1.51
</td>
<td style="text-align:left;color: red !important;">
0.953
</td>
<td style="text-align:left;">
0.953
</td>
<td style="text-align:left;">
1.3e-12
</td>
<td style="text-align:left;">
3
</td>
</tr>
<tr grouplength="2">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L3</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M39417
</td>
<td style="text-align:left;">
Electron transport chain: OXPHOS system in mitochondria
</td>
<td style="text-align:left;">
63
</td>
<td style="text-align:left;">
9
</td>
<td style="text-align:left;">
-0.781
</td>
<td style="text-align:left;">
-1.55
</td>
<td style="text-align:left;color: red !important;">
0.936
</td>
<td style="text-align:left;">
0.936
</td>
<td style="text-align:left;">
3.29e-09
</td>
<td style="text-align:left;">
24
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M18264
</td>
<td style="text-align:left;">
Genes involved in oxidative phosphorylation; based on literature and sequence annotation resources and converted to Affymetrix HG-U133A probe sets.
</td>
<td style="text-align:left;">
57
</td>
<td style="text-align:left;">
9
</td>
<td style="text-align:left;">
-0.729
</td>
<td style="text-align:left;">
-0.0372
</td>
<td style="text-align:left;color: red !important;">
0.0244
</td>
<td style="text-align:left;">
0.0248
</td>
<td style="text-align:left;">
8.47e-08
</td>
<td style="text-align:left;">
43
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="liver" class="section level2">
<h2>Liver</h2>
<div id="prop-genes-0.05-db-all_msigdb-1" class="section level3">
<h3>prop genes = 0.05 , db = all_msigdb</h3>
<p><img src="figure/baboon_diet.Rmd/experiment.level.results-9.png" width="672" style="display: block; margin: auto;" /><img src="figure/baboon_diet.Rmd/experiment.level.results-10.png" width="672" style="display: block; margin: auto;" /></p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
geneSet
</th>
<th style="text-align:left;">
description
</th>
<th style="text-align:left;">
geneSetSize
</th>
<th style="text-align:left;">
overlap
</th>
<th style="text-align:left;">
logOddsRatio
</th>
<th style="text-align:left;">
beta
</th>
<th style="text-align:left;">
alpha
</th>
<th style="text-align:left;">
pip
</th>
<th style="text-align:left;">
pFishersExact
</th>
<th style="text-align:left;">
fisherRank
</th>
</tr>
</thead>
<tbody>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L1</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M8295
</td>
<td style="text-align:left;">
Neighborhood of TTN
</td>
<td style="text-align:left;">
25
</td>
<td style="text-align:left;">
22
</td>
<td style="text-align:left;">
2.16
</td>
<td style="text-align:left;">
4.25
</td>
<td style="text-align:left;color: green !important;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
3.52e-26
</td>
<td style="text-align:left;">
5
</td>
</tr>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L2</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M12628
</td>
<td style="text-align:left;">
A process in which force is generated within muscle tissue, resulting in a change in muscle geometry. Force generation involves a chemo-mechanical energy conversion step that is carried out by the actin/myosin complex activity, which generates force through ATP hydrolysis. [GOC:ef, GOC:mtg_muscle, <a href="ISBN:0198506732" class="uri">ISBN:0198506732</a>]
</td>
<td style="text-align:left;">
286
</td>
<td style="text-align:left;">
71
</td>
<td style="text-align:left;">
0.837
</td>
<td style="text-align:left;">
1.45
</td>
<td style="text-align:left;color: green !important;">
0.999
</td>
<td style="text-align:left;">
0.999
</td>
<td style="text-align:left;">
1.55e-30
</td>
<td style="text-align:left;">
1
</td>
</tr>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L3</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M9898
</td>
<td style="text-align:left;">
Set ‘Suz12 targets’: genes identified by ChIP on chip as targets of the Polycomb protein SUZ12 [GeneID=23512] in human embryonic stem cells.
</td>
<td style="text-align:left;">
711
</td>
<td style="text-align:left;">
97
</td>
<td style="text-align:left;">
0.521
</td>
<td style="text-align:left;">
1.07
</td>
<td style="text-align:left;color: green !important;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
4.54e-20
</td>
<td style="text-align:left;">
9
</td>
</tr>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L4</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M17471
</td>
<td style="text-align:left;">
Genes up-regulated in invasive ductal carcinoma (IDC) relative to ductal carcinoma in situ (DCIS, non-invasive).
</td>
<td style="text-align:left;">
314
</td>
<td style="text-align:left;">
52
</td>
<td style="text-align:left;">
0.601
</td>
<td style="text-align:left;">
1.22
</td>
<td style="text-align:left;color: green !important;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
1.6e-14
</td>
<td style="text-align:left;">
43
</td>
</tr>
</tbody>
</table>
</div>
<div id="prop-genes-0.1-db-all_msigdb-1" class="section level3">
<h3>prop genes = 0.1 , db = all_msigdb</h3>
<p><img src="figure/baboon_diet.Rmd/experiment.level.results-11.png" width="672" style="display: block; margin: auto;" /><img src="figure/baboon_diet.Rmd/experiment.level.results-12.png" width="672" style="display: block; margin: auto;" /></p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
geneSet
</th>
<th style="text-align:left;">
description
</th>
<th style="text-align:left;">
geneSetSize
</th>
<th style="text-align:left;">
overlap
</th>
<th style="text-align:left;">
logOddsRatio
</th>
<th style="text-align:left;">
beta
</th>
<th style="text-align:left;">
alpha
</th>
<th style="text-align:left;">
pip
</th>
<th style="text-align:left;">
pFishersExact
</th>
<th style="text-align:left;">
fisherRank
</th>
</tr>
</thead>
<tbody>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L1</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M1920
</td>
<td style="text-align:left;">
Genes forming the macrophage-enriched metabolic network (MEMN) claimed to have a causal relationship with the metabolic syndrom traits.
</td>
<td style="text-align:left;">
1080
</td>
<td style="text-align:left;">
241
</td>
<td style="text-align:left;">
0.466
</td>
<td style="text-align:left;">
0.785
</td>
<td style="text-align:left;color: green !important;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
7.26e-36
</td>
<td style="text-align:left;">
1
</td>
</tr>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L10</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M2009
</td>
<td style="text-align:left;">
Genes with high-CpG-density promoters (HCP) bearing histone H3 trimethylation mark at K27 (H3K27me3) in neural progenitor cells (NPC).
</td>
<td style="text-align:left;">
226
</td>
<td style="text-align:left;">
58
</td>
<td style="text-align:left;">
0.505
</td>
<td style="text-align:left;">
1.04
</td>
<td style="text-align:left;color: green !important;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
8.7e-12
</td>
<td style="text-align:left;">
145
</td>
</tr>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L2</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M8295
</td>
<td style="text-align:left;">
Neighborhood of TTN
</td>
<td style="text-align:left;">
25
</td>
<td style="text-align:left;">
23
</td>
<td style="text-align:left;">
2.02
</td>
<td style="text-align:left;">
3.83
</td>
<td style="text-align:left;color: green !important;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
2.09e-21
</td>
<td style="text-align:left;">
19
</td>
</tr>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L3</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M5389
</td>
<td style="text-align:left;">
Genes whose expression increases with age in normal kidney.
</td>
<td style="text-align:left;">
413
</td>
<td style="text-align:left;">
122
</td>
<td style="text-align:left;">
0.606
</td>
<td style="text-align:left;">
0.99
</td>
<td style="text-align:left;color: green !important;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
1.38e-29
</td>
<td style="text-align:left;">
4
</td>
</tr>
<tr grouplength="3">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L4</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M42761
</td>
<td style="text-align:left;">
Any process that modulates the frequency, rate or extent of the directed movement of ions from one side of a membrane to the other. [GOC:mah]
</td>
<td style="text-align:left;">
362
</td>
<td style="text-align:left;">
97
</td>
<td style="text-align:left;">
0.54
</td>
<td style="text-align:left;">
0.706
</td>
<td style="text-align:left;color: green !important;">
0.764
</td>
<td style="text-align:left;">
0.764
</td>
<td style="text-align:left;">
2.65e-20
</td>
<td style="text-align:left;">
28
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M15848
</td>
<td style="text-align:left;">
Any process that modulates the frequency, rate or extent of the directed movement of charged atoms or small charged molecules into, out of or within a cell, or between cells, by means of some agent such as a transporter or pore. [GOC:jl]
</td>
<td style="text-align:left;">
513
</td>
<td style="text-align:left;">
123
</td>
<td style="text-align:left;">
0.479
</td>
<td style="text-align:left;">
0.126
</td>
<td style="text-align:left;color: green !important;">
0.162
</td>
<td style="text-align:left;">
0.162
</td>
<td style="text-align:left;">
7.64e-21
</td>
<td style="text-align:left;">
22
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M16782
</td>
<td style="text-align:left;">
Any process that modulates the frequency, rate or extent of the directed movement of a solute from one side of a membrane to the other. [GOC:mah]
</td>
<td style="text-align:left;">
437
</td>
<td style="text-align:left;">
107
</td>
<td style="text-align:left;">
0.488
</td>
<td style="text-align:left;">
0.0606
</td>
<td style="text-align:left;color: green !important;">
0.0732
</td>
<td style="text-align:left;">
0.0733
</td>
<td style="text-align:left;">
6.81e-19
</td>
<td style="text-align:left;">
36
</td>
</tr>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L5</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M1578
</td>
<td style="text-align:left;">
Genes down-regulated in freshly isolated CD31- [GeneID=5175] (stromal stem cells from adipose tissue) versus the CD31+ (non-stem) counterparts.
</td>
<td style="text-align:left;">
188
</td>
<td style="text-align:left;">
58
</td>
<td style="text-align:left;">
0.618
</td>
<td style="text-align:left;">
1.24
</td>
<td style="text-align:left;color: green !important;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
1.43e-15
</td>
<td style="text-align:left;">
64
</td>
</tr>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L6</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M17471
</td>
<td style="text-align:left;">
Genes up-regulated in invasive ductal carcinoma (IDC) relative to ductal carcinoma in situ (DCIS, non-invasive).
</td>
<td style="text-align:left;">
314
</td>
<td style="text-align:left;">
105
</td>
<td style="text-align:left;">
0.682
</td>
<td style="text-align:left;">
0.911
</td>
<td style="text-align:left;color: green !important;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
1.45e-30
</td>
<td style="text-align:left;">
3
</td>
</tr>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L7</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M6298
</td>
<td style="text-align:left;">
Fibers, composed of actin, myosin, and associated proteins, found in cells of smooth or striated muscle. [GOC:go_curators, <a href="ISBN:0815316194" class="uri">ISBN:0815316194</a>]
</td>
<td style="text-align:left;">
212
</td>
<td style="text-align:left;">
76
</td>
<td style="text-align:left;">
0.721
</td>
<td style="text-align:left;">
1.22
</td>
<td style="text-align:left;color: green !important;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
1.53e-24
</td>
<td style="text-align:left;">
10
</td>
</tr>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L8</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M29940
</td>
<td style="text-align:left;">
Genes containing one or more binding sites for UniProt:Q9BTC0 (DIDO1) in their promoter regions (TSS -1000,+100 bp) as identified by GTRD version 20.06 ChIP-seq harmonization.
</td>
<td style="text-align:left;">
1300
</td>
<td style="text-align:left;">
30
</td>
<td style="text-align:left;">
-0.708
</td>
<td style="text-align:left;">
-1.13
</td>
<td style="text-align:left;color: red !important;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
2.16e-29
</td>
<td style="text-align:left;">
5
</td>
</tr>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L9</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M42508
</td>
<td style="text-align:left;">
List of 1130 human and mouse housekeeping genes. This list shows the overlap of human genes stably expressed across 52 tissues and cells types and mouse genes with at least one of their transcripts expressed across 14 tissues and cells types.
</td>
<td style="text-align:left;">
1070
</td>
<td style="text-align:left;">
16
</td>
<td style="text-align:left;">
-0.897
</td>
<td style="text-align:left;">
-1.3
</td>
<td style="text-align:left;color: red !important;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
6.51e-31
</td>
<td style="text-align:left;">
2
</td>
</tr>
</tbody>
</table>
</div>
<div id="prop-genes-0.2-db-all_msigdb-1" class="section level3">
<h3>prop genes = 0.2 , db = all_msigdb</h3>
<p><img src="figure/baboon_diet.Rmd/experiment.level.results-13.png" width="672" style="display: block; margin: auto;" /><img src="figure/baboon_diet.Rmd/experiment.level.results-14.png" width="672" style="display: block; margin: auto;" /></p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
geneSet
</th>
<th style="text-align:left;">
description
</th>
<th style="text-align:left;">
geneSetSize
</th>
<th style="text-align:left;">
overlap
</th>
<th style="text-align:left;">
logOddsRatio
</th>
<th style="text-align:left;">
beta
</th>
<th style="text-align:left;">
alpha
</th>
<th style="text-align:left;">
pip
</th>
<th style="text-align:left;">
pFishersExact
</th>
<th style="text-align:left;">
fisherRank
</th>
</tr>
</thead>
<tbody>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L1</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M8838
</td>
<td style="text-align:left;">
Immune (humoral) and inflammatory response.
</td>
<td style="text-align:left;">
445
</td>
<td style="text-align:left;">
220
</td>
<td style="text-align:left;">
0.619
</td>
<td style="text-align:left;">
0.801
</td>
<td style="text-align:left;color: green !important;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
2.21e-45
</td>
<td style="text-align:left;">
1
</td>
</tr>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L10</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M3268
</td>
<td style="text-align:left;">
Top 200 marker genes up-regulated in the ‘proliferation’ subclass of hepatocellular carcinoma (HCC); characterized by increased proliferation, high levels of serum AFP [GeneID=174], and chromosomal instability.
</td>
<td style="text-align:left;">
163
</td>
<td style="text-align:left;">
69
</td>
<td style="text-align:left;">
0.475
</td>
<td style="text-align:left;">
1.12
</td>
<td style="text-align:left;color: green !important;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
7.14e-11
</td>
<td style="text-align:left;">
176
</td>
</tr>
<tr grouplength="2">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L11</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M40425
</td>
<td style="text-align:left;">
Self-propelled movement of a cell or organism from one location to another. [GOC:dgh]
</td>
<td style="text-align:left;">
1470
</td>
<td style="text-align:left;">
459
</td>
<td style="text-align:left;">
0.297
</td>
<td style="text-align:left;">
0.222
</td>
<td style="text-align:left;color: green !important;">
0.595
</td>
<td style="text-align:left;">
0.595
</td>
<td style="text-align:left;">
1.81e-27
</td>
<td style="text-align:left;">
15
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M40389
</td>
<td style="text-align:left;">
The controlled self-propelled movement of a cell from one site to a destination guided by molecular cues. Cell migration is a central process in the development and maintenance of multicellular organisms. [GOC:cjm, GOC:dph, GOC:ems, GOC:pf, Wikipedia:Cell_migration]
</td>
<td style="text-align:left;">
1220
</td>
<td style="text-align:left;">
398
</td>
<td style="text-align:left;">
0.322
</td>
<td style="text-align:left;">
0.16
</td>
<td style="text-align:left;color: green !important;">
0.403
</td>
<td style="text-align:left;">
0.403
</td>
<td style="text-align:left;">
6.79e-28
</td>
<td style="text-align:left;">
12
</td>
</tr>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L12</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M15549
</td>
<td style="text-align:left;">
Genes turned on in Hep3B cells (hepatocellular carcinoma, HCC) upon knockdown of SDHB [GeneID=6390] by RNAi.
</td>
<td style="text-align:left;">
105
</td>
<td style="text-align:left;">
54
</td>
<td style="text-align:left;">
0.633
</td>
<td style="text-align:left;">
1.19
</td>
<td style="text-align:left;color: green !important;">
0.998
</td>
<td style="text-align:left;">
0.998
</td>
<td style="text-align:left;">
6.35e-13
</td>
<td style="text-align:left;">
114
</td>
</tr>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L13</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M2200
</td>
<td style="text-align:left;">
Genes enriched at every T lymphocyte differentiation stage compared to the early passage fetal thymic stromal cultures (TSC).
</td>
<td style="text-align:left;">
175
</td>
<td style="text-align:left;">
89
</td>
<td style="text-align:left;">
0.628
</td>
<td style="text-align:left;">
0.977
</td>
<td style="text-align:left;color: green !important;">
0.999
</td>
<td style="text-align:left;">
0.999
</td>
<td style="text-align:left;">
5.51e-20
</td>
<td style="text-align:left;">
31
</td>
</tr>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L14</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M15848
</td>
<td style="text-align:left;">
Any process that modulates the frequency, rate or extent of the directed movement of charged atoms or small charged molecules into, out of or within a cell, or between cells, by means of some agent such as a transporter or pore. [GOC:jl]
</td>
<td style="text-align:left;">
513
</td>
<td style="text-align:left;">
191
</td>
<td style="text-align:left;">
0.393
</td>
<td style="text-align:left;">
0.612
</td>
<td style="text-align:left;color: green !important;">
0.997
</td>
<td style="text-align:left;">
0.997
</td>
<td style="text-align:left;">
2.9e-20
</td>
<td style="text-align:left;">
30
</td>
</tr>
<tr grouplength="3">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L15</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M39608
</td>
<td style="text-align:left;">
Cholesterol biosynthesis pathway
</td>
<td style="text-align:left;">
14
</td>
<td style="text-align:left;">
10
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
2.72
</td>
<td style="text-align:left;color: green !important;">
0.909
</td>
<td style="text-align:left;">
0.909
</td>
<td style="text-align:left;">
4.56e-05
</td>
<td style="text-align:left;">
918
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M763
</td>
<td style="text-align:left;">
Genes in the cancer module 432.
</td>
<td style="text-align:left;">
15
</td>
<td style="text-align:left;">
10
</td>
<td style="text-align:left;">
0.904
</td>
<td style="text-align:left;">
0.065
</td>
<td style="text-align:left;color: green !important;">
0.0258
</td>
<td style="text-align:left;">
0.0259
</td>
<td style="text-align:left;">
0.000112
</td>
<td style="text-align:left;">
1090
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M16227
</td>
<td style="text-align:left;">
Cholesterol biosynthesis
</td>
<td style="text-align:left;">
23
</td>
<td style="text-align:left;">
12
</td>
<td style="text-align:left;">
0.641
</td>
<td style="text-align:left;">
0.0359
</td>
<td style="text-align:left;color: green !important;">
0.0175
</td>
<td style="text-align:left;">
0.0176
</td>
<td style="text-align:left;">
0.000591
</td>
<td style="text-align:left;">
1500
</td>
</tr>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L2</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M1920
</td>
<td style="text-align:left;">
Genes forming the macrophage-enriched metabolic network (MEMN) claimed to have a causal relationship with the metabolic syndrom traits.
</td>
<td style="text-align:left;">
1080
</td>
<td style="text-align:left;">
407
</td>
<td style="text-align:left;">
0.426
</td>
<td style="text-align:left;">
0.697
</td>
<td style="text-align:left;color: green !important;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
4.39e-45
</td>
<td style="text-align:left;">
2
</td>
</tr>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L3</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M8513
</td>
<td style="text-align:left;">
Genes up-regulated in the normal-like subtype of breast cancer.
</td>
<td style="text-align:left;">
400
</td>
<td style="text-align:left;">
199
</td>
<td style="text-align:left;">
0.622
</td>
<td style="text-align:left;">
0.916
</td>
<td style="text-align:left;color: green !important;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
1.39e-41
</td>
<td style="text-align:left;">
3
</td>
</tr>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L4</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M8295
</td>
<td style="text-align:left;">
Neighborhood of TTN
</td>
<td style="text-align:left;">
25
</td>
<td style="text-align:left;">
25
</td>
<td style="text-align:left;">
Inf
</td>
<td style="text-align:left;">
5.75
</td>
<td style="text-align:left;color: green !important;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
3.08e-18
</td>
<td style="text-align:left;">
39
</td>
</tr>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L5</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M9232
</td>
<td style="text-align:left;">
The component of the plasma membrane consisting of the gene products and protein complexes having either part of their peptide sequence embedded in the hydrophobic region of the membrane or some other covalently attached group such as a GPI anchor that is similarly embedded in the membrane. [GOC:dos, GOC:mah]
</td>
<td style="text-align:left;">
1240
</td>
<td style="text-align:left;">
404
</td>
<td style="text-align:left;">
0.322
</td>
<td style="text-align:left;">
0.377
</td>
<td style="text-align:left;color: green !important;">
0.997
</td>
<td style="text-align:left;">
0.997
</td>
<td style="text-align:left;">
2.91e-28
</td>
<td style="text-align:left;">
10
</td>
</tr>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L6</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M29955
</td>
<td style="text-align:left;">
Genes containing one or more binding sites for UniProt:Q15723 (ELF2) in their promoter regions (TSS -1000,+100 bp) as identified by GTRD version 20.06 ChIP-seq harmonization.
</td>
<td style="text-align:left;">
1220
</td>
<td style="text-align:left;">
107
</td>
<td style="text-align:left;">
-0.443
</td>
<td style="text-align:left;">
-0.65
</td>
<td style="text-align:left;color: red !important;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
8.23e-29
</td>
<td style="text-align:left;">
7
</td>
</tr>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L7</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M13283
</td>
<td style="text-align:left;">
Liver selective genes
</td>
<td style="text-align:left;">
203
</td>
<td style="text-align:left;">
5
</td>
<td style="text-align:left;">
-1
</td>
<td style="text-align:left;">
-2.44
</td>
<td style="text-align:left;color: red !important;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
7.08e-14
</td>
<td style="text-align:left;">
98
</td>
</tr>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L8</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M17471
</td>
<td style="text-align:left;">
Genes up-regulated in invasive ductal carcinoma (IDC) relative to ductal carcinoma in situ (DCIS, non-invasive).
</td>
<td style="text-align:left;">
314
</td>
<td style="text-align:left;">
159
</td>
<td style="text-align:left;">
0.632
</td>
<td style="text-align:left;">
0.779
</td>
<td style="text-align:left;color: green !important;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
1.62e-34
</td>
<td style="text-align:left;">
4
</td>
</tr>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L9</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M29940
</td>
<td style="text-align:left;">
Genes containing one or more binding sites for UniProt:Q9BTC0 (DIDO1) in their promoter regions (TSS -1000,+100 bp) as identified by GTRD version 20.06 ChIP-seq harmonization.
</td>
<td style="text-align:left;">
1300
</td>
<td style="text-align:left;">
112
</td>
<td style="text-align:left;">
-0.453
</td>
<td style="text-align:left;">
-0.658
</td>
<td style="text-align:left;color: red !important;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
2.16e-31
</td>
<td style="text-align:left;">
6
</td>
</tr>
</tbody>
</table>
</div>
<div id="prop-genes-0.5-db-all_msigdb-1" class="section level3">
<h3>prop genes = 0.5 , db = all_msigdb</h3>
<p><img src="figure/baboon_diet.Rmd/experiment.level.results-15.png" width="672" style="display: block; margin: auto;" /><img src="figure/baboon_diet.Rmd/experiment.level.results-16.png" width="672" style="display: block; margin: auto;" /></p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
geneSet
</th>
<th style="text-align:left;">
description
</th>
<th style="text-align:left;">
geneSetSize
</th>
<th style="text-align:left;">
overlap
</th>
<th style="text-align:left;">
logOddsRatio
</th>
<th style="text-align:left;">
beta
</th>
<th style="text-align:left;">
alpha
</th>
<th style="text-align:left;">
pip
</th>
<th style="text-align:left;">
pFishersExact
</th>
<th style="text-align:left;">
fisherRank
</th>
</tr>
</thead>
<tbody>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L1</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M13283
</td>
<td style="text-align:left;">
Liver selective genes
</td>
<td style="text-align:left;">
203
</td>
<td style="text-align:left;">
27
</td>
<td style="text-align:left;">
-0.824
</td>
<td style="text-align:left;">
-1.74
</td>
<td style="text-align:left;color: red !important;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
1.41e-28
</td>
<td style="text-align:left;">
1
</td>
</tr>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L2</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M1920
</td>
<td style="text-align:left;">
Genes forming the macrophage-enriched metabolic network (MEMN) claimed to have a causal relationship with the metabolic syndrom traits.
</td>
<td style="text-align:left;">
1080
</td>
<td style="text-align:left;">
681
</td>
<td style="text-align:left;">
0.253
</td>
<td style="text-align:left;">
0.535
</td>
<td style="text-align:left;color: green !important;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
1.88e-19
</td>
<td style="text-align:left;">
2
</td>
</tr>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L3</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M8513
</td>
<td style="text-align:left;">
Genes up-regulated in the normal-like subtype of breast cancer.
</td>
<td style="text-align:left;">
400
</td>
<td style="text-align:left;">
279
</td>
<td style="text-align:left;">
0.373
</td>
<td style="text-align:left;">
0.753
</td>
<td style="text-align:left;color: green !important;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
7.12e-16
</td>
<td style="text-align:left;">
8
</td>
</tr>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L4</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M2194
</td>
<td style="text-align:left;">
Genes down-regulated in liver samples of liver-specific knockout of HNF4A [GeneID=3172].
</td>
<td style="text-align:left;">
138
</td>
<td style="text-align:left;">
22
</td>
<td style="text-align:left;">
-0.728
</td>
<td style="text-align:left;">
-1.38
</td>
<td style="text-align:left;color: red !important;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
5.48e-17
</td>
<td style="text-align:left;">
6
</td>
</tr>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L5</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M16932
</td>
<td style="text-align:left;">
Top 200 marker genes down-regulated in the ‘proliferation’ subclass of hepatocellular carcinoma (HCC); characterized by increased proliferation, high levels of serum AFP [GeneID=174], and chromosomal instability.
</td>
<td style="text-align:left;">
148
</td>
<td style="text-align:left;">
23
</td>
<td style="text-align:left;">
-0.742
</td>
<td style="text-align:left;">
-1.31
</td>
<td style="text-align:left;color: red !important;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
1.45e-18
</td>
<td style="text-align:left;">
5
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="muscle" class="section level2">
<h2>Muscle</h2>
<div id="prop-genes-0.05-db-all_msigdb-2" class="section level3">
<h3>prop genes = 0.05 , db = all_msigdb</h3>
<p><img src="figure/baboon_diet.Rmd/experiment.level.results-17.png" width="672" style="display: block; margin: auto;" /><img src="figure/baboon_diet.Rmd/experiment.level.results-18.png" width="672" style="display: block; margin: auto;" /></p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
geneSet
</th>
<th style="text-align:left;">
description
</th>
<th style="text-align:left;">
geneSetSize
</th>
<th style="text-align:left;">
overlap
</th>
<th style="text-align:left;">
logOddsRatio
</th>
<th style="text-align:left;">
beta
</th>
<th style="text-align:left;">
alpha
</th>
<th style="text-align:left;">
pip
</th>
<th style="text-align:left;">
pFishersExact
</th>
<th style="text-align:left;">
fisherRank
</th>
</tr>
</thead>
<tbody>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L1</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M13283
</td>
<td style="text-align:left;">
Liver selective genes
</td>
<td style="text-align:left;">
203
</td>
<td style="text-align:left;">
100
</td>
<td style="text-align:left;">
1.33
</td>
<td style="text-align:left;">
1.95
</td>
<td style="text-align:left;color: green !important;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
5.52e-76
</td>
<td style="text-align:left;">
1
</td>
</tr>
<tr grouplength="2">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L2</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M10917
</td>
<td style="text-align:left;">
Neighborhood of HPN
</td>
<td style="text-align:left;">
108
</td>
<td style="text-align:left;">
70
</td>
<td style="text-align:left;">
1.59
</td>
<td style="text-align:left;">
1.7
</td>
<td style="text-align:left;color: green !important;">
0.827
</td>
<td style="text-align:left;">
0.827
</td>
<td style="text-align:left;">
1.12e-64
</td>
<td style="text-align:left;">
2
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M132
</td>
<td style="text-align:left;">
Neighborhood of LCAT
</td>
<td style="text-align:left;">
101
</td>
<td style="text-align:left;">
65
</td>
<td style="text-align:left;">
1.58
</td>
<td style="text-align:left;">
0.344
</td>
<td style="text-align:left;color: green !important;">
0.165
</td>
<td style="text-align:left;">
0.166
</td>
<td style="text-align:left;">
9.04e-60
</td>
<td style="text-align:left;">
4
</td>
</tr>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L3</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M624
</td>
<td style="text-align:left;">
Genes that that have low expression in mammary tumors of epithelial-mesenchymal transition (EMT) histology.
</td>
<td style="text-align:left;">
106
</td>
<td style="text-align:left;">
34
</td>
<td style="text-align:left;">
0.972
</td>
<td style="text-align:left;">
1.79
</td>
<td style="text-align:left;color: green !important;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
5.71e-19
</td>
<td style="text-align:left;">
37
</td>
</tr>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L4</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M22767
</td>
<td style="text-align:left;">
The directed movement of a C4-dicarboxylate into, out of or within a cell, or between cells, by means of some agent such as a transporter or pore. A C4-dicarboxylate is the anion of a dicarboxylic acid that contains four carbon atoms. [GOC:krc, GOC:mah]
</td>
<td style="text-align:left;">
28
</td>
<td style="text-align:left;">
14
</td>
<td style="text-align:left;">
1.29
</td>
<td style="text-align:left;">
3.05
</td>
<td style="text-align:left;color: green !important;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
1.13e-11
</td>
<td style="text-align:left;">
85
</td>
</tr>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L5</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M38993
</td>
<td style="text-align:left;">
Transcriptome of human HepaRG hepatocellular carcinoma liver progenitors in responses to a WNT3A-enriched microenvironment and dissection of pathways dependent on _-catenin and/or blocked by the SFRP-like Wnt inhibitor FZD8_CRD.
</td>
<td style="text-align:left;">
335
</td>
<td style="text-align:left;">
81
</td>
<td style="text-align:left;">
0.827
</td>
<td style="text-align:left;">
1.31
</td>
<td style="text-align:left;color: green !important;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
6.82e-34
</td>
<td style="text-align:left;">
16
</td>
</tr>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L6</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M13014
</td>
<td style="text-align:left;">
Genes down-regulated in liver tumor compared to the normal adjacent tissue.
</td>
<td style="text-align:left;">
236
</td>
<td style="text-align:left;">
58
</td>
<td style="text-align:left;">
0.823
</td>
<td style="text-align:left;">
1.45
</td>
<td style="text-align:left;color: green !important;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
7.73e-25
</td>
<td style="text-align:left;">
23
</td>
</tr>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L7</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M4306
</td>
<td style="text-align:left;">
Genes down-regulated in HMLE cells (immortalized nontransformed mammary epithelium) after E-cadhedrin (CDH1) [GeneID=999] knockdown by RNAi.
</td>
<td style="text-align:left;">
386
</td>
<td style="text-align:left;">
68
</td>
<td style="text-align:left;">
0.642
</td>
<td style="text-align:left;">
1.21
</td>
<td style="text-align:left;color: green !important;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
3.52e-20
</td>
<td style="text-align:left;">
31
</td>
</tr>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L8</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M10829
</td>
<td style="text-align:left;">
The chemical reactions and pathways involving an alpha-amino acid. [GOC:TermGenie]
</td>
<td style="text-align:left;">
181
</td>
<td style="text-align:left;">
51
</td>
<td style="text-align:left;">
0.901
</td>
<td style="text-align:left;">
1.37
</td>
<td style="text-align:left;color: green !important;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
6.38e-25
</td>
<td style="text-align:left;">
22
</td>
</tr>
</tbody>
</table>
</div>
<div id="prop-genes-0.1-db-all_msigdb-2" class="section level3">
<h3>prop genes = 0.1 , db = all_msigdb</h3>
<p><img src="figure/baboon_diet.Rmd/experiment.level.results-19.png" width="672" style="display: block; margin: auto;" /><img src="figure/baboon_diet.Rmd/experiment.level.results-20.png" width="672" style="display: block; margin: auto;" /></p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
geneSet
</th>
<th style="text-align:left;">
description
</th>
<th style="text-align:left;">
geneSetSize
</th>
<th style="text-align:left;">
overlap
</th>
<th style="text-align:left;">
logOddsRatio
</th>
<th style="text-align:left;">
beta
</th>
<th style="text-align:left;">
alpha
</th>
<th style="text-align:left;">
pip
</th>
<th style="text-align:left;">
pFishersExact
</th>
<th style="text-align:left;">
fisherRank
</th>
</tr>
</thead>
<tbody>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L1</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M13283
</td>
<td style="text-align:left;">
Liver selective genes
</td>
<td style="text-align:left;">
203
</td>
<td style="text-align:left;">
142
</td>
<td style="text-align:left;">
1.37
</td>
<td style="text-align:left;">
2.13
</td>
<td style="text-align:left;color: green !important;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
1.85e-95
</td>
<td style="text-align:left;">
1
</td>
</tr>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L10</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M42508
</td>
<td style="text-align:left;">
List of 1130 human and mouse housekeeping genes. This list shows the overlap of human genes stably expressed across 52 tissues and cells types and mouse genes with at least one of their transcripts expressed across 14 tissues and cells types.
</td>
<td style="text-align:left;">
1070
</td>
<td style="text-align:left;">
21
</td>
<td style="text-align:left;">
-0.775
</td>
<td style="text-align:left;">
-1.16
</td>
<td style="text-align:left;color: red !important;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
6.61e-27
</td>
<td style="text-align:left;">
26
</td>
</tr>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L11</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M2383
</td>
<td style="text-align:left;">
Genes up-regulated in E12.5 forelimb buds with POR [GeneID=5447] knockout.
</td>
<td style="text-align:left;">
24
</td>
<td style="text-align:left;">
14
</td>
<td style="text-align:left;">
1.1
</td>
<td style="text-align:left;">
2.4
</td>
<td style="text-align:left;color: green !important;">
0.977
</td>
<td style="text-align:left;">
0.977
</td>
<td style="text-align:left;">
7.03e-09
</td>
<td style="text-align:left;">
197
</td>
</tr>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L12</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M2098
</td>
<td style="text-align:left;">
Genes up-regulated in NB4 cells (acute promyelocytic leukemia, APL) in response to tretinoin [PubChem=444795]; based on Chip-seq data.
</td>
<td style="text-align:left;">
501
</td>
<td style="text-align:left;">
104
</td>
<td style="text-align:left;">
0.392
</td>
<td style="text-align:left;">
0.753
</td>
<td style="text-align:left;color: green !important;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
3e-13
</td>
<td style="text-align:left;">
80
</td>
</tr>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L2</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M2194
</td>
<td style="text-align:left;">
Genes down-regulated in liver samples of liver-specific knockout of HNF4A [GeneID=3172].
</td>
<td style="text-align:left;">
138
</td>
<td style="text-align:left;">
85
</td>
<td style="text-align:left;">
1.18
</td>
<td style="text-align:left;">
1.79
</td>
<td style="text-align:left;color: green !important;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
3.03e-50
</td>
<td style="text-align:left;">
10
</td>
</tr>
<tr grouplength="2">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L3</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M10917
</td>
<td style="text-align:left;">
Neighborhood of HPN
</td>
<td style="text-align:left;">
108
</td>
<td style="text-align:left;">
92
</td>
<td style="text-align:left;">
1.74
</td>
<td style="text-align:left;">
1.8
</td>
<td style="text-align:left;color: green !important;">
0.81
</td>
<td style="text-align:left;">
0.81
</td>
<td style="text-align:left;">
6.79e-76
</td>
<td style="text-align:left;">
2
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M132
</td>
<td style="text-align:left;">
Neighborhood of LCAT
</td>
<td style="text-align:left;">
101
</td>
<td style="text-align:left;">
86
</td>
<td style="text-align:left;">
1.74
</td>
<td style="text-align:left;">
0.424
</td>
<td style="text-align:left;color: green !important;">
0.187
</td>
<td style="text-align:left;">
0.187
</td>
<td style="text-align:left;">
6.2e-71
</td>
<td style="text-align:left;">
4
</td>
</tr>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L4</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M624
</td>
<td style="text-align:left;">
Genes that that have low expression in mammary tumors of epithelial-mesenchymal transition (EMT) histology.
</td>
<td style="text-align:left;">
106
</td>
<td style="text-align:left;">
44
</td>
<td style="text-align:left;">
0.817
</td>
<td style="text-align:left;">
1.73
</td>
<td style="text-align:left;color: green !important;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
1.52e-17
</td>
<td style="text-align:left;">
46
</td>
</tr>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L5</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M38993
</td>
<td style="text-align:left;">
Transcriptome of human HepaRG hepatocellular carcinoma liver progenitors in responses to a WNT3A-enriched microenvironment and dissection of pathways dependent on _-catenin and/or blocked by the SFRP-like Wnt inhibitor FZD8_CRD.
</td>
<td style="text-align:left;">
335
</td>
<td style="text-align:left;">
126
</td>
<td style="text-align:left;">
0.768
</td>
<td style="text-align:left;">
1.11
</td>
<td style="text-align:left;color: green !important;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
9.58e-43
</td>
<td style="text-align:left;">
16
</td>
</tr>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L6</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M22767
</td>
<td style="text-align:left;">
The directed movement of a C4-dicarboxylate into, out of or within a cell, or between cells, by means of some agent such as a transporter or pore. A C4-dicarboxylate is the anion of a dicarboxylic acid that contains four carbon atoms. [GOC:krc, GOC:mah]
</td>
<td style="text-align:left;">
28
</td>
<td style="text-align:left;">
18
</td>
<td style="text-align:left;">
1.21
</td>
<td style="text-align:left;">
2.77
</td>
<td style="text-align:left;color: green !important;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
4.46e-12
</td>
<td style="text-align:left;">
95
</td>
</tr>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L7</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M1941
</td>
<td style="text-align:left;">
Genes with high-CpG-density promoters (HCP) bearing histone H3 dimethylation at K4 (H3K4me2) and trimethylation at K27 (H3K27me3) in brain.
</td>
<td style="text-align:left;">
882
</td>
<td style="text-align:left;">
163
</td>
<td style="text-align:left;">
0.338
</td>
<td style="text-align:left;">
0.727
</td>
<td style="text-align:left;color: green !important;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
2.07e-15
</td>
<td style="text-align:left;">
59
</td>
</tr>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L8</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M13014
</td>
<td style="text-align:left;">
Genes down-regulated in liver tumor compared to the normal adjacent tissue.
</td>
<td style="text-align:left;">
236
</td>
<td style="text-align:left;">
82
</td>
<td style="text-align:left;">
0.701
</td>
<td style="text-align:left;">
1.15
</td>
<td style="text-align:left;color: green !important;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
2.45e-25
</td>
<td style="text-align:left;">
29
</td>
</tr>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L9</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M16932
</td>
<td style="text-align:left;">
Top 200 marker genes down-regulated in the ‘proliferation’ subclass of hepatocellular carcinoma (HCC); characterized by increased proliferation, high levels of serum AFP [GeneID=174], and chromosomal instability.
</td>
<td style="text-align:left;">
148
</td>
<td style="text-align:left;">
78
</td>
<td style="text-align:left;">
1.02
</td>
<td style="text-align:left;">
1.43
</td>
<td style="text-align:left;color: green !important;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
2.64e-39
</td>
<td style="text-align:left;">
18
</td>
</tr>
</tbody>
</table>
</div>
<div id="prop-genes-0.2-db-all_msigdb-2" class="section level3">
<h3>prop genes = 0.2 , db = all_msigdb</h3>
<p><img src="figure/baboon_diet.Rmd/experiment.level.results-21.png" width="672" style="display: block; margin: auto;" /><img src="figure/baboon_diet.Rmd/experiment.level.results-22.png" width="672" style="display: block; margin: auto;" /></p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
geneSet
</th>
<th style="text-align:left;">
description
</th>
<th style="text-align:left;">
geneSetSize
</th>
<th style="text-align:left;">
overlap
</th>
<th style="text-align:left;">
logOddsRatio
</th>
<th style="text-align:left;">
beta
</th>
<th style="text-align:left;">
alpha
</th>
<th style="text-align:left;">
pip
</th>
<th style="text-align:left;">
pFishersExact
</th>
<th style="text-align:left;">
fisherRank
</th>
</tr>
</thead>
<tbody>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L1</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M13283
</td>
<td style="text-align:left;">
Liver selective genes
</td>
<td style="text-align:left;">
203
</td>
<td style="text-align:left;">
164
</td>
<td style="text-align:left;">
1.25
</td>
<td style="text-align:left;">
1.76
</td>
<td style="text-align:left;color: green !important;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
1.26e-78
</td>
<td style="text-align:left;">
1
</td>
</tr>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L10</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M1872
</td>
<td style="text-align:left;">
Liver genes - metabolism and xenobiotics.
</td>
<td style="text-align:left;">
449
</td>
<td style="text-align:left;">
231
</td>
<td style="text-align:left;">
0.656
</td>
<td style="text-align:left;">
0.72
</td>
<td style="text-align:left;color: green !important;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
1.49e-51
</td>
<td style="text-align:left;">
5
</td>
</tr>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L2</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M2194
</td>
<td style="text-align:left;">
Genes down-regulated in liver samples of liver-specific knockout of HNF4A [GeneID=3172].
</td>
<td style="text-align:left;">
138
</td>
<td style="text-align:left;">
99
</td>
<td style="text-align:left;">
1.02
</td>
<td style="text-align:left;">
1.52
</td>
<td style="text-align:left;color: green !important;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
1.36e-39
</td>
<td style="text-align:left;">
10
</td>
</tr>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L3</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M38993
</td>
<td style="text-align:left;">
Transcriptome of human HepaRG hepatocellular carcinoma liver progenitors in responses to a WNT3A-enriched microenvironment and dissection of pathways dependent on _-catenin and/or blocked by the SFRP-like Wnt inhibitor FZD8_CRD.
</td>
<td style="text-align:left;">
335
</td>
<td style="text-align:left;">
168
</td>
<td style="text-align:left;">
0.625
</td>
<td style="text-align:left;">
0.918
</td>
<td style="text-align:left;color: green !important;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
9.57e-36
</td>
<td style="text-align:left;">
14
</td>
</tr>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L4</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M624
</td>
<td style="text-align:left;">
Genes that that have low expression in mammary tumors of epithelial-mesenchymal transition (EMT) histology.
</td>
<td style="text-align:left;">
106
</td>
<td style="text-align:left;">
59
</td>
<td style="text-align:left;">
0.708
</td>
<td style="text-align:left;">
1.43
</td>
<td style="text-align:left;color: green !important;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
4.65e-16
</td>
<td style="text-align:left;">
35
</td>
</tr>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L5</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M10917
</td>
<td style="text-align:left;">
Neighborhood of HPN
</td>
<td style="text-align:left;">
108
</td>
<td style="text-align:left;">
102
</td>
<td style="text-align:left;">
1.85
</td>
<td style="text-align:left;">
2.56
</td>
<td style="text-align:left;color: green !important;">
0.969
</td>
<td style="text-align:left;">
0.969
</td>
<td style="text-align:left;">
6.09e-64
</td>
<td style="text-align:left;">
2
</td>
</tr>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L6</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M1941
</td>
<td style="text-align:left;">
Genes with high-CpG-density promoters (HCP) bearing histone H3 dimethylation at K4 (H3K4me2) and trimethylation at K27 (H3K27me3) in brain.
</td>
<td style="text-align:left;">
882
</td>
<td style="text-align:left;">
257
</td>
<td style="text-align:left;">
0.233
</td>
<td style="text-align:left;">
0.479
</td>
<td style="text-align:left;color: green !important;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
2.21e-11
</td>
<td style="text-align:left;">
63
</td>
</tr>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L7</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M16932
</td>
<td style="text-align:left;">
Top 200 marker genes down-regulated in the ‘proliferation’ subclass of hepatocellular carcinoma (HCC); characterized by increased proliferation, high levels of serum AFP [GeneID=174], and chromosomal instability.
</td>
<td style="text-align:left;">
148
</td>
<td style="text-align:left;">
98
</td>
<td style="text-align:left;">
0.908
</td>
<td style="text-align:left;">
1.3
</td>
<td style="text-align:left;color: green !important;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
1.62e-34
</td>
<td style="text-align:left;">
16
</td>
</tr>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L8</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M30383
</td>
<td style="text-align:left;">
Genes containing one or more binding sites for UniProt:Q9Y462 (ZNF711) in their promoter regions (TSS -1000,+100 bp) as identified by GTRD version 20.06 ChIP-seq harmonization.
</td>
<td style="text-align:left;">
1350
</td>
<td style="text-align:left;">
157
</td>
<td style="text-align:left;">
-0.301
</td>
<td style="text-align:left;">
-0.529
</td>
<td style="text-align:left;color: red !important;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
2.42e-17
</td>
<td style="text-align:left;">
32
</td>
</tr>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L9</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M13014
</td>
<td style="text-align:left;">
Genes down-regulated in liver tumor compared to the normal adjacent tissue.
</td>
<td style="text-align:left;">
236
</td>
<td style="text-align:left;">
113
</td>
<td style="text-align:left;">
0.578
</td>
<td style="text-align:left;">
0.919
</td>
<td style="text-align:left;color: green !important;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
3.06e-22
</td>
<td style="text-align:left;">
26
</td>
</tr>
</tbody>
</table>
</div>
<div id="prop-genes-0.5-db-all_msigdb-2" class="section level3">
<h3>prop genes = 0.5 , db = all_msigdb</h3>
<p><img src="figure/baboon_diet.Rmd/experiment.level.results-23.png" width="672" style="display: block; margin: auto;" /><img src="figure/baboon_diet.Rmd/experiment.level.results-24.png" width="672" style="display: block; margin: auto;" /></p>
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
geneSet
</th>
<th style="text-align:left;">
description
</th>
<th style="text-align:left;">
geneSetSize
</th>
<th style="text-align:left;">
overlap
</th>
<th style="text-align:left;">
logOddsRatio
</th>
<th style="text-align:left;">
beta
</th>
<th style="text-align:left;">
alpha
</th>
<th style="text-align:left;">
pip
</th>
<th style="text-align:left;">
pFishersExact
</th>
<th style="text-align:left;">
fisherRank
</th>
</tr>
</thead>
<tbody>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L1</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M13283
</td>
<td style="text-align:left;">
Liver selective genes
</td>
<td style="text-align:left;">
203
</td>
<td style="text-align:left;">
182
</td>
<td style="text-align:left;">
0.948
</td>
<td style="text-align:left;">
1.83
</td>
<td style="text-align:left;color: green !important;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
1.32e-33
</td>
<td style="text-align:left;">
1
</td>
</tr>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L2</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M2194
</td>
<td style="text-align:left;">
Genes down-regulated in liver samples of liver-specific knockout of HNF4A [GeneID=3172].
</td>
<td style="text-align:left;">
138
</td>
<td style="text-align:left;">
117
</td>
<td style="text-align:left;">
0.752
</td>
<td style="text-align:left;">
1.33
</td>
<td style="text-align:left;color: green !important;">
1
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
1.71e-17
</td>
<td style="text-align:left;">
9
</td>
</tr>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L3</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M6298
</td>
<td style="text-align:left;">
Fibers, composed of actin, myosin, and associated proteins, found in cells of smooth or striated muscle. [GOC:go_curators, <a href="ISBN:0815316194" class="uri">ISBN:0815316194</a>]
</td>
<td style="text-align:left;">
212
</td>
<td style="text-align:left;">
60
</td>
<td style="text-align:left;">
-0.41
</td>
<td style="text-align:left;">
-0.891
</td>
<td style="text-align:left;color: red !important;">
0.999
</td>
<td style="text-align:left;">
0.999
</td>
<td style="text-align:left;">
1.09e-10
</td>
<td style="text-align:left;">
20
</td>
</tr>
<tr grouplength="1">
<td colspan="10" style="border-bottom: 1px solid;">
<strong>L4</strong>
</td>
</tr>
<tr>
<td style="text-align:left;padding-left: 2em;" indentlevel="1">
M38993
</td>
<td style="text-align:left;">
Transcriptome of human HepaRG hepatocellular carcinoma liver progenitors in responses to a WNT3A-enriched microenvironment and dissection of pathways dependent on _-catenin and/or blocked by the SFRP-like Wnt inhibitor FZD8_CRD.
</td>
<td style="text-align:left;">
335
</td>
<td style="text-align:left;">
238
</td>
<td style="text-align:left;">
0.399
</td>
<td style="text-align:left;">
0.654
</td>
<td style="text-align:left;color: green !important;">
0.99
</td>
<td style="text-align:left;">
0.99
</td>
<td style="text-align:left;">
3.91e-15
</td>
<td style="text-align:left;">
12
</td>
</tr>
</tbody>
</table>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>#&gt; R version 4.1.0 (2021-05-18)
#&gt; Platform: x86_64-pc-linux-gnu (64-bit)
#&gt; Running under: Scientific Linux 7.4 (Nitrogen)
#&gt; 
#&gt; Matrix products: default
#&gt; BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so
#&gt; 
#&gt; locale:
#&gt; [1] en_US.UTF-8
#&gt; 
#&gt; attached base packages:
#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     
#&gt; 
#&gt; other attached packages:
#&gt;  [1] kableExtra_1.3.4     targets_0.12.0       forcats_0.5.1       
#&gt;  [4] stringr_1.4.0        dplyr_1.0.9          purrr_0.3.4         
#&gt;  [7] readr_2.1.2          tidyr_1.1.3          tibble_3.1.7        
#&gt; [10] tidyverse_1.3.1      ggplot2_3.3.6        gseasusie_0.0.0.9000
#&gt; 
#&gt; loaded via a namespace (and not attached):
#&gt;  [1] httr_1.4.2        sass_0.4.0        viridisLite_0.4.0 bit64_4.0.5      
#&gt;  [5] jsonlite_1.8.0    modelr_0.1.8      bslib_0.2.5.1     assertthat_0.2.1 
#&gt;  [9] highr_0.9         base64url_1.4     cellranger_1.1.0  yaml_2.2.1       
#&gt; [13] pillar_1.7.0      backports_1.2.1   glue_1.6.2        digest_0.6.29    
#&gt; [17] promises_1.2.0.1  rvest_1.0.0       colorspace_2.0-3  htmltools_0.5.1.1
#&gt; [21] httpuv_1.6.1      pkgconfig_2.0.3   broom_0.7.8       haven_2.4.1      
#&gt; [25] webshot_0.5.2     scales_1.2.0      processx_3.5.2    svglite_2.1.0    
#&gt; [29] whisker_0.4       later_1.2.0       tzdb_0.3.0        git2r_0.28.0     
#&gt; [33] generics_0.1.2    farver_2.1.0      ellipsis_0.3.2    withr_2.5.0      
#&gt; [37] cli_3.3.0         magrittr_2.0.3    crayon_1.5.1      readxl_1.3.1     
#&gt; [41] evaluate_0.15     ps_1.6.0          fs_1.5.0          fansi_1.0.3      
#&gt; [45] xml2_1.3.2        tools_4.1.0       data.table_1.14.0 hms_1.1.1        
#&gt; [49] lifecycle_1.0.1   munsell_0.5.0     reprex_2.0.0      callr_3.7.0      
#&gt; [53] compiler_4.1.0    jquerylib_0.1.4   systemfonts_1.0.4 rlang_1.0.2      
#&gt; [57] grid_4.1.0        rstudioapi_0.13   igraph_1.2.6      labeling_0.4.2   
#&gt; [61] rmarkdown_2.9     gtable_0.3.0      codetools_0.2-18  DBI_1.1.1        
#&gt; [65] R6_2.5.1          lubridate_1.7.10  knitr_1.39        bit_4.0.4        
#&gt; [69] utf8_1.2.2        workflowr_1.6.2   rprojroot_2.0.2   stringi_1.7.6    
#&gt; [73] Rcpp_1.0.8.3      vctrs_0.4.1       dbplyr_2.1.1      tidyselect_1.1.2 
#&gt; [77] xfun_0.30</code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
