<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="karltayeb" />

<meta name="date" content="2022-03-16" />

<title>Single-cell PBMC Example</title>

<script src="site_libs/header-attrs-2.13/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">logistic-susie-gsea</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/karltayeb/logistic-susie-gsea">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Single-cell PBMC Example</h1>
<h4 class="author">karltayeb</h4>
<h4 class="date">2022-03-16</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-03-29
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>logistic-susie-gsea/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220105code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20220105)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220105code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220105)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomkarltayeblogisticsusiegseatree6143c4434f0e139fa367ece633d114fd392c0ec4targetblank6143c44a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/karltayeb/logistic-susie-gsea/tree/6143c4434f0e139fa367ece633d114fd392c0ec4" target="_blank">6143c44</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomkarltayeblogisticsusiegseatree6143c4434f0e139fa367ece633d114fd392c0ec4targetblank6143c44a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/karltayeb/logistic-susie-gsea/tree/6143c4434f0e139fa367ece633d114fd392c0ec4" target="_blank">6143c44</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    library/
    Ignored:    renv/library/
    Ignored:    renv/staging/
    Ignored:    staging/

Untracked files:
    Untracked:  _targets.R
    Untracked:  _targets.html
    Untracked:  _targets.md
    Untracked:  _targets/
    Untracked:  _targets_r/
    Untracked:  analysis/fetal_reference_cellid_gsea.Rmd
    Untracked:  analysis/fixed_intercept.Rmd
    Untracked:  analysis/iDEA_examples.Rmd
    Untracked:  analysis/latent_gene_list.Rmd
    Untracked:  analysis/latent_logistic_susie.Rmd
    Untracked:  analysis/libra_setup.Rmd
    Untracked:  analysis/linear_method_failure_modes.Rmd
    Untracked:  analysis/linear_regression_failure_regime.Rmd
    Untracked:  analysis/logistic_susie_veb_boost_vs_vb.Rmd
    Untracked:  analysis/references.bib
    Untracked:  analysis/simulations.Rmd
    Untracked:  analysis/test.Rmd
    Untracked:  analysis/wenhe_baboon_example.Rmd
    Untracked:  build_site.R
    Untracked:  cache/
    Untracked:  code/latent_logistic_susie.R
    Untracked:  code/marginal_sumstat_gsea_collapsed.R
    Untracked:  data/adipose_2yr_topsnp.txt
    Untracked:  data/fetal_reference_cellid_gene_sets.RData
    Untracked:  data/pbmc-purified/
    Untracked:  docs.zip
    Untracked:  index.md
    Untracked:  latent_logistic_susie_cache/
    Untracked:  simulation_targets/
    Untracked:  single_cell_pbmc_cache/
    Untracked:  summary_stat_gsea_exploration_cache/

Unstaged changes:
    Modified:   _simulation_targets.R
    Modified:   _targets.Rmd
    Modified:   analysis/gseabenchmark_tcga.Rmd
    Modified:   code/fit_baselines.R
    Modified:   code/fit_logistic_susie.R
    Modified:   code/fit_mr_ash.R
    Modified:   code/fit_susie.R
    Modified:   code/load_gene_sets.R
    Modified:   code/marginal_sumstat_gsea.R
    Modified:   code/simulate_gene_lists.R
    Modified:   code/utils.R
    Modified:   target_components/factories.R
    Modified:   target_components/methods.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/single_cell_pbmc.Rmd</code>) and HTML (<code>docs/single_cell_pbmc.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/karltayeb/logistic-susie-gsea/blob/6143c4434f0e139fa367ece633d114fd392c0ec4/analysis/single_cell_pbmc.Rmd" target="_blank">6143c44</a>
</td>
<td>
karltayeb
</td>
<td>
2022-03-29
</td>
<td>
wflow_publish(“analysis/single_cell_pbmc.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/karltayeb/logistic-susie-gsea/56f81300a9dcac6d44f597108b7cf24e948e6459/docs/single_cell_pbmc.html" target="_blank">56f8130</a>
</td>
<td>
karltayeb
</td>
<td>
2022-03-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/karltayeb/logistic-susie-gsea/a2bdb5680eea3d5caf14d0b7423351c4232628b4/docs/single_cell_pbmc.html" target="_blank">a2bdb56</a>
</td>
<td>
karltayeb
</td>
<td>
2022-03-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/karltayeb/logistic-susie-gsea/blob/122deece85c32abc1d72b089a34a2f24c62d8972/analysis/single_cell_pbmc.Rmd" target="_blank">122deec</a>
</td>
<td>
karltayeb
</td>
<td>
2022-03-29
</td>
<td>
wflow_publish(pages)
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>Our goals here are to run Logistic SuSiE on differential expression results from TCGA. We want to assess:</p>
<ol style="list-style-type: decimal">
<li>If the resulting enrichment results look good/interpretable across multiple/concatenated gene sets</li>
<li>Assess sensitivity to a range of p-value thresholds</li>
<li>Evaluate the potential of the summary stat latent model</li>
</ol>
<pre class="r"><code>library(GSEABenchmarkeR)
library(EnrichmentBrowser)
library(tidyverse)
library(susieR)
library(DT)
source(&#39;code/load_gene_sets.R&#39;)
source(&#39;code/utils.R&#39;)
source(&#39;code/logistic_susie_vb.R&#39;)
source(&#39;code/logistic_susie_veb_boost.R&#39;)
source(&#39;code/latent_logistic_susie.R&#39;)</code></pre>
</div>
<div id="setup" class="section level2">
<h2>Setup</h2>
<div id="load-gene-sets" class="section level3">
<h3>Load Gene Sets</h3>
<p><code>loadGeneSetX</code> uniformly formats gene sets and generates the <span class="math inline">\(X\)</span> matrix We can source any gene set from <code>WebGestaltR::listGeneSet()</code></p>
<pre class="r"><code>gs_list &lt;- WebGestaltR::listGeneSet()
gobp &lt;- loadGeneSetX(&#39;geneontology_Biological_Process&#39;, min.size=50)  # just huge number of gene sets
gobp_nr &lt;- loadGeneSetX(&#39;geneontology_Biological_Process_noRedundant&#39;, min.size=1)
gomf &lt;- loadGeneSetX(&#39;geneontology_Molecular_Function&#39;, min.size=1)
kegg &lt;- loadGeneSetX(&#39;pathway_KEGG&#39;, min.size=1)
reactome &lt;- loadGeneSetX(&#39;pathway_Reactome&#39;, min.size=1)
wikipathway_cancer &lt;- loadGeneSetX(&#39;pathway_Wikipathway_cancer&#39;, min.size=1)
wikipathway &lt;- loadGeneSetX(&#39;pathway_Wikipathway&#39;, min.size=1)

genesets &lt;- list(
  gobp=gobp,
  gobp_nr=gobp_nr,
  gomf=gomf,
  kegg=kegg,
  reactome=reactome,
  wikipathway_cancer=wikipathway_cancer,
  wikipathway=wikipathway
)</code></pre>
<pre class="r"><code>load(&#39;data/pbmc-purified/deseq2-pbmc-purified.RData&#39;)

convert_labels &lt;- function(y, from=&#39;SYMBOL&#39;, to=&#39;ENTREZID&#39;){
  hs &lt;- org.Hs.eg.db::org.Hs.eg.db
  gene_symbols &lt;- names(y)
  symbol2entrez &lt;- AnnotationDbi::select(hs, keys=gene_symbols, columns=c(to, from), keytype = from)
  symbol2entrez &lt;- symbol2entrez[!duplicated(symbol2entrez[[from]]),]
  symbol2entrez &lt;- symbol2entrez[!is.na(symbol2entrez[[to]]),]
  symbol2entrez &lt;- symbol2entrez[!is.na(symbol2entrez[[from]]),]
  rownames(symbol2entrez) &lt;- symbol2entrez[[from]]
  ysub &lt;- y[names(y) %in% symbol2entrez[[from]]]
  names(ysub) &lt;- symbol2entrez[names(ysub),][[to]]
  return(ysub)
}


par(mfrow=c(1,1))
deseq$`CD19+ B` %&gt;% .$padj %&gt;% hist(main=&#39;CD19+B p-values&#39;)</code></pre>
<pre><code>Loading required package: DESeq2</code></pre>
<p><img src="figure/single_cell_pbmc.Rmd/data.setup-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-data.setup-1">
Past versions of data.setup-1.png
</button>
</p>
<div id="fig-data.setup-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/karltayeb/logistic-susie-gsea/blob/a2bdb5680eea3d5caf14d0b7423351c4232628b4/docs/figure/single_cell_pbmc.Rmd/data.setup-1.png" target="_blank">a2bdb56</a>
</td>
<td>
karltayeb
</td>
<td>
2022-03-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="fit-logistic-susie" class="section level3">
<h3>Fit logistic SuSiE</h3>
<pre class="r"><code>logistic_susie_driver = function(db, celltype, thresh){
  gs &lt;- genesets[[db]]
  data &lt;- deseq[[celltype]]
  
  # set up binary y
  y &lt;- data %&gt;%
    as.data.frame %&gt;%
    rownames_to_column(&#39;gene&#39;) %&gt;%
    dplyr::select(gene, padj) %&gt;%
    filter(!is.na(padj)) %&gt;%
    mutate(y = as.integer(padj &lt; thresh)) %&gt;%
    select(gene, y) %&gt;%
    tibble2namedlist %&gt;%
    convert_labels(&#39;ENSEMBL&#39;)
  
  u &lt;- process_input(gs$X, y)  # subset to common genes
  vb.fit &lt;- logistic.susie(  # fit model
    u$X, u$y, L=10, init.intercept = 0, verbose=1, maxit=100)

  # summarise results
  set.summary &lt;- vb.fit$pip %&gt;% 
    as_tibble(rownames=&#39;geneSet&#39;) %&gt;%
    rename(pip=value) %&gt;%
    mutate(
      top_component = apply(vb.fit$alpha, 2, which.max),
      active_set = top_component %in% vb.fit$sets$cs_index,
      top_component = paste0(&#39;L&#39;, top_component),
      cs = purrr::map(top_component, ~tryCatch(
        colnames(gs$X)[get(.x, vb.fit$sets$cs)], error = function(e) list())),
      in_cs = geneSet %in% cs,
      beta = colSums(vb.fit$mu * vb.fit$alpha),
      geneListSize = sum(u$y),
      geneSetSize = colSums(u$X),
      overlap = (u$y %*% u$X)[1,],
      nGenes = length(u$y),
      propSetInList = overlap / geneSetSize,
      oddsRatio = (overlap / (geneListSize - overlap)) / (
        (geneSetSize - overlap) / (nGenes - geneSetSize + overlap)),
    pValueHypergeometric = phyper(
      overlap-1, geneListSize, nGenes, geneSetSize, lower.tail= FALSE),
    db = db,
    celltype = celltype,
    thresh = thresh
    ) %&gt;% left_join(gs$geneSet$geneSetDes)
  return(list(fit = vb.fit, set.summary=set.summary))
}</code></pre>
<p>For each celltype, we fit logistic SuSiE using multiple gene set sources at various threshold of <code>padj</code>.</p>
<pre class="r"><code>celltypes &lt;- names(deseq)
pthresh &lt;- c(0.1, 0.01, 0.001, 0.0001, 0.00001, 0.000001)
db_name &lt;- names(genesets)
crossed &lt;- cross3(db_name, celltypes, pthresh)

pbmc_res &lt;- xfun::cache_rds({
  res &lt;- purrr::map(crossed, purrr::lift_dl(logistic_susie_driver))
  for (i in 1:length(res)){  # save some space
    res[[i]]$fit$dat &lt;- NULL
  }
  res
  }, file = &#39;logistic_susie_pbmc_genesets_pthresh.rds&#39;
)

pbmc_res_set_summary &lt;- dplyr::bind_rows(purrr::map(pbmc_res, ~ pluck(.x, &#39;set.summary&#39;)))</code></pre>
</div>
<div id="summary-functions" class="section level3">
<h3>Summary functions</h3>
<p>Just a few functions to help streamline looking at output</p>
<pre class="r"><code>pval_focussed_table = function(thresh=1e-3, filter_db=NULL, filter_celltype=NULL, top.n=50){
  pbmc_res_set_summary %&gt;%
  filter(
    case_when(
      is.null(filter_db) ~ TRUE,
      !is.null(filter_db) ~ db %in% filter_db
    ) &amp;
    thresh == thresh &amp;
    case_when(
      is.null(filter_celltype) ~ TRUE,
      !is.null(filter_celltype) ~ celltype %in% filter_celltype
    )
  )  %&gt;%
  dplyr::arrange(celltype, db, pValueHypergeometric) %&gt;%
  group_by(celltype, db) %&gt;% slice(1:top.n) %&gt;%
  select(celltype, db, geneSet, description, pip, top_component, oddsRatio, propSetInList, pValueHypergeometric) %&gt;%
  mutate_at(vars(celltype, db), factor) %&gt;%
  datatable(filter = &#39;top&#39;)
}

set_focussed_table = function(thresh=1e-3, filter_db=NULL, filter_celltype=NULL){
  pbmc_res_set_summary %&gt;%
  filter(
    case_when(
      is.null(filter_db) ~ TRUE,
      !is.null(filter_db) ~ db %in% filter_db
    ) &amp;
    thresh == 1e-3 &amp;
    in_cs &amp; active_set &amp;
    case_when(
      is.null(filter_celltype) ~ TRUE,
      !is.null(filter_celltype) ~ celltype %in% filter_celltype
    )
  )  %&gt;%
  dplyr::arrange(celltype, db, desc(pip)) %&gt;%
  select(celltype, db, geneSet, description, pip, top_component, oddsRatio, propSetInList, pValueHypergeometric) %&gt;%
  mutate_at(vars(celltype, geneSet, db), factor) %&gt;%
  datatable(filter = &#39;top&#39;)
}</code></pre>
</div>
</div>
<div id="resultsexplore-enrichments" class="section level2">
<h2>Results/Explore enrichments</h2>
<p>Our goal is to assess 1. The quality of the gene set enrichments we get from each celltype - do reported gene set enrichments seem celltype specific/celltype relevant? - how much “interesting” marginal enrichment do we fail to capture in the multivariate model - how sensitive are we to the choice of pvalue threshold</p>
</div>
<div id="results" class="section level2">
<h2>Results</h2>
<p>Lets take a look at what enrichment we’re getting across cell-types.</p>
<div id="cd19-b" class="section level3">
<h3>CD19+ B</h3>
<pre class="r"><code>pbmc_res_set_summary %&gt;%
  filter(active_set, in_cs, thresh == 1e-4, celltype == &#39;CD19+ B&#39;) %&gt;%
  select(
    geneSet, description, top_component, pip,
    beta, geneListSize, geneSetSize, propSetInList, pValueHypergeometric)</code></pre>
<pre><code># A tibble: 17 × 9
   geneSet      description  top_component   pip   beta geneListSize geneSetSize
   &lt;chr&gt;        &lt;chr&gt;        &lt;chr&gt;         &lt;dbl&gt;  &lt;dbl&gt;        &lt;int&gt;       &lt;dbl&gt;
 1 GO:0001775   cell activa… L5            0.978  0.467         6911        1128
 2 GO:0002376   immune syst… L1            1.00   0.508         6911        2228
 3 GO:0045047   protein tar… L2            0.962  2.46          6911         105
 4 GO:0002764   immune resp… L5            0.996  0.565         6327         411
 5 GO:0009123   nucleoside … L4            0.984  0.742         6327         286
 6 GO:0070972   protein loc… L2            1.00   1.63          6327         132
 7 GO:0003723   RNA binding  L1            1.00   0.353         6975        1483
 8 GO:0000981   DNA-binding… L2            0.995 -0.657         6975        1161
 9 GO:0003735   structural … L6            0.995  0.964         6975         152
10 hsa00190     Oxidative p… L1            0.998  1.46          3273         120
11 hsa03010     Ribosome     L3            1.00   1.41          3273         129
12 hsa04640     Hematopoiet… L2            1.00   1.50          3273          80
13 R-HSA-168256 Immune Syst… L1            1      0.749         4620        1627
14 R-HSA-983168 Antigen pro… L5            0.997 -1.07          4620         280
15 WP619        Type II int… L1            1.00   2.49           966          30
16 WP111        Electron Tr… L2            1.00   1.64          3220         101
17 WP477        Cytoplasmic… L1            1.00   2.40          3220          85
# … with 2 more variables: propSetInList &lt;dbl&gt;, pValueHypergeometric &lt;dbl&gt;</code></pre>
</div>
<div id="cd56-nk" class="section level3">
<h3>CD56+ NK</h3>
<pre class="r"><code>pbmc_res_set_summary %&gt;%
  filter(active_set, in_cs, thresh == 1e-4, celltype == &#39;CD56+ NK&#39;) %&gt;%
  select(
    geneSet, description, top_component, pip,
    beta, geneListSize, geneSetSize, propSetInList, pValueHypergeometric)</code></pre>
<pre><code># A tibble: 16 × 9
   geneSet       description top_component   pip   beta geneListSize geneSetSize
   &lt;chr&gt;         &lt;chr&gt;       &lt;chr&gt;         &lt;dbl&gt;  &lt;dbl&gt;        &lt;int&gt;       &lt;dbl&gt;
 1 GO:0002376    immune sys… L1            1.00   0.653         7231        2228
 2 GO:0006119    oxidative … L4            0.982  1.32          7231         118
 3 GO:0006413    translatio… L2            0.995  1.16          6598         182
 4 GO:0009123    nucleoside… L4            0.982  0.674         6598         286
 5 GO:0042110    T cell act… L3            0.986  0.591         6598         389
 6 GO:0042113    B cell act… L5            0.997  0.842         6598         198
 7 GO:0000981    DNA-bindin… L2            0.997 -0.734         7305        1161
 8 GO:0003735    structural… L1            1.00   1.90          7305         152
 9 hsa03010      Ribosome    L1            1.00   1.78          3406         129
10 hsa04640      Hematopoie… L3            0.998  1.31          3406          80
11 hsa05012      Parkinson … L2            1.00   1.56          3406         126
12 R-HSA-168256  Immune Sys… L1            1.00   0.689         4855        1627
13 R-HSA-8878171 Transcript… L5            0.971  0.966         4855         189
14 R-HSA-163200  Respirator… L4            0.993  1.32          4855         121
15 WP111         Electron T… L2            0.991  1.32          3316         101
16 WP477         Cytoplasmi… L1            1      2.97          3316          85
# … with 2 more variables: propSetInList &lt;dbl&gt;, pValueHypergeometric &lt;dbl&gt;</code></pre>
</div>
<div id="t-cell" class="section level3">
<h3>T cell</h3>
<pre class="r"><code>pbmc_res_set_summary %&gt;%
  filter(active_set, in_cs, thresh == 1e-4, celltype == &#39;T cell&#39;) %&gt;%
  select(
    geneSet, description, top_component, pip,
    beta, geneListSize, geneSetSize, propSetInList, pValueHypergeometric)</code></pre>
<pre><code># A tibble: 8 × 9
  geneSet       description   top_component   pip  beta geneListSize geneSetSize
  &lt;chr&gt;         &lt;chr&gt;         &lt;chr&gt;         &lt;dbl&gt; &lt;dbl&gt;        &lt;int&gt;       &lt;dbl&gt;
1 GO:0001775    cell activat… L1            0.998 0.753         8802        1127
2 GO:0002376    immune syste… L5            0.998 0.424         8802        2227
3 GO:0006119    oxidative ph… L2            0.999 1.79          8802         118
4 GO:0042110    T cell activ… L2            1.00  0.921         7959         389
5 GO:0070972    protein loca… L3            1.00  1.33          7959         132
6 GO:0005515    protein bind… L1            1.00  0.370         8880        9033
7 hsa05010      Alzheimer di… L1            1.00  1.68          3983         154
8 R-HSA-6798695 Neutrophil d… L1            1.00  1.37          5790         431
# … with 2 more variables: propSetInList &lt;dbl&gt;, pValueHypergeometric &lt;dbl&gt;</code></pre>
</div>
<div id="cd14-monocyte" class="section level3">
<h3>CD14+ Monocyte</h3>
<pre class="r"><code>pbmc_res_set_summary %&gt;%
  filter(active_set, in_cs, thresh == 1e-4, celltype == &#39;CD14+ Monocyte&#39;) %&gt;%
  select(
    geneSet, description, top_component, pip,
    beta, geneListSize, geneSetSize, propSetInList, pValueHypergeometric)</code></pre>
<pre><code># A tibble: 17 × 9
   geneSet       description top_component   pip   beta geneListSize geneSetSize
   &lt;chr&gt;         &lt;chr&gt;       &lt;chr&gt;         &lt;dbl&gt;  &lt;dbl&gt;        &lt;int&gt;       &lt;dbl&gt;
 1 GO:0016192    vesicle-me… L4            0.985  0.324         6633        1605
 2 GO:0006119    oxidative … L3            0.999  1.79          6633         118
 3 GO:0006413    translatio… L2            1.00   1.21          6073         182
 4 GO:0009123    nucleoside… L3            0.988  0.905         6073         286
 5 GO:0036230    granulocyt… L1            0.976  1.17          6073         450
 6 GO:0003723    RNA binding L1            1.00   0.416         6697        1480
 7 GO:0000981    DNA-bindin… L2            1.00  -0.633         6697        1160
 8 GO:0003735    structural… L4            1.00   1.61          6697         152
 9 hsa03010      Ribosome    L1            1.00   1.75          3110         129
10 hsa05012      Parkinson … L2            1.00   1.70          3110         125
11 R-HSA-6798695 Neutrophil… L1            1.00   1.21          4451         431
12 R-HSA-198933  Immunoregu… L6            0.997  1.19          4451         103
13 R-HSA-72766   Translation L2            1.00   1.66          4451         286
14 R-HSA-163200  Respirator… L3            1.00   1.71          4451         121
15 R-HSA-379726  Mitochondr… L7            0.997 -2.40          4451          21
16 WP111         Electron T… L2            1.00   1.76          3062         100
17 WP477         Cytoplasmi… L1            1.00   2.53          3062          85
# … with 2 more variables: propSetInList &lt;dbl&gt;, pValueHypergeometric &lt;dbl&gt;</code></pre>
</div>
<div id="cd34" class="section level3">
<h3>CD34+</h3>
<pre class="r"><code>pbmc_res_set_summary %&gt;%
  filter(active_set, in_cs, thresh == 1e-4, celltype == &#39;CD34+&#39;) %&gt;%
  select(
    geneSet, description, top_component, pip,
    beta, geneListSize, geneSetSize, propSetInList, pValueHypergeometric)</code></pre>
<pre><code># A tibble: 13 × 9
   geneSet       description  top_component   pip  beta geneListSize geneSetSize
   &lt;chr&gt;         &lt;chr&gt;        &lt;chr&gt;         &lt;dbl&gt; &lt;dbl&gt;        &lt;int&gt;       &lt;dbl&gt;
 1 GO:0001775    cell activa… L1            0.987 0.731         8363        1128
 2 GO:0006119    oxidative p… L3            0.999 1.56          8363         118
 3 GO:0006413    translation… L2            1.00  1.07          7544         182
 4 GO:0005515    protein bin… L2            1.00  0.337         8421        9031
 5 GO:0003735    structural … L1            1.00  1.76          8421         152
 6 hsa00190      Oxidative p… L1            0.992 1.51          3772         120
 7 hsa03010      Ribosome     L2            1.00  1.52          3772         129
 8 R-HSA-6798695 Neutrophil … L2            0.986 0.779         5481         432
 9 R-HSA-198933  Immunoregul… L4            1.00  1.43          5481         103
10 R-HSA-163200  Respiratory… L3            0.998 1.57          5481         121
11 R-HSA-72764   Eukaryotic … L1            0.993 3.41          5481          90
12 WP111         Electron Tr… L2            1.00  2.13          3698         101
13 WP477         Cytoplasmic… L1            1     3.25          3698          85
# … with 2 more variables: propSetInList &lt;dbl&gt;, pValueHypergeometric &lt;dbl&gt;</code></pre>
<pre class="r"><code>knitr::knit_exit()</code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
