<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="karltayeb" />

<meta name="date" content="2022-03-16" />

<title>Single-cell PBMC Example</title>

<script src="site_libs/header-attrs-2.13/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">logistic-susie-gsea</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/karltayeb/logistic-susie-gsea">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Single-cell PBMC Example</h1>
<h4 class="author">karltayeb</h4>
<h4 class="date">2022-03-16</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-03-29
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>logistic-susie-gsea/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220105code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20220105)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220105code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220105)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomkarltayeblogisticsusiegseatreefc1f3c1d433ec35d805a712a30e006a116027e5atargetblankfc1f3c1a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/karltayeb/logistic-susie-gsea/tree/fc1f3c1d433ec35d805a712a30e006a116027e5a" target="_blank">fc1f3c1</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomkarltayeblogisticsusiegseatreefc1f3c1d433ec35d805a712a30e006a116027e5atargetblankfc1f3c1a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/karltayeb/logistic-susie-gsea/tree/fc1f3c1d433ec35d805a712a30e006a116027e5a" target="_blank">fc1f3c1</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    library/
    Ignored:    renv/library/
    Ignored:    renv/staging/
    Ignored:    staging/

Untracked files:
    Untracked:  _targets.R
    Untracked:  _targets.html
    Untracked:  _targets.md
    Untracked:  _targets/
    Untracked:  _targets_r/
    Untracked:  analysis/fetal_reference_cellid_gsea.Rmd
    Untracked:  analysis/fixed_intercept.Rmd
    Untracked:  analysis/iDEA_examples.Rmd
    Untracked:  analysis/latent_gene_list.Rmd
    Untracked:  analysis/latent_logistic_susie.Rmd
    Untracked:  analysis/libra_setup.Rmd
    Untracked:  analysis/linear_method_failure_modes.Rmd
    Untracked:  analysis/linear_regression_failure_regime.Rmd
    Untracked:  analysis/logistic_susie_veb_boost_vs_vb.Rmd
    Untracked:  analysis/references.bib
    Untracked:  analysis/simulations.Rmd
    Untracked:  analysis/test.Rmd
    Untracked:  analysis/wenhe_baboon_example.Rmd
    Untracked:  build_site.R
    Untracked:  cache/
    Untracked:  code/latent_logistic_susie.R
    Untracked:  code/marginal_sumstat_gsea_collapsed.R
    Untracked:  data/adipose_2yr_topsnp.txt
    Untracked:  data/fetal_reference_cellid_gene_sets.RData
    Untracked:  data/pbmc-purified/
    Untracked:  docs.zip
    Untracked:  index.md
    Untracked:  latent_logistic_susie_cache/
    Untracked:  simulation_targets/
    Untracked:  single_cell_pbmc_cache/
    Untracked:  summary_stat_gsea_exploration_cache/

Unstaged changes:
    Modified:   _simulation_targets.R
    Modified:   _targets.Rmd
    Modified:   analysis/gseabenchmark_tcga.Rmd
    Modified:   code/fit_baselines.R
    Modified:   code/fit_logistic_susie.R
    Modified:   code/fit_mr_ash.R
    Modified:   code/fit_susie.R
    Modified:   code/load_gene_sets.R
    Modified:   code/marginal_sumstat_gsea.R
    Modified:   code/simulate_gene_lists.R
    Modified:   code/utils.R
    Modified:   target_components/factories.R
    Modified:   target_components/methods.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/single_cell_pbmc.Rmd</code>) and HTML (<code>docs/single_cell_pbmc.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/karltayeb/logistic-susie-gsea/blob/fc1f3c1d433ec35d805a712a30e006a116027e5a/analysis/single_cell_pbmc.Rmd" target="_blank">fc1f3c1</a>
</td>
<td>
karltayeb
</td>
<td>
2022-03-29
</td>
<td>
wflow_publish(“analysis/single_cell_pbmc.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/karltayeb/logistic-susie-gsea/8646723be9a94fab6ea656001b458271b64bcb67/docs/single_cell_pbmc.html" target="_blank">8646723</a>
</td>
<td>
karltayeb
</td>
<td>
2022-03-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/karltayeb/logistic-susie-gsea/blob/6143c4434f0e139fa367ece633d114fd392c0ec4/analysis/single_cell_pbmc.Rmd" target="_blank">6143c44</a>
</td>
<td>
karltayeb
</td>
<td>
2022-03-29
</td>
<td>
wflow_publish(“analysis/single_cell_pbmc.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/karltayeb/logistic-susie-gsea/56f81300a9dcac6d44f597108b7cf24e948e6459/docs/single_cell_pbmc.html" target="_blank">56f8130</a>
</td>
<td>
karltayeb
</td>
<td>
2022-03-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/karltayeb/logistic-susie-gsea/a2bdb5680eea3d5caf14d0b7423351c4232628b4/docs/single_cell_pbmc.html" target="_blank">a2bdb56</a>
</td>
<td>
karltayeb
</td>
<td>
2022-03-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/karltayeb/logistic-susie-gsea/blob/122deece85c32abc1d72b089a34a2f24c62d8972/analysis/single_cell_pbmc.Rmd" target="_blank">122deec</a>
</td>
<td>
karltayeb
</td>
<td>
2022-03-29
</td>
<td>
wflow_publish(pages)
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>Our goals here are to run Logistic SuSiE on differential expression results from TCGA. We want to assess:</p>
<ol style="list-style-type: decimal">
<li>If the resulting enrichment results look good/interpretable across multiple/concatenated gene sets</li>
<li>Assess sensitivity to a range of p-value thresholds</li>
<li>Evaluate the potential of the summary stat latent model</li>
</ol>
<pre class="r"><code>library(GSEABenchmarkeR)
library(EnrichmentBrowser)
library(tidyverse)
library(susieR)
library(DT)
source(&#39;code/load_gene_sets.R&#39;)
source(&#39;code/utils.R&#39;)
source(&#39;code/logistic_susie_vb.R&#39;)
source(&#39;code/logistic_susie_veb_boost.R&#39;)
source(&#39;code/latent_logistic_susie.R&#39;)</code></pre>
</div>
<div id="setup" class="section level2">
<h2>Setup</h2>
<div id="load-gene-sets" class="section level3">
<h3>Load Gene Sets</h3>
<p><code>loadGeneSetX</code> uniformly formats gene sets and generates the <span class="math inline">\(X\)</span> matrix We can source any gene set from <code>WebGestaltR::listGeneSet()</code></p>
<pre class="r"><code>gs_list &lt;- WebGestaltR::listGeneSet()
gobp &lt;- loadGeneSetX(&#39;geneontology_Biological_Process&#39;, min.size=50)  # just huge number of gene sets
gobp_nr &lt;- loadGeneSetX(&#39;geneontology_Biological_Process_noRedundant&#39;, min.size=1)
gomf &lt;- loadGeneSetX(&#39;geneontology_Molecular_Function&#39;, min.size=1)
kegg &lt;- loadGeneSetX(&#39;pathway_KEGG&#39;, min.size=1)
reactome &lt;- loadGeneSetX(&#39;pathway_Reactome&#39;, min.size=1)
wikipathway_cancer &lt;- loadGeneSetX(&#39;pathway_Wikipathway_cancer&#39;, min.size=1)
wikipathway &lt;- loadGeneSetX(&#39;pathway_Wikipathway&#39;, min.size=1)

genesets &lt;- list(
  gobp=gobp,
  gobp_nr=gobp_nr,
  gomf=gomf,
  kegg=kegg,
  reactome=reactome,
  wikipathway_cancer=wikipathway_cancer,
  wikipathway=wikipathway
)</code></pre>
<pre class="r"><code>load(&#39;data/pbmc-purified/deseq2-pbmc-purified.RData&#39;)

convert_labels &lt;- function(y, from=&#39;SYMBOL&#39;, to=&#39;ENTREZID&#39;){
  hs &lt;- org.Hs.eg.db::org.Hs.eg.db
  gene_symbols &lt;- names(y)
  symbol2entrez &lt;- AnnotationDbi::select(hs, keys=gene_symbols, columns=c(to, from), keytype = from)
  symbol2entrez &lt;- symbol2entrez[!duplicated(symbol2entrez[[from]]),]
  symbol2entrez &lt;- symbol2entrez[!is.na(symbol2entrez[[to]]),]
  symbol2entrez &lt;- symbol2entrez[!is.na(symbol2entrez[[from]]),]
  rownames(symbol2entrez) &lt;- symbol2entrez[[from]]
  ysub &lt;- y[names(y) %in% symbol2entrez[[from]]]
  names(ysub) &lt;- symbol2entrez[names(ysub),][[to]]
  return(ysub)
}


par(mfrow=c(1,1))
deseq$`CD19+ B` %&gt;% .$padj %&gt;% hist(main=&#39;CD19+B p-values&#39;)</code></pre>
<pre><code>Loading required package: DESeq2</code></pre>
<p><img src="figure/single_cell_pbmc.Rmd/data.setup-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-data.setup-1">
Past versions of data.setup-1.png
</button>
</p>
<div id="fig-data.setup-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/karltayeb/logistic-susie-gsea/blob/a2bdb5680eea3d5caf14d0b7423351c4232628b4/docs/figure/single_cell_pbmc.Rmd/data.setup-1.png" target="_blank">a2bdb56</a>
</td>
<td>
karltayeb
</td>
<td>
2022-03-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="fit-logistic-susie" class="section level3">
<h3>Fit logistic SuSiE</h3>
<pre class="r"><code>logistic_susie_driver = function(db, celltype, thresh){
  gs &lt;- genesets[[db]]
  data &lt;- deseq[[celltype]]
  
  # set up binary y
  y &lt;- data %&gt;%
    as.data.frame %&gt;%
    rownames_to_column(&#39;gene&#39;) %&gt;%
    dplyr::select(gene, padj) %&gt;%
    filter(!is.na(padj)) %&gt;%
    mutate(y = as.integer(padj &lt; thresh)) %&gt;%
    select(gene, y) %&gt;%
    tibble2namedlist %&gt;%
    convert_labels(&#39;ENSEMBL&#39;)
  
  u &lt;- process_input(gs$X, y)  # subset to common genes
  vb.fit &lt;- logistic.susie(  # fit model
    u$X, u$y, L=10, init.intercept = 0, verbose=1, maxit=100)

  # summarise results
  set.summary &lt;- vb.fit$pip %&gt;% 
    as_tibble(rownames=&#39;geneSet&#39;) %&gt;%
    rename(pip=value) %&gt;%
    mutate(
      top_component = apply(vb.fit$alpha, 2, which.max),
      active_set = top_component %in% vb.fit$sets$cs_index,
      top_component = paste0(&#39;L&#39;, top_component),
      cs = purrr::map(top_component, ~tryCatch(
        colnames(gs$X)[get(.x, vb.fit$sets$cs)], error = function(e) list())),
      in_cs = geneSet %in% cs,
      beta = colSums(vb.fit$mu * vb.fit$alpha),
      geneListSize = sum(u$y),
      geneSetSize = colSums(u$X),
      overlap = (u$y %*% u$X)[1,],
      nGenes = length(u$y),
      propSetInList = overlap / geneSetSize,
      oddsRatio = (overlap / (geneListSize - overlap)) / (
        (geneSetSize - overlap) / (nGenes - geneSetSize + overlap)),
    pValueHypergeometric = phyper(
      overlap-1, geneListSize, nGenes, geneSetSize, lower.tail= FALSE),
    db = db,
    celltype = celltype,
    thresh = thresh
    ) %&gt;% left_join(gs$geneSet$geneSetDes)
  return(list(fit = vb.fit, set.summary=set.summary))
}</code></pre>
<p>For each celltype, we fit logistic SuSiE using multiple gene set sources at various threshold of <code>padj</code>.</p>
<pre class="r"><code>celltypes &lt;- names(deseq)
pthresh &lt;- c(0.1, 0.01, 0.001, 0.0001, 0.00001, 0.000001)
db_name &lt;- names(genesets)
crossed &lt;- cross3(db_name, celltypes, pthresh)

pbmc_res &lt;- xfun::cache_rds({
  res &lt;- purrr::map(crossed, purrr::lift_dl(logistic_susie_driver))
  for (i in 1:length(res)){  # save some space
    res[[i]]$fit$dat &lt;- NULL
  }
  res
  }, file = &#39;logistic_susie_pbmc_genesets_pthresh.rds&#39;
)

pbmc_res_set_summary &lt;- dplyr::bind_rows(purrr::map(pbmc_res, ~ pluck(.x, &#39;set.summary&#39;)))</code></pre>
</div>
<div id="summary-functions" class="section level3">
<h3>Summary functions</h3>
<p>Just a few functions to help streamline looking at output</p>
<pre class="r"><code>pval_focussed_table = function(thresh=1e-3, filter_db=NULL, filter_celltype=NULL, top.n=50){
  pbmc_res_set_summary %&gt;%
  filter(
    case_when(
      is.null(filter_db) ~ TRUE,
      !is.null(filter_db) ~ db %in% filter_db
    ) &amp;
    thresh == thresh &amp;
    case_when(
      is.null(filter_celltype) ~ TRUE,
      !is.null(filter_celltype) ~ celltype %in% filter_celltype
    )
  )  %&gt;%
  dplyr::arrange(celltype, db, pValueHypergeometric) %&gt;%
  group_by(celltype, db) %&gt;% slice(1:top.n) %&gt;%
  select(celltype, db, geneSet, description, pip, top_component, oddsRatio, propSetInList, pValueHypergeometric) %&gt;%
  mutate_at(vars(celltype, db), factor) %&gt;%
  datatable(filter = &#39;top&#39;)
}

set_focussed_table = function(thresh=1e-3, filter_db=NULL, filter_celltype=NULL){
  pbmc_res_set_summary %&gt;%
  filter(
    case_when(
      is.null(filter_db) ~ TRUE,
      !is.null(filter_db) ~ db %in% filter_db
    ) &amp;
    thresh == 1e-3 &amp;
    in_cs &amp; active_set &amp;
    case_when(
      is.null(filter_celltype) ~ TRUE,
      !is.null(filter_celltype) ~ celltype %in% filter_celltype
    )
  )  %&gt;%
  dplyr::arrange(celltype, db, desc(pip)) %&gt;%
  select(celltype, db, geneSet, description, pip, top_component, oddsRatio, propSetInList, pValueHypergeometric) %&gt;%
  mutate_at(vars(celltype, geneSet, db), factor) %&gt;%
  datatable(filter = &#39;top&#39;)
}</code></pre>
</div>
</div>
<div id="resultsexplore-enrichments" class="section level2">
<h2>Results/Explore enrichments</h2>
<p>Our goal is to assess 1. The quality of the gene set enrichments we get from each celltype - do reported gene set enrichments seem celltype specific/celltype relevant? - how much “interesting” marginal enrichment do we fail to capture in the multivariate model - how sensitive are we to the choice of pvalue threshold</p>
</div>
<div id="results" class="section level2">
<h2>Results</h2>
<p>Lets take a look at what enrichment we’re getting across cell-types.</p>
<div id="cd19-b" class="section level3">
<h3>CD19+ B</h3>
<pre class="r"><code>pbmc_res_set_summary %&gt;%
  filter(active_set, in_cs, thresh == 1e-4, celltype == &#39;CD19+ B&#39;) %&gt;%
  select(
    geneSet, description, top_component, pip,
    beta, geneListSize, geneSetSize, propSetInList, pValueHypergeometric) %&gt;%
  knitr::kable()</code></pre>
<table>
<colgroup>
<col width="7%" />
<col width="39%" />
<col width="7%" />
<col width="5%" />
<col width="6%" />
<col width="7%" />
<col width="6%" />
<col width="7%" />
<col width="11%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">geneSet</th>
<th align="left">description</th>
<th align="left">top_component</th>
<th align="right">pip</th>
<th align="right">beta</th>
<th align="right">geneListSize</th>
<th align="right">geneSetSize</th>
<th align="right">propSetInList</th>
<th align="right">pValueHypergeometric</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left"><a href="GO:0001775" class="uri">GO:0001775</a></td>
<td align="left">cell activation</td>
<td align="left">L5</td>
<td align="right">0.9780542</td>
<td align="right">0.4672903</td>
<td align="right">6911</td>
<td align="right">1128</td>
<td align="right">0.7517730</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0002376" class="uri">GO:0002376</a></td>
<td align="left">immune system process</td>
<td align="left">L1</td>
<td align="right">0.9999575</td>
<td align="right">0.5078753</td>
<td align="right">6911</td>
<td align="right">2228</td>
<td align="right">0.7064632</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0045047" class="uri">GO:0045047</a></td>
<td align="left">protein targeting to ER</td>
<td align="left">L2</td>
<td align="right">0.9620521</td>
<td align="right">2.4624775</td>
<td align="right">6911</td>
<td align="right">105</td>
<td align="right">0.9428571</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0002764" class="uri">GO:0002764</a></td>
<td align="left">immune response-regulating signaling pathway</td>
<td align="left">L5</td>
<td align="right">0.9961853</td>
<td align="right">0.5645340</td>
<td align="right">6327</td>
<td align="right">411</td>
<td align="right">0.7445255</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0009123" class="uri">GO:0009123</a></td>
<td align="left">nucleoside monophosphate metabolic process</td>
<td align="left">L4</td>
<td align="right">0.9844976</td>
<td align="right">0.7415436</td>
<td align="right">6327</td>
<td align="right">286</td>
<td align="right">0.7517483</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0070972" class="uri">GO:0070972</a></td>
<td align="left">protein localization to endoplasmic reticulum</td>
<td align="left">L2</td>
<td align="right">0.9999580</td>
<td align="right">1.6294991</td>
<td align="right">6327</td>
<td align="right">132</td>
<td align="right">0.8712121</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0003723" class="uri">GO:0003723</a></td>
<td align="left">RNA binding</td>
<td align="left">L1</td>
<td align="right">0.9997440</td>
<td align="right">0.3528854</td>
<td align="right">6975</td>
<td align="right">1483</td>
<td align="right">0.6581254</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0000981" class="uri">GO:0000981</a></td>
<td align="left">DNA-binding transcription factor activity, RNA polymerase II-specific</td>
<td align="left">L2</td>
<td align="right">0.9950038</td>
<td align="right">-0.6568235</td>
<td align="right">6975</td>
<td align="right">1161</td>
<td align="right">0.4771748</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0003735" class="uri">GO:0003735</a></td>
<td align="left">structural constituent of ribosome</td>
<td align="left">L6</td>
<td align="right">0.9953441</td>
<td align="right">0.9638753</td>
<td align="right">6975</td>
<td align="right">152</td>
<td align="right">0.8289474</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">hsa00190</td>
<td align="left">Oxidative phosphorylation</td>
<td align="left">L1</td>
<td align="right">0.9979902</td>
<td align="right">1.4574888</td>
<td align="right">3273</td>
<td align="right">120</td>
<td align="right">0.8750000</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">hsa03010</td>
<td align="left">Ribosome</td>
<td align="left">L3</td>
<td align="right">1.0000000</td>
<td align="right">1.4064281</td>
<td align="right">3273</td>
<td align="right">129</td>
<td align="right">0.8682171</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">hsa04640</td>
<td align="left">Hematopoietic cell lineage</td>
<td align="left">L2</td>
<td align="right">0.9997951</td>
<td align="right">1.4990711</td>
<td align="right">3273</td>
<td align="right">80</td>
<td align="right">0.8875000</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">R-HSA-168256</td>
<td align="left">Immune System</td>
<td align="left">L1</td>
<td align="right">1.0000000</td>
<td align="right">0.7492562</td>
<td align="right">4620</td>
<td align="right">1627</td>
<td align="right">0.6969883</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">R-HSA-983168</td>
<td align="left">Antigen processing: Ubiquitination &amp; Proteasome degradation</td>
<td align="left">L5</td>
<td align="right">0.9970497</td>
<td align="right">-1.0658222</td>
<td align="right">4620</td>
<td align="right">280</td>
<td align="right">0.5392857</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">WP619</td>
<td align="left">Type II interferon signaling (IFNG)</td>
<td align="left">L1</td>
<td align="right">0.9998636</td>
<td align="right">2.4876068</td>
<td align="right">966</td>
<td align="right">30</td>
<td align="right">0.9666667</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">WP111</td>
<td align="left">Electron Transport Chain (OXPHOS system in mitochondria)</td>
<td align="left">L2</td>
<td align="right">0.9999189</td>
<td align="right">1.6369406</td>
<td align="right">3220</td>
<td align="right">101</td>
<td align="right">0.9009901</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">WP477</td>
<td align="left">Cytoplasmic Ribosomal Proteins</td>
<td align="left">L1</td>
<td align="right">1.0000000</td>
<td align="right">2.4011250</td>
<td align="right">3220</td>
<td align="right">85</td>
<td align="right">0.9529412</td>
<td align="right">0</td>
</tr>
</tbody>
</table>
</div>
<div id="cd56-nk" class="section level3">
<h3>CD56+ NK</h3>
<pre class="r"><code>pbmc_res_set_summary %&gt;%
  filter(active_set, in_cs, thresh == 1e-4, celltype == &#39;CD56+ NK&#39;) %&gt;%
  select(
    geneSet, description, top_component, pip,
    beta, geneListSize, geneSetSize, propSetInList, pValueHypergeometric) %&gt;%
  knitr::kable()</code></pre>
<table style="width:100%;">
<colgroup>
<col width="6%" />
<col width="51%" />
<col width="6%" />
<col width="4%" />
<col width="4%" />
<col width="5%" />
<col width="5%" />
<col width="6%" />
<col width="9%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">geneSet</th>
<th align="left">description</th>
<th align="left">top_component</th>
<th align="right">pip</th>
<th align="right">beta</th>
<th align="right">geneListSize</th>
<th align="right">geneSetSize</th>
<th align="right">propSetInList</th>
<th align="right">pValueHypergeometric</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left"><a href="GO:0002376" class="uri">GO:0002376</a></td>
<td align="left">immune system process</td>
<td align="left">L1</td>
<td align="right">0.9999984</td>
<td align="right">0.6534004</td>
<td align="right">7231</td>
<td align="right">2228</td>
<td align="right">0.7221724</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0006119" class="uri">GO:0006119</a></td>
<td align="left">oxidative phosphorylation</td>
<td align="left">L4</td>
<td align="right">0.9815754</td>
<td align="right">1.3155955</td>
<td align="right">7231</td>
<td align="right">118</td>
<td align="right">0.8474576</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0006413" class="uri">GO:0006413</a></td>
<td align="left">translational initiation</td>
<td align="left">L2</td>
<td align="right">0.9954877</td>
<td align="right">1.1585848</td>
<td align="right">6598</td>
<td align="right">182</td>
<td align="right">0.8351648</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0009123" class="uri">GO:0009123</a></td>
<td align="left">nucleoside monophosphate metabolic process</td>
<td align="left">L4</td>
<td align="right">0.9815927</td>
<td align="right">0.6736545</td>
<td align="right">6598</td>
<td align="right">286</td>
<td align="right">0.7622378</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0042110" class="uri">GO:0042110</a></td>
<td align="left">T cell activation</td>
<td align="left">L3</td>
<td align="right">0.9856601</td>
<td align="right">0.5907739</td>
<td align="right">6598</td>
<td align="right">389</td>
<td align="right">0.7763496</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0042113" class="uri">GO:0042113</a></td>
<td align="left">B cell activation</td>
<td align="left">L5</td>
<td align="right">0.9972725</td>
<td align="right">0.8422029</td>
<td align="right">6598</td>
<td align="right">198</td>
<td align="right">0.8232323</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0000981" class="uri">GO:0000981</a></td>
<td align="left">DNA-binding transcription factor activity, RNA polymerase II-specific</td>
<td align="left">L2</td>
<td align="right">0.9967706</td>
<td align="right">-0.7338417</td>
<td align="right">7305</td>
<td align="right">1161</td>
<td align="right">0.4952627</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0003735" class="uri">GO:0003735</a></td>
<td align="left">structural constituent of ribosome</td>
<td align="left">L1</td>
<td align="right">1.0000000</td>
<td align="right">1.9030114</td>
<td align="right">7305</td>
<td align="right">152</td>
<td align="right">0.9144737</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">hsa03010</td>
<td align="left">Ribosome</td>
<td align="left">L1</td>
<td align="right">1.0000000</td>
<td align="right">1.7794168</td>
<td align="right">3406</td>
<td align="right">129</td>
<td align="right">0.9147287</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">hsa04640</td>
<td align="left">Hematopoietic cell lineage</td>
<td align="left">L3</td>
<td align="right">0.9978541</td>
<td align="right">1.3141833</td>
<td align="right">3406</td>
<td align="right">80</td>
<td align="right">0.8750000</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">hsa05012</td>
<td align="left">Parkinson disease</td>
<td align="left">L2</td>
<td align="right">0.9999978</td>
<td align="right">1.5611100</td>
<td align="right">3406</td>
<td align="right">126</td>
<td align="right">0.8968254</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">R-HSA-168256</td>
<td align="left">Immune System</td>
<td align="left">L1</td>
<td align="right">1.0000000</td>
<td align="right">0.6885907</td>
<td align="right">4855</td>
<td align="right">1627</td>
<td align="right">0.7209588</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">R-HSA-8878171</td>
<td align="left">Transcriptional regulation by RUNX1</td>
<td align="left">L5</td>
<td align="right">0.9712818</td>
<td align="right">0.9657802</td>
<td align="right">4855</td>
<td align="right">189</td>
<td align="right">0.7513228</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">R-HSA-163200</td>
<td align="left">Respiratory electron transport, ATP synthesis by chemiosmotic coupling, and heat production by uncoupling proteins.</td>
<td align="left">L4</td>
<td align="right">0.9933245</td>
<td align="right">1.3233274</td>
<td align="right">4855</td>
<td align="right">121</td>
<td align="right">0.8429752</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">WP111</td>
<td align="left">Electron Transport Chain (OXPHOS system in mitochondria)</td>
<td align="left">L2</td>
<td align="right">0.9908620</td>
<td align="right">1.3218233</td>
<td align="right">3316</td>
<td align="right">101</td>
<td align="right">0.8811881</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">WP477</td>
<td align="left">Cytoplasmic Ribosomal Proteins</td>
<td align="left">L1</td>
<td align="right">1.0000000</td>
<td align="right">2.9682233</td>
<td align="right">3316</td>
<td align="right">85</td>
<td align="right">0.9764706</td>
<td align="right">0</td>
</tr>
</tbody>
</table>
</div>
<div id="t-cell" class="section level3">
<h3>T cell</h3>
<pre class="r"><code>pbmc_res_set_summary %&gt;%
  filter(active_set, in_cs, thresh == 1e-4, celltype == &#39;T cell&#39;) %&gt;%
  select(
    geneSet, description, top_component, pip,
    beta, geneListSize, geneSetSize, propSetInList, pValueHypergeometric) %&gt;%
  knitr::kable()</code></pre>
<table>
<colgroup>
<col width="9%" />
<col width="29%" />
<col width="9%" />
<col width="6%" />
<col width="6%" />
<col width="8%" />
<col width="7%" />
<col width="9%" />
<col width="13%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">geneSet</th>
<th align="left">description</th>
<th align="left">top_component</th>
<th align="right">pip</th>
<th align="right">beta</th>
<th align="right">geneListSize</th>
<th align="right">geneSetSize</th>
<th align="right">propSetInList</th>
<th align="right">pValueHypergeometric</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left"><a href="GO:0001775" class="uri">GO:0001775</a></td>
<td align="left">cell activation</td>
<td align="left">L1</td>
<td align="right">0.9982435</td>
<td align="right">0.7533519</td>
<td align="right">8802</td>
<td align="right">1127</td>
<td align="right">0.8881988</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0002376" class="uri">GO:0002376</a></td>
<td align="left">immune system process</td>
<td align="left">L5</td>
<td align="right">0.9984689</td>
<td align="right">0.4243072</td>
<td align="right">8802</td>
<td align="right">2227</td>
<td align="right">0.8378985</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0006119" class="uri">GO:0006119</a></td>
<td align="left">oxidative phosphorylation</td>
<td align="left">L2</td>
<td align="right">0.9988673</td>
<td align="right">1.7903483</td>
<td align="right">8802</td>
<td align="right">118</td>
<td align="right">0.9406780</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0042110" class="uri">GO:0042110</a></td>
<td align="left">T cell activation</td>
<td align="left">L2</td>
<td align="right">0.9999906</td>
<td align="right">0.9206753</td>
<td align="right">7959</td>
<td align="right">389</td>
<td align="right">0.8868895</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0070972" class="uri">GO:0070972</a></td>
<td align="left">protein localization to endoplasmic reticulum</td>
<td align="left">L3</td>
<td align="right">0.9998559</td>
<td align="right">1.3305778</td>
<td align="right">7959</td>
<td align="right">132</td>
<td align="right">0.9166667</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0005515" class="uri">GO:0005515</a></td>
<td align="left">protein binding</td>
<td align="left">L1</td>
<td align="right">1.0000000</td>
<td align="right">0.3704248</td>
<td align="right">8880</td>
<td align="right">9033</td>
<td align="right">0.7474815</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">hsa05010</td>
<td align="left">Alzheimer disease</td>
<td align="left">L1</td>
<td align="right">0.9999993</td>
<td align="right">1.6786291</td>
<td align="right">3983</td>
<td align="right">154</td>
<td align="right">0.9480519</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">R-HSA-6798695</td>
<td align="left">Neutrophil degranulation</td>
<td align="left">L1</td>
<td align="right">1.0000000</td>
<td align="right">1.3714212</td>
<td align="right">5790</td>
<td align="right">431</td>
<td align="right">0.9164733</td>
<td align="right">0</td>
</tr>
</tbody>
</table>
</div>
<div id="cd14-monocyte" class="section level3">
<h3>CD14+ Monocyte</h3>
<pre class="r"><code>pbmc_res_set_summary %&gt;%
  filter(active_set, in_cs, thresh == 1e-4, celltype == &#39;CD14+ Monocyte&#39;) %&gt;%
  select(
    geneSet, description, top_component, pip,
    beta, geneListSize, geneSetSize, propSetInList, pValueHypergeometric) %&gt;%
  knitr::kable()</code></pre>
<table style="width:100%;">
<colgroup>
<col width="6%" />
<col width="51%" />
<col width="6%" />
<col width="4%" />
<col width="4%" />
<col width="5%" />
<col width="5%" />
<col width="6%" />
<col width="9%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">geneSet</th>
<th align="left">description</th>
<th align="left">top_component</th>
<th align="right">pip</th>
<th align="right">beta</th>
<th align="right">geneListSize</th>
<th align="right">geneSetSize</th>
<th align="right">propSetInList</th>
<th align="right">pValueHypergeometric</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left"><a href="GO:0016192" class="uri">GO:0016192</a></td>
<td align="left">vesicle-mediated transport</td>
<td align="left">L4</td>
<td align="right">0.9851434</td>
<td align="right">0.3239679</td>
<td align="right">6633</td>
<td align="right">1605</td>
<td align="right">0.6560748</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0006119" class="uri">GO:0006119</a></td>
<td align="left">oxidative phosphorylation</td>
<td align="left">L3</td>
<td align="right">0.9990485</td>
<td align="right">1.7896878</td>
<td align="right">6633</td>
<td align="right">118</td>
<td align="right">0.8728814</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0006413" class="uri">GO:0006413</a></td>
<td align="left">translational initiation</td>
<td align="left">L2</td>
<td align="right">0.9999940</td>
<td align="right">1.2066226</td>
<td align="right">6073</td>
<td align="right">182</td>
<td align="right">0.8461538</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0009123" class="uri">GO:0009123</a></td>
<td align="left">nucleoside monophosphate metabolic process</td>
<td align="left">L3</td>
<td align="right">0.9883528</td>
<td align="right">0.9050661</td>
<td align="right">6073</td>
<td align="right">286</td>
<td align="right">0.7622378</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0036230" class="uri">GO:0036230</a></td>
<td align="left">granulocyte activation</td>
<td align="left">L1</td>
<td align="right">0.9759280</td>
<td align="right">1.1699222</td>
<td align="right">6073</td>
<td align="right">450</td>
<td align="right">0.8088889</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0003723" class="uri">GO:0003723</a></td>
<td align="left">RNA binding</td>
<td align="left">L1</td>
<td align="right">0.9999998</td>
<td align="right">0.4163546</td>
<td align="right">6697</td>
<td align="right">1480</td>
<td align="right">0.6655405</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0000981" class="uri">GO:0000981</a></td>
<td align="left">DNA-binding transcription factor activity, RNA polymerase II-specific</td>
<td align="left">L2</td>
<td align="right">0.9998275</td>
<td align="right">-0.6329681</td>
<td align="right">6697</td>
<td align="right">1160</td>
<td align="right">0.4646552</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0003735" class="uri">GO:0003735</a></td>
<td align="left">structural constituent of ribosome</td>
<td align="left">L4</td>
<td align="right">1.0000000</td>
<td align="right">1.6077468</td>
<td align="right">6697</td>
<td align="right">152</td>
<td align="right">0.8947368</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="left">hsa03010</td>
<td align="left">Ribosome</td>
<td align="left">L1</td>
<td align="right">1.0000000</td>
<td align="right">1.7470056</td>
<td align="right">3110</td>
<td align="right">129</td>
<td align="right">0.8914729</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="left">hsa05012</td>
<td align="left">Parkinson disease</td>
<td align="left">L2</td>
<td align="right">0.9999021</td>
<td align="right">1.6964808</td>
<td align="right">3110</td>
<td align="right">125</td>
<td align="right">0.8880000</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="left">R-HSA-6798695</td>
<td align="left">Neutrophil degranulation</td>
<td align="left">L1</td>
<td align="right">0.9999954</td>
<td align="right">1.2111808</td>
<td align="right">4451</td>
<td align="right">431</td>
<td align="right">0.8051044</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="left">R-HSA-198933</td>
<td align="left">Immunoregulatory interactions between a Lymphoid and a non-Lymphoid cell</td>
<td align="left">L6</td>
<td align="right">0.9974877</td>
<td align="right">1.1921499</td>
<td align="right">4451</td>
<td align="right">103</td>
<td align="right">0.8155340</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="left">R-HSA-72766</td>
<td align="left">Translation</td>
<td align="left">L2</td>
<td align="right">1.0000000</td>
<td align="right">1.6620629</td>
<td align="right">4451</td>
<td align="right">286</td>
<td align="right">0.8251748</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="left">R-HSA-163200</td>
<td align="left">Respiratory electron transport, ATP synthesis by chemiosmotic coupling, and heat production by uncoupling proteins.</td>
<td align="left">L3</td>
<td align="right">0.9996111</td>
<td align="right">1.7113805</td>
<td align="right">4451</td>
<td align="right">121</td>
<td align="right">0.8677686</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="left">R-HSA-379726</td>
<td align="left">Mitochondrial tRNA aminoacylation</td>
<td align="left">L7</td>
<td align="right">0.9972986</td>
<td align="right">-2.4026838</td>
<td align="right">4451</td>
<td align="right">21</td>
<td align="right">0.3333333</td>
<td align="right">0.6931782</td>
</tr>
<tr class="even">
<td align="left">WP111</td>
<td align="left">Electron Transport Chain (OXPHOS system in mitochondria)</td>
<td align="left">L2</td>
<td align="right">0.9998312</td>
<td align="right">1.7563572</td>
<td align="right">3062</td>
<td align="right">100</td>
<td align="right">0.9000000</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="left">WP477</td>
<td align="left">Cytoplasmic Ribosomal Proteins</td>
<td align="left">L1</td>
<td align="right">1.0000000</td>
<td align="right">2.5280541</td>
<td align="right">3062</td>
<td align="right">85</td>
<td align="right">0.9529412</td>
<td align="right">0.0000000</td>
</tr>
</tbody>
</table>
</div>
<div id="cd34" class="section level3">
<h3>CD34+</h3>
<pre class="r"><code>pbmc_res_set_summary %&gt;%
  filter(active_set, in_cs, thresh == 1e-4, celltype == &#39;CD34+&#39;) %&gt;%
  select(
    geneSet, description, top_component, pip,
    beta, geneListSize, geneSetSize, propSetInList, pValueHypergeometric) %&gt;%
  knitr::kable()</code></pre>
<table>
<colgroup>
<col width="6%" />
<col width="51%" />
<col width="6%" />
<col width="4%" />
<col width="4%" />
<col width="5%" />
<col width="5%" />
<col width="6%" />
<col width="9%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">geneSet</th>
<th align="left">description</th>
<th align="left">top_component</th>
<th align="right">pip</th>
<th align="right">beta</th>
<th align="right">geneListSize</th>
<th align="right">geneSetSize</th>
<th align="right">propSetInList</th>
<th align="right">pValueHypergeometric</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left"><a href="GO:0001775" class="uri">GO:0001775</a></td>
<td align="left">cell activation</td>
<td align="left">L1</td>
<td align="right">0.9870540</td>
<td align="right">0.7305079</td>
<td align="right">8363</td>
<td align="right">1128</td>
<td align="right">0.8226950</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0006119" class="uri">GO:0006119</a></td>
<td align="left">oxidative phosphorylation</td>
<td align="left">L3</td>
<td align="right">0.9989413</td>
<td align="right">1.5639047</td>
<td align="right">8363</td>
<td align="right">118</td>
<td align="right">0.9152542</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0006413" class="uri">GO:0006413</a></td>
<td align="left">translational initiation</td>
<td align="left">L2</td>
<td align="right">0.9996608</td>
<td align="right">1.0665346</td>
<td align="right">7544</td>
<td align="right">182</td>
<td align="right">0.8736264</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0005515" class="uri">GO:0005515</a></td>
<td align="left">protein binding</td>
<td align="left">L2</td>
<td align="right">1.0000000</td>
<td align="right">0.3367103</td>
<td align="right">8421</td>
<td align="right">9031</td>
<td align="right">0.7087809</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0003735" class="uri">GO:0003735</a></td>
<td align="left">structural constituent of ribosome</td>
<td align="left">L1</td>
<td align="right">1.0000000</td>
<td align="right">1.7570426</td>
<td align="right">8421</td>
<td align="right">152</td>
<td align="right">0.9276316</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">hsa00190</td>
<td align="left">Oxidative phosphorylation</td>
<td align="left">L1</td>
<td align="right">0.9921759</td>
<td align="right">1.5131062</td>
<td align="right">3772</td>
<td align="right">120</td>
<td align="right">0.9250000</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">hsa03010</td>
<td align="left">Ribosome</td>
<td align="left">L2</td>
<td align="right">1.0000000</td>
<td align="right">1.5232912</td>
<td align="right">3772</td>
<td align="right">129</td>
<td align="right">0.9224806</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">R-HSA-6798695</td>
<td align="left">Neutrophil degranulation</td>
<td align="left">L2</td>
<td align="right">0.9862716</td>
<td align="right">0.7788667</td>
<td align="right">5481</td>
<td align="right">432</td>
<td align="right">0.8356481</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">R-HSA-198933</td>
<td align="left">Immunoregulatory interactions between a Lymphoid and a non-Lymphoid cell</td>
<td align="left">L4</td>
<td align="right">0.9999503</td>
<td align="right">1.4291485</td>
<td align="right">5481</td>
<td align="right">103</td>
<td align="right">0.9126214</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">R-HSA-163200</td>
<td align="left">Respiratory electron transport, ATP synthesis by chemiosmotic coupling, and heat production by uncoupling proteins.</td>
<td align="left">L3</td>
<td align="right">0.9977325</td>
<td align="right">1.5694015</td>
<td align="right">5481</td>
<td align="right">121</td>
<td align="right">0.9173554</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">R-HSA-72764</td>
<td align="left">Eukaryotic Translation Termination</td>
<td align="left">L1</td>
<td align="right">0.9930562</td>
<td align="right">3.4135736</td>
<td align="right">5481</td>
<td align="right">90</td>
<td align="right">0.9888889</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">WP111</td>
<td align="left">Electron Transport Chain (OXPHOS system in mitochondria)</td>
<td align="left">L2</td>
<td align="right">0.9999971</td>
<td align="right">2.1291975</td>
<td align="right">3698</td>
<td align="right">101</td>
<td align="right">0.9603960</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">WP477</td>
<td align="left">Cytoplasmic Ribosomal Proteins</td>
<td align="left">L1</td>
<td align="right">1.0000000</td>
<td align="right">3.2474751</td>
<td align="right">3698</td>
<td align="right">85</td>
<td align="right">0.9882353</td>
<td align="right">0</td>
</tr>
</tbody>
</table>
<pre class="r"><code>knitr::knit_exit()</code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
