<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="karltayeb" />

<meta name="date" content="2022-03-16" />

<title>Single-cell PBMC Example</title>

<script src="site_libs/header-attrs-2.13/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">logistic-susie-gsea</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/karltayeb/logistic-susie-gsea">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Single-cell PBMC Example</h1>
<h4 class="author">karltayeb</h4>
<h4 class="date">2022-03-16</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-03-29
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>logistic-susie-gsea/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220105code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20220105)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220105code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220105)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomkarltayeblogisticsusiegseatree54209afb933b39ba8a596f008997341e6723c294targetblank54209afa"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/karltayeb/logistic-susie-gsea/tree/54209afb933b39ba8a596f008997341e6723c294" target="_blank">54209af</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomkarltayeblogisticsusiegseatree54209afb933b39ba8a596f008997341e6723c294targetblank54209afa" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/karltayeb/logistic-susie-gsea/tree/54209afb933b39ba8a596f008997341e6723c294" target="_blank">54209af</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    library/
    Ignored:    renv/library/
    Ignored:    renv/staging/
    Ignored:    staging/

Untracked files:
    Untracked:  _targets.R
    Untracked:  _targets.html
    Untracked:  _targets.md
    Untracked:  _targets/
    Untracked:  _targets_r/
    Untracked:  analysis/fetal_reference_cellid_gsea.Rmd
    Untracked:  analysis/fixed_intercept.Rmd
    Untracked:  analysis/iDEA_examples.Rmd
    Untracked:  analysis/latent_gene_list.Rmd
    Untracked:  analysis/latent_logistic_susie.Rmd
    Untracked:  analysis/libra_setup.Rmd
    Untracked:  analysis/linear_method_failure_modes.Rmd
    Untracked:  analysis/linear_regression_failure_regime.Rmd
    Untracked:  analysis/logistic_susie_veb_boost_vs_vb.Rmd
    Untracked:  analysis/references.bib
    Untracked:  analysis/simulations.Rmd
    Untracked:  analysis/test.Rmd
    Untracked:  analysis/wenhe_baboon_example.Rmd
    Untracked:  build_site.R
    Untracked:  cache/
    Untracked:  code/latent_logistic_susie.R
    Untracked:  code/marginal_sumstat_gsea_collapsed.R
    Untracked:  data/adipose_2yr_topsnp.txt
    Untracked:  data/fetal_reference_cellid_gene_sets.RData
    Untracked:  data/pbmc-purified/
    Untracked:  docs.zip
    Untracked:  index.md
    Untracked:  latent_logistic_susie_cache/
    Untracked:  simulation_targets/
    Untracked:  single_cell_pbmc_cache/
    Untracked:  summary_stat_gsea_exploration_cache/

Unstaged changes:
    Modified:   _simulation_targets.R
    Modified:   _targets.Rmd
    Modified:   analysis/gseabenchmark_tcga.Rmd
    Modified:   code/fit_baselines.R
    Modified:   code/fit_logistic_susie.R
    Modified:   code/fit_mr_ash.R
    Modified:   code/fit_susie.R
    Modified:   code/load_gene_sets.R
    Modified:   code/marginal_sumstat_gsea.R
    Modified:   code/simulate_gene_lists.R
    Modified:   code/utils.R
    Modified:   target_components/factories.R
    Modified:   target_components/methods.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/single_cell_pbmc.Rmd</code>) and HTML (<code>docs/single_cell_pbmc.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/karltayeb/logistic-susie-gsea/blob/54209afb933b39ba8a596f008997341e6723c294/analysis/single_cell_pbmc.Rmd" target="_blank">54209af</a>
</td>
<td>
karltayeb
</td>
<td>
2022-03-29
</td>
<td>
wflow_publish(“analysis/single_cell_pbmc.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/karltayeb/logistic-susie-gsea/67bf8cfa4262bcb1c6ae9ff17b0ae9e208629b3f/docs/single_cell_pbmc.html" target="_blank">67bf8cf</a>
</td>
<td>
karltayeb
</td>
<td>
2022-03-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/karltayeb/logistic-susie-gsea/blob/170e480377710a7d97bfbb90273e314939513a99/analysis/single_cell_pbmc.Rmd" target="_blank">170e480</a>
</td>
<td>
karltayeb
</td>
<td>
2022-03-29
</td>
<td>
wflow_publish(“analysis/single_cell_pbmc.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/karltayeb/logistic-susie-gsea/90c6006fce9f55a35dcfe0e29465c052093db831/docs/single_cell_pbmc.html" target="_blank">90c6006</a>
</td>
<td>
karltayeb
</td>
<td>
2022-03-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/karltayeb/logistic-susie-gsea/blob/8159b83cf4ae6a16db3e46ca7d03372671d5fb6c/analysis/single_cell_pbmc.Rmd" target="_blank">8159b83</a>
</td>
<td>
karltayeb
</td>
<td>
2022-03-29
</td>
<td>
wflow_publish(“analysis/single_cell_pbmc.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/karltayeb/logistic-susie-gsea/9d7fd295ff89bd12920097345723d95029a123ca/docs/single_cell_pbmc.html" target="_blank">9d7fd29</a>
</td>
<td>
karltayeb
</td>
<td>
2022-03-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/karltayeb/logistic-susie-gsea/blob/51647d478484cabd6aa755b3b4be4907ea5f20bb/analysis/single_cell_pbmc.Rmd" target="_blank">51647d4</a>
</td>
<td>
karltayeb
</td>
<td>
2022-03-29
</td>
<td>
wflow_publish(“analysis/single_cell_pbmc.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/karltayeb/logistic-susie-gsea/3da7c5b2302522c09487bce436ab06c3188e638c/docs/single_cell_pbmc.html" target="_blank">3da7c5b</a>
</td>
<td>
karltayeb
</td>
<td>
2022-03-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/karltayeb/logistic-susie-gsea/blob/fc1f3c1d433ec35d805a712a30e006a116027e5a/analysis/single_cell_pbmc.Rmd" target="_blank">fc1f3c1</a>
</td>
<td>
karltayeb
</td>
<td>
2022-03-29
</td>
<td>
wflow_publish(“analysis/single_cell_pbmc.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/karltayeb/logistic-susie-gsea/8646723be9a94fab6ea656001b458271b64bcb67/docs/single_cell_pbmc.html" target="_blank">8646723</a>
</td>
<td>
karltayeb
</td>
<td>
2022-03-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/karltayeb/logistic-susie-gsea/blob/6143c4434f0e139fa367ece633d114fd392c0ec4/analysis/single_cell_pbmc.Rmd" target="_blank">6143c44</a>
</td>
<td>
karltayeb
</td>
<td>
2022-03-29
</td>
<td>
wflow_publish(“analysis/single_cell_pbmc.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/karltayeb/logistic-susie-gsea/56f81300a9dcac6d44f597108b7cf24e948e6459/docs/single_cell_pbmc.html" target="_blank">56f8130</a>
</td>
<td>
karltayeb
</td>
<td>
2022-03-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/karltayeb/logistic-susie-gsea/a2bdb5680eea3d5caf14d0b7423351c4232628b4/docs/single_cell_pbmc.html" target="_blank">a2bdb56</a>
</td>
<td>
karltayeb
</td>
<td>
2022-03-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/karltayeb/logistic-susie-gsea/blob/122deece85c32abc1d72b089a34a2f24c62d8972/analysis/single_cell_pbmc.Rmd" target="_blank">122deec</a>
</td>
<td>
karltayeb
</td>
<td>
2022-03-29
</td>
<td>
wflow_publish(pages)
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>Our goals here are to run Logistic SuSiE on differential expression results from TCGA. We want to assess:</p>
<ol style="list-style-type: decimal">
<li>If the resulting enrichment results look good/interpretable across multiple/concatenated gene sets</li>
<li>Assess sensitivity to a range of p-value thresholds</li>
<li>Evaluate the potential of the summary stat latent model</li>
</ol>
<pre class="r"><code>library(GSEABenchmarkeR)
library(EnrichmentBrowser)
library(tidyverse)
library(susieR)
library(DT)
source(&#39;code/load_gene_sets.R&#39;)
source(&#39;code/utils.R&#39;)
source(&#39;code/logistic_susie_vb.R&#39;)
source(&#39;code/logistic_susie_veb_boost.R&#39;)
source(&#39;code/latent_logistic_susie.R&#39;)</code></pre>
</div>
<div id="setup" class="section level2">
<h2>Setup</h2>
<div id="load-gene-sets" class="section level3">
<h3>Load Gene Sets</h3>
<p><code>loadGeneSetX</code> uniformly formats gene sets and generates the <span class="math inline">\(X\)</span> matrix We can source any gene set from <code>WebGestaltR::listGeneSet()</code></p>
<pre class="r"><code>gs_list &lt;- WebGestaltR::listGeneSet()
gobp &lt;- loadGeneSetX(&#39;geneontology_Biological_Process&#39;, min.size=50)  # just huge number of gene sets
gobp_nr &lt;- loadGeneSetX(&#39;geneontology_Biological_Process_noRedundant&#39;, min.size=1)
gomf &lt;- loadGeneSetX(&#39;geneontology_Molecular_Function&#39;, min.size=1)
kegg &lt;- loadGeneSetX(&#39;pathway_KEGG&#39;, min.size=1)
reactome &lt;- loadGeneSetX(&#39;pathway_Reactome&#39;, min.size=1)
wikipathway_cancer &lt;- loadGeneSetX(&#39;pathway_Wikipathway_cancer&#39;, min.size=1)
wikipathway &lt;- loadGeneSetX(&#39;pathway_Wikipathway&#39;, min.size=1)

genesets &lt;- list(
  gobp=gobp,
  gobp_nr=gobp_nr,
  gomf=gomf,
  kegg=kegg,
  reactome=reactome,
  wikipathway_cancer=wikipathway_cancer,
  wikipathway=wikipathway
)</code></pre>
<pre class="r"><code>load(&#39;data/pbmc-purified/deseq2-pbmc-purified.RData&#39;)

convert_labels &lt;- function(y, from=&#39;SYMBOL&#39;, to=&#39;ENTREZID&#39;){
  hs &lt;- org.Hs.eg.db::org.Hs.eg.db
  gene_symbols &lt;- names(y)
  symbol2entrez &lt;- AnnotationDbi::select(hs, keys=gene_symbols, columns=c(to, from), keytype = from)
  symbol2entrez &lt;- symbol2entrez[!duplicated(symbol2entrez[[from]]),]
  symbol2entrez &lt;- symbol2entrez[!is.na(symbol2entrez[[to]]),]
  symbol2entrez &lt;- symbol2entrez[!is.na(symbol2entrez[[from]]),]
  rownames(symbol2entrez) &lt;- symbol2entrez[[from]]
  ysub &lt;- y[names(y) %in% symbol2entrez[[from]]]
  names(ysub) &lt;- symbol2entrez[names(ysub),][[to]]
  return(ysub)
}


par(mfrow=c(1,1))
deseq$`CD19+ B` %&gt;% .$padj %&gt;% hist(main=&#39;CD19+B p-values&#39;)</code></pre>
<pre><code>Loading required package: DESeq2</code></pre>
<p><img src="figure/single_cell_pbmc.Rmd/data.setup-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-data.setup-1">
Past versions of data.setup-1.png
</button>
</p>
<div id="fig-data.setup-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/karltayeb/logistic-susie-gsea/blob/a2bdb5680eea3d5caf14d0b7423351c4232628b4/docs/figure/single_cell_pbmc.Rmd/data.setup-1.png" target="_blank">a2bdb56</a>
</td>
<td>
karltayeb
</td>
<td>
2022-03-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="fit-logistic-susie" class="section level3">
<h3>Fit logistic SuSiE</h3>
<pre class="r"><code>logistic_susie_driver = function(db, celltype, thresh){
  gs &lt;- genesets[[db]]
  data &lt;- deseq[[celltype]]
  
  # set up binary y
  y &lt;- data %&gt;%
    as.data.frame %&gt;%
    rownames_to_column(&#39;gene&#39;) %&gt;%
    dplyr::select(gene, padj) %&gt;%
    filter(!is.na(padj)) %&gt;%
    mutate(y = as.integer(padj &lt; thresh)) %&gt;%
    select(gene, y) %&gt;%
    tibble2namedlist %&gt;%
    convert_labels(&#39;ENSEMBL&#39;)
  
  u &lt;- process_input(gs$X, y)  # subset to common genes
  vb.fit &lt;- logistic.susie(  # fit model
    u$X, u$y, L=10, init.intercept = 0, verbose=1, maxit=100)

  # summarise results
  set.summary &lt;- vb.fit$pip %&gt;% 
    as_tibble(rownames=&#39;geneSet&#39;) %&gt;%
    rename(pip=value) %&gt;%
    mutate(
      top_component = apply(vb.fit$alpha, 2, which.max),
      active_set = top_component %in% vb.fit$sets$cs_index,
      top_component = paste0(&#39;L&#39;, top_component),
      cs = purrr::map(top_component, ~tryCatch(
        colnames(gs$X)[get(.x, vb.fit$sets$cs)], error = function(e) list())),
      in_cs = geneSet %in% cs,
      beta = colSums(vb.fit$mu * vb.fit$alpha),
      geneListSize = sum(u$y),
      geneSetSize = colSums(u$X),
      overlap = (u$y %*% u$X)[1,],
      nGenes = length(u$y),
      propSetInList = overlap / geneSetSize,
      oddsRatio = (overlap / (geneListSize - overlap)) / (
        (geneSetSize - overlap) / (nGenes - geneSetSize + overlap)),
    pValueHypergeometric = phyper(
      overlap-1, geneListSize, nGenes, geneSetSize, lower.tail= FALSE),
    db = db,
    celltype = celltype,
    thresh = thresh
    ) %&gt;% left_join(gs$geneSet$geneSetDes)
  return(list(fit = vb.fit, set.summary=set.summary))
}</code></pre>
<p>For each celltype, we fit logistic SuSiE using multiple gene set sources at various threshold of <code>padj</code>.</p>
<pre class="r"><code>celltypes &lt;- names(deseq)
pthresh &lt;- c(0.1, 0.01, 0.001, 0.0001, 0.00001, 0.000001)
db_name &lt;- names(genesets)
crossed &lt;- cross3(db_name, celltypes, pthresh)

pbmc_res &lt;- xfun::cache_rds({
  res &lt;- purrr::map(crossed, purrr::lift_dl(logistic_susie_driver))
  for (i in 1:length(res)){  # save some space
    res[[i]]$fit$dat &lt;- NULL
  }
  res
  }, file = &#39;logistic_susie_pbmc_genesets_pthresh.rds&#39;
)

pbmc_res_set_summary &lt;- dplyr::bind_rows(purrr::map(pbmc_res, ~ pluck(.x, &#39;set.summary&#39;)))</code></pre>
</div>
<div id="summary-functions" class="section level3">
<h3>Summary functions</h3>
<p>Just a few functions to help streamline looking at output</p>
<pre class="r"><code>pval_focussed_table = function(thresh=1e-3, filter_db=NULL, filter_celltype=NULL, top.n=50){
  pbmc_res_set_summary %&gt;%
  filter(
    case_when(
      is.null(filter_db) ~ TRUE,
      !is.null(filter_db) ~ db %in% filter_db
    ) &amp;
    thresh == thresh &amp;
    case_when(
      is.null(filter_celltype) ~ TRUE,
      !is.null(filter_celltype) ~ celltype %in% filter_celltype
    )
  )  %&gt;%
  dplyr::arrange(celltype, db, pValueHypergeometric) %&gt;%
  group_by(celltype, db) %&gt;% slice(1:top.n) %&gt;%
  select(celltype, db, geneSet, description, pip, top_component, oddsRatio, propSetInList, pValueHypergeometric) %&gt;%
  mutate_at(vars(celltype, db), factor) %&gt;%
  datatable(filter = &#39;top&#39;)
}

set_focussed_table = function(thresh=1e-3, filter_db=NULL, filter_celltype=NULL){
  pbmc_res_set_summary %&gt;%
  filter(
    case_when(
      is.null(filter_db) ~ TRUE,
      !is.null(filter_db) ~ db %in% filter_db
    ) &amp;
    thresh == 1e-3 &amp;
    in_cs &amp; active_set &amp;
    case_when(
      is.null(filter_celltype) ~ TRUE,
      !is.null(filter_celltype) ~ celltype %in% filter_celltype
    )
  )  %&gt;%
  dplyr::arrange(celltype, db, desc(pip)) %&gt;%
  select(celltype, db, geneSet, description, pip, top_component, oddsRatio, propSetInList, pValueHypergeometric) %&gt;%
  mutate_at(vars(celltype, geneSet, db), factor) %&gt;%
  datatable(filter = &#39;top&#39;)
}</code></pre>
</div>
</div>
<div id="resultsexplore-enrichments" class="section level2">
<h2>Results/Explore enrichments</h2>
<p>Our goal is to assess 1. The quality of the gene set enrichments we get from each celltype - do reported gene set enrichments seem celltype specific/celltype relevant? - how much “interesting” marginal enrichment do we fail to capture in the multivariate model - how sensitive are we to the choice of pvalue threshold</p>
</div>
<div id="results" class="section level2">
<h2>Results</h2>
<p>Lets take a look at what enrichment we’re getting across cell-types. ### Overview</p>
<pre class="r"><code>pbmc_res_set_summary %&gt;%
  filter(active_set, thresh==1e-4) %&gt;%
  group_by(db, celltype, top_component) %&gt;%
  arrange(db, celltype, top_component, desc(pip)) %&gt;%
  select(geneSet, description, pip) %&gt;% chop(c(geneSet, description, pip)) %&gt;%
  knitr::kable()</code></pre>
<pre><code>Adding missing grouping variables: `db`, `celltype`, `top_component`</code></pre>
<table style="width:100%;">
<colgroup>
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="11%" />
<col width="77%" />
<col width="10%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">db</th>
<th align="left">celltype</th>
<th align="left">top_component</th>
<th align="left">geneSet</th>
<th align="left">description</th>
<th align="left">pip</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">gobp</td>
<td align="left">CD14+ Monocyte</td>
<td align="left">L1</td>
<td align="left"><a href="GO:0045321" class="uri">GO:0045321</a>, <a href="GO:0001775" class="uri">GO:0001775</a></td>
<td align="left">leukocyte activation, cell activation</td>
<td align="left">0.9139560, 0.0861039</td>
</tr>
<tr class="even">
<td align="left">gobp</td>
<td align="left">CD14+ Monocyte</td>
<td align="left">L2</td>
<td align="left"><a href="GO:0045047" class="uri">GO:0045047</a>, <a href="GO:0006614" class="uri">GO:0006614</a>, <a href="GO:0006613" class="uri">GO:0006613</a>, <a href="GO:0072599" class="uri">GO:0072599</a>, <a href="GO:0070972" class="uri">GO:0070972</a></td>
<td align="left">protein targeting to ER , SRP-dependent cotranslational protein targeting to membrane , cotranslational protein targeting to membrane , establishment of protein localization to endoplasmic reticulum, protein localization to endoplasmic reticulum</td>
<td align="left">0.8615391988, 0.0574691370, 0.0477713854, 0.0336165786, 0.0004015912</td>
</tr>
<tr class="odd">
<td align="left">gobp</td>
<td align="left">CD14+ Monocyte</td>
<td align="left">L3</td>
<td align="left"><a href="GO:0006119" class="uri">GO:0006119</a>, <a href="GO:0042773" class="uri">GO:0042773</a>, <a href="GO:0042775" class="uri">GO:0042775</a></td>
<td align="left">oxidative phosphorylation , ATP synthesis coupled electron transport , mitochondrial ATP synthesis coupled electron transport</td>
<td align="left">0.9990484928, 0.0006152697, 0.0004379516</td>
</tr>
<tr class="even">
<td align="left">gobp</td>
<td align="left">CD14+ Monocyte</td>
<td align="left">L4</td>
<td align="left"><a href="GO:0016192" class="uri">GO:0016192</a>, <a href="GO:0002376" class="uri">GO:0002376</a>, <a href="GO:0006955" class="uri">GO:0006955</a>, <a href="GO:0045055" class="uri">GO:0045055</a>, <a href="GO:0006810" class="uri">GO:0006810</a>, <a href="GO:0042119" class="uri">GO:0042119</a>, <a href="GO:0006887" class="uri">GO:0006887</a>, <a href="GO:0051234" class="uri">GO:0051234</a></td>
<td align="left">vesicle-mediated transport , immune system process , immune response , regulated exocytosis , transport , neutrophil activation , exocytosis , establishment of localization</td>
<td align="left">0.9851433899, 0.0210152324, 0.0059841488, 0.0010584611, 0.0009289568, 0.0008104012, 0.0007570762, 0.0007478103</td>
</tr>
<tr class="odd">
<td align="left">gobp</td>
<td align="left">CD14+ Monocyte</td>
<td align="left">L5</td>
<td align="left"><a href="GO:0006915" class="uri">GO:0006915</a>, <a href="GO:0012501" class="uri">GO:0012501</a>, <a href="GO:0043067" class="uri">GO:0043067</a>, <a href="GO:0042981" class="uri">GO:0042981</a>, <a href="GO:0010941" class="uri">GO:0010941</a>, <a href="GO:0008219" class="uri">GO:0008219</a>, <a href="GO:0097190" class="uri">GO:0097190</a>, <a href="GO:0006950" class="uri">GO:0006950</a>, <a href="GO:0010942" class="uri">GO:0010942</a>, <a href="GO:0050790" class="uri">GO:0050790</a>, <a href="GO:0048518" class="uri">GO:0048518</a>, <a href="GO:0010033" class="uri">GO:0010033</a>, <a href="GO:0070887" class="uri">GO:0070887</a>, <a href="GO:2001233" class="uri">GO:2001233</a>, <a href="GO:0043065" class="uri">GO:0043065</a>, <a href="GO:0043068" class="uri">GO:0043068</a>, <a href="GO:0043069" class="uri">GO:0043069</a>, <a href="GO:0043066" class="uri">GO:0043066</a>, <a href="GO:0051716" class="uri">GO:0051716</a>, <a href="GO:0048522" class="uri">GO:0048522</a>, <a href="GO:0060548" class="uri">GO:0060548</a>, <a href="GO:0050896" class="uri">GO:0050896</a>, <a href="GO:0048523" class="uri">GO:0048523</a>, <a href="GO:0048519" class="uri">GO:0048519</a></td>
<td align="left">apoptotic process , programmed cell death , regulation of programmed cell death , regulation of apoptotic process , regulation of cell death , cell death , apoptotic signaling pathway , response to stress , positive regulation of cell death , regulation of catalytic activity , positive regulation of biological process , response to organic substance , cellular response to chemical stimulus , regulation of apoptotic signaling pathway , positive regulation of apoptotic process , positive regulation of programmed cell death, negative regulation of programmed cell death, negative regulation of apoptotic process , cellular response to stimulus , positive regulation of cellular process , negative regulation of cell death , response to stimulus , negative regulation of cellular process , negative regulation of biological process</td>
<td align="left">2.720176e-01, 1.948576e-01, 1.940882e-01, 1.583264e-01, 1.008376e-01, 5.132999e-02, 2.549188e-03, 9.307323e-04, 6.576479e-04, 5.735087e-04, 5.459875e-04, 4.811908e-04, 4.506495e-04, 3.491954e-04, 3.298411e-04, 2.900530e-04, 1.817724e-04, 1.476441e-04, 1.365127e-04, 1.359366e-04, 1.228290e-04, 1.125502e-04, 7.493961e-05, 6.920531e-05</td>
</tr>
<tr class="even">
<td align="left">gobp</td>
<td align="left">CD14+ Monocyte</td>
<td align="left">L6</td>
<td align="left"><a href="GO:0006518" class="uri">GO:0006518</a>, <a href="GO:0006412" class="uri">GO:0006412</a>, <a href="GO:0043043" class="uri">GO:0043043</a>, <a href="GO:0043603" class="uri">GO:0043603</a>, <a href="GO:0043604" class="uri">GO:0043604</a></td>
<td align="left">peptide metabolic process , translation , peptide biosynthetic process , cellular amide metabolic process, amide biosynthetic process</td>
<td align="left">0.790995030, 0.119614037, 0.082857945, 0.002543295, 0.001245957</td>
</tr>
<tr class="odd">
<td align="left">gobp</td>
<td align="left">CD14+ Monocyte</td>
<td align="left">L7</td>
<td align="left"><a href="GO:0008380" class="uri">GO:0008380</a>, <a href="GO:0000375" class="uri">GO:0000375</a>, <a href="GO:0000377" class="uri">GO:0000377</a>, <a href="GO:0000398" class="uri">GO:0000398</a>, <a href="GO:0006397" class="uri">GO:0006397</a>, <a href="GO:0016071" class="uri">GO:0016071</a></td>
<td align="left">RNA splicing , RNA splicing, via transesterification reactions , RNA splicing, via transesterification reactions with bulged adenosine as nucleophile, mRNA splicing, via spliceosome , mRNA processing , mRNA metabolic process</td>
<td align="left">0.422448316, 0.196794593, 0.178759471, 0.178759471, 0.021244858, 0.001609996</td>
</tr>
<tr class="even">
<td align="left">gobp</td>
<td align="left">CD19+ B</td>
<td align="left">L1</td>
<td align="left"><a href="GO:0002376" class="uri">GO:0002376</a></td>
<td align="left">immune system process</td>
<td align="left">0.9999575</td>
</tr>
<tr class="odd">
<td align="left">gobp</td>
<td align="left">CD19+ B</td>
<td align="left">L2</td>
<td align="left"><a href="GO:0045047" class="uri">GO:0045047</a>, <a href="GO:0006613" class="uri">GO:0006613</a>, <a href="GO:0072599" class="uri">GO:0072599</a>, <a href="GO:0006614" class="uri">GO:0006614</a></td>
<td align="left">protein targeting to ER , cotranslational protein targeting to membrane , establishment of protein localization to endoplasmic reticulum, SRP-dependent cotranslational protein targeting to membrane</td>
<td align="left">0.96205208, 0.02165616, 0.01453688, 0.00410197</td>
</tr>
<tr class="even">
<td align="left">gobp</td>
<td align="left">CD19+ B</td>
<td align="left">L3</td>
<td align="left"><a href="GO:0042773" class="uri">GO:0042773</a>, <a href="GO:0042775" class="uri">GO:0042775</a>, <a href="GO:0006119" class="uri">GO:0006119</a></td>
<td align="left">ATP synthesis coupled electron transport , mitochondrial ATP synthesis coupled electron transport, oxidative phosphorylation</td>
<td align="left">0.62449631, 0.36002221, 0.01752387</td>
</tr>
<tr class="odd">
<td align="left">gobp</td>
<td align="left">CD19+ B</td>
<td align="left">L5</td>
<td align="left"><a href="GO:0001775" class="uri">GO:0001775</a>, <a href="GO:0045321" class="uri">GO:0045321</a>, <a href="GO:0002366" class="uri">GO:0002366</a>, <a href="GO:0002263" class="uri">GO:0002263</a></td>
<td align="left">cell activation , leukocyte activation , leukocyte activation involved in immune response, cell activation involved in immune response</td>
<td align="left">0.9780541751, 0.0218972721, 0.0005957131, 0.0005770067</td>
</tr>
<tr class="even">
<td align="left">gobp</td>
<td align="left">CD19+ B</td>
<td align="left">L6</td>
<td align="left"><a href="GO:0008380" class="uri">GO:0008380</a>, <a href="GO:0000377" class="uri">GO:0000377</a>, <a href="GO:0000398" class="uri">GO:0000398</a>, <a href="GO:0000375" class="uri">GO:0000375</a>, <a href="GO:0006397" class="uri">GO:0006397</a>, <a href="GO:0050684" class="uri">GO:0050684</a>, <a href="GO:0043484" class="uri">GO:0043484</a>, <a href="GO:0000380" class="uri">GO:0000380</a>, <a href="GO:0048024" class="uri">GO:0048024</a>, <a href="GO:0016071" class="uri">GO:0016071</a></td>
<td align="left">RNA splicing , RNA splicing, via transesterification reactions with bulged adenosine as nucleophile, mRNA splicing, via spliceosome , RNA splicing, via transesterification reactions , mRNA processing , regulation of mRNA processing , regulation of RNA splicing , alternative mRNA splicing, via spliceosome , regulation of mRNA splicing, via spliceosome , mRNA metabolic process</td>
<td align="left">0.822736133, 0.049644134, 0.049644134, 0.033005117, 0.017234053, 0.009417204, 0.007220671, 0.002939483, 0.001834918, 0.001051637</td>
</tr>
<tr class="odd">
<td align="left">gobp</td>
<td align="left">CD34+</td>
<td align="left">L1</td>
<td align="left"><a href="GO:0001775" class="uri">GO:0001775</a>, <a href="GO:0045321" class="uri">GO:0045321</a></td>
<td align="left">cell activation , leukocyte activation</td>
<td align="left">0.9870540, 0.0137773</td>
</tr>
<tr class="even">
<td align="left">gobp</td>
<td align="left">CD34+</td>
<td align="left">L2</td>
<td align="left"><a href="GO:0006613" class="uri">GO:0006613</a>, <a href="GO:0006614" class="uri">GO:0006614</a>, <a href="GO:0045047" class="uri">GO:0045047</a>, <a href="GO:0072599" class="uri">GO:0072599</a></td>
<td align="left">cotranslational protein targeting to membrane , SRP-dependent cotranslational protein targeting to membrane , protein targeting to ER , establishment of protein localization to endoplasmic reticulum</td>
<td align="left">0.813781011, 0.153626340, 0.029062966, 0.008527809</td>
</tr>
<tr class="odd">
<td align="left">gobp</td>
<td align="left">CD34+</td>
<td align="left">L3</td>
<td align="left"><a href="GO:0006119" class="uri">GO:0006119</a>, <a href="GO:0042773" class="uri">GO:0042773</a>, <a href="GO:0042775" class="uri">GO:0042775</a></td>
<td align="left">oxidative phosphorylation , ATP synthesis coupled electron transport , mitochondrial ATP synthesis coupled electron transport</td>
<td align="left">0.998941297, 0.002197687, 0.002040038</td>
</tr>
<tr class="even">
<td align="left">gobp</td>
<td align="left">CD34+</td>
<td align="left">L4</td>
<td align="left"><a href="GO:0044419" class="uri">GO:0044419</a>, <a href="GO:0016032" class="uri">GO:0016032</a>, <a href="GO:0044403" class="uri">GO:0044403</a></td>
<td align="left">interspecies interaction between organisms, viral process , symbiont process</td>
<td align="left">0.4203517, 0.3833143, 0.1952495</td>
</tr>
<tr class="odd">
<td align="left">gobp</td>
<td align="left">CD56+ NK</td>
<td align="left">L1</td>
<td align="left"><a href="GO:0002376" class="uri">GO:0002376</a></td>
<td align="left">immune system process</td>
<td align="left">0.9999984</td>
</tr>
<tr class="even">
<td align="left">gobp</td>
<td align="left">CD56+ NK</td>
<td align="left">L2</td>
<td align="left"><a href="GO:0045047" class="uri">GO:0045047</a>, <a href="GO:0072599" class="uri">GO:0072599</a>, <a href="GO:0006614" class="uri">GO:0006614</a>, <a href="GO:0006613" class="uri">GO:0006613</a></td>
<td align="left">protein targeting to ER , establishment of protein localization to endoplasmic reticulum, SRP-dependent cotranslational protein targeting to membrane , cotranslational protein targeting to membrane</td>
<td align="left">0.68278614, 0.20978513, 0.07123053, 0.03822676</td>
</tr>
<tr class="odd">
<td align="left">gobp</td>
<td align="left">CD56+ NK</td>
<td align="left">L4</td>
<td align="left"><a href="GO:0006119" class="uri">GO:0006119</a>, <a href="GO:0009123" class="uri">GO:0009123</a>, <a href="GO:0042773" class="uri">GO:0042773</a>, <a href="GO:0042775" class="uri">GO:0042775</a>, <a href="GO:0009161" class="uri">GO:0009161</a>, <a href="GO:0009126" class="uri">GO:0009126</a>, <a href="GO:0009167" class="uri">GO:0009167</a>, <a href="GO:0022900" class="uri">GO:0022900</a>, <a href="GO:0022904" class="uri">GO:0022904</a>, <a href="GO:0046034" class="uri">GO:0046034</a></td>
<td align="left">oxidative phosphorylation , nucleoside monophosphate metabolic process , ATP synthesis coupled electron transport , mitochondrial ATP synthesis coupled electron transport, ribonucleoside monophosphate metabolic process , purine nucleoside monophosphate metabolic process , purine ribonucleoside monophosphate metabolic process , electron transport chain , respiratory electron transport chain , ATP metabolic process</td>
<td align="left">0.9815754083, 0.0075172568, 0.0050461669, 0.0035658370, 0.0031462719, 0.0018470112, 0.0018470112, 0.0017166414, 0.0015391764, 0.0009472366</td>
</tr>
<tr class="even">
<td align="left">gobp</td>
<td align="left">CD56+ NK</td>
<td align="left">L5</td>
<td align="left"><a href="GO:0008380" class="uri">GO:0008380</a>, <a href="GO:0006397" class="uri">GO:0006397</a>, <a href="GO:0000375" class="uri">GO:0000375</a>, <a href="GO:0000377" class="uri">GO:0000377</a>, <a href="GO:0000398" class="uri">GO:0000398</a>, <a href="GO:0016071" class="uri">GO:0016071</a></td>
<td align="left">RNA splicing , mRNA processing , RNA splicing, via transesterification reactions , RNA splicing, via transesterification reactions with bulged adenosine as nucleophile, mRNA splicing, via spliceosome , mRNA metabolic process</td>
<td align="left">0.59074452, 0.22252471, 0.05825611, 0.05441422, 0.05441422, 0.01452235</td>
</tr>
<tr class="odd">
<td align="left">gobp</td>
<td align="left">T cell</td>
<td align="left">L1</td>
<td align="left"><a href="GO:0001775" class="uri">GO:0001775</a>, <a href="GO:0045321" class="uri">GO:0045321</a></td>
<td align="left">cell activation , leukocyte activation</td>
<td align="left">0.998243491, 0.002864201</td>
</tr>
<tr class="even">
<td align="left">gobp</td>
<td align="left">T cell</td>
<td align="left">L2</td>
<td align="left"><a href="GO:0006119" class="uri">GO:0006119</a>, <a href="GO:0042773" class="uri">GO:0042773</a>, <a href="GO:0042775" class="uri">GO:0042775</a></td>
<td align="left">oxidative phosphorylation , ATP synthesis coupled electron transport , mitochondrial ATP synthesis coupled electron transport</td>
<td align="left">0.998867321, 0.002306692, 0.002090733</td>
</tr>
<tr class="odd">
<td align="left">gobp</td>
<td align="left">T cell</td>
<td align="left">L3</td>
<td align="left"><a href="GO:0010941" class="uri">GO:0010941</a>, <a href="GO:0008219" class="uri">GO:0008219</a>, <a href="GO:0043067" class="uri">GO:0043067</a>, <a href="GO:0042981" class="uri">GO:0042981</a>, <a href="GO:0012501" class="uri">GO:0012501</a>, <a href="GO:0006915" class="uri">GO:0006915</a>, <a href="GO:0097190" class="uri">GO:0097190</a>, <a href="GO:0048518" class="uri">GO:0048518</a>, <a href="GO:0010942" class="uri">GO:0010942</a>, <a href="GO:0048522" class="uri">GO:0048522</a>, <a href="GO:0048519" class="uri">GO:0048519</a>, <a href="GO:0043068" class="uri">GO:0043068</a>, <a href="GO:0043065" class="uri">GO:0043065</a></td>
<td align="left">regulation of cell death , cell death , regulation of programmed cell death , regulation of apoptotic process , programmed cell death , apoptotic process , apoptotic signaling pathway , positive regulation of biological process , positive regulation of cell death , positive regulation of cellular process , negative regulation of biological process , positive regulation of programmed cell death, positive regulation of apoptotic process</td>
<td align="left">0.404035437, 0.367067203, 0.059812581, 0.059590178, 0.042848010, 0.038547811, 0.020693460, 0.009702335, 0.006246010, 0.006037632, 0.005468877, 0.002090144, 0.001933994</td>
</tr>
<tr class="even">
<td align="left">gobp</td>
<td align="left">T cell</td>
<td align="left">L4</td>
<td align="left"><a href="GO:0045047" class="uri">GO:0045047</a>, <a href="GO:0006614" class="uri">GO:0006614</a>, <a href="GO:0072599" class="uri">GO:0072599</a>, <a href="GO:0070972" class="uri">GO:0070972</a>, <a href="GO:0006613" class="uri">GO:0006613</a></td>
<td align="left">protein targeting to ER , SRP-dependent cotranslational protein targeting to membrane , establishment of protein localization to endoplasmic reticulum, protein localization to endoplasmic reticulum , cotranslational protein targeting to membrane</td>
<td align="left">0.42251775, 0.37005939, 0.13809458, 0.04385614, 0.03174252</td>
</tr>
<tr class="odd">
<td align="left">gobp</td>
<td align="left">T cell</td>
<td align="left">L5</td>
<td align="left"><a href="GO:0002376" class="uri">GO:0002376</a>, <a href="GO:0006955" class="uri">GO:0006955</a></td>
<td align="left">immune system process, immune response</td>
<td align="left">0.998468901, 0.002881243</td>
</tr>
<tr class="even">
<td align="left">gobp_nr</td>
<td align="left">CD14+ Monocyte</td>
<td align="left">L1</td>
<td align="left"><a href="GO:0036230" class="uri">GO:0036230</a>, <a href="GO:0002446" class="uri">GO:0002446</a></td>
<td align="left">granulocyte activation , neutrophil mediated immunity</td>
<td align="left">0.97592803, 0.02469514</td>
</tr>
<tr class="odd">
<td align="left">gobp_nr</td>
<td align="left">CD14+ Monocyte</td>
<td align="left">L2</td>
<td align="left"><a href="GO:0006413" class="uri">GO:0006413</a></td>
<td align="left">translational initiation</td>
<td align="left">0.999994</td>
</tr>
<tr class="even">
<td align="left">gobp_nr</td>
<td align="left">CD14+ Monocyte</td>
<td align="left">L3</td>
<td align="left"><a href="GO:0009123" class="uri">GO:0009123</a>, <a href="GO:0009141" class="uri">GO:0009141</a></td>
<td align="left">nucleoside monophosphate metabolic process, nucleoside triphosphate metabolic process</td>
<td align="left">0.98835280, 0.01259324</td>
</tr>
<tr class="odd">
<td align="left">gobp_nr</td>
<td align="left">CD14+ Monocyte</td>
<td align="left">L4</td>
<td align="left"><a href="GO:0002521" class="uri">GO:0002521</a>, <a href="GO:0042110" class="uri">GO:0042110</a>, <a href="GO:0002694" class="uri">GO:0002694</a>, <a href="GO:1903706" class="uri">GO:1903706</a></td>
<td align="left">leukocyte differentiation , T cell activation , regulation of leukocyte activation, regulation of hemopoiesis</td>
<td align="left">0.619222289, 0.358464783, 0.007935017, 0.004456591</td>
</tr>
<tr class="even">
<td align="left">gobp_nr</td>
<td align="left">CD14+ Monocyte</td>
<td align="left">L5</td>
<td align="left"><a href="GO:0008380" class="uri">GO:0008380</a>, <a href="GO:0006397" class="uri">GO:0006397</a>, <a href="GO:1903311" class="uri">GO:1903311</a></td>
<td align="left">RNA splicing , mRNA processing , regulation of mRNA metabolic process</td>
<td align="left">0.876773133, 0.089632405, 0.004303895</td>
</tr>
<tr class="odd">
<td align="left">gobp_nr</td>
<td align="left">CD14+ Monocyte</td>
<td align="left">L6</td>
<td align="left"><a href="GO:0090150" class="uri">GO:0090150</a>, <a href="GO:0070972" class="uri">GO:0070972</a>, <a href="GO:0006605" class="uri">GO:0006605</a></td>
<td align="left">establishment of protein localization to membrane, protein localization to endoplasmic reticulum , protein targeting</td>
<td align="left">0.756201036, 0.243465201, 0.004950252</td>
</tr>
<tr class="even">
<td align="left">gobp_nr</td>
<td align="left">CD19+ B</td>
<td align="left">L1</td>
<td align="left"><a href="GO:0002446" class="uri">GO:0002446</a>, <a href="GO:0036230" class="uri">GO:0036230</a></td>
<td align="left">neutrophil mediated immunity, granulocyte activation</td>
<td align="left">0.5151440, 0.4857358</td>
</tr>
<tr class="odd">
<td align="left">gobp_nr</td>
<td align="left">CD19+ B</td>
<td align="left">L2</td>
<td align="left"><a href="GO:0070972" class="uri">GO:0070972</a></td>
<td align="left">protein localization to endoplasmic reticulum</td>
<td align="left">0.999958</td>
</tr>
<tr class="even">
<td align="left">gobp_nr</td>
<td align="left">CD19+ B</td>
<td align="left">L4</td>
<td align="left"><a href="GO:0009123" class="uri">GO:0009123</a>, <a href="GO:0009141" class="uri">GO:0009141</a></td>
<td align="left">nucleoside monophosphate metabolic process, nucleoside triphosphate metabolic process</td>
<td align="left">0.98449756, 0.01648675</td>
</tr>
<tr class="odd">
<td align="left">gobp_nr</td>
<td align="left">CD19+ B</td>
<td align="left">L5</td>
<td align="left"><a href="GO:0002764" class="uri">GO:0002764</a></td>
<td align="left">immune response-regulating signaling pathway</td>
<td align="left">0.9961853</td>
</tr>
<tr class="even">
<td align="left">gobp_nr</td>
<td align="left">CD34+</td>
<td align="left">L1</td>
<td align="left"><a href="GO:0002446" class="uri">GO:0002446</a>, <a href="GO:0036230" class="uri">GO:0036230</a></td>
<td align="left">neutrophil mediated immunity, granulocyte activation</td>
<td align="left">0.6078439, 0.3946596</td>
</tr>
<tr class="odd">
<td align="left">gobp_nr</td>
<td align="left">CD34+</td>
<td align="left">L2</td>
<td align="left"><a href="GO:0006413" class="uri">GO:0006413</a></td>
<td align="left">translational initiation</td>
<td align="left">0.9996608</td>
</tr>
<tr class="even">
<td align="left">gobp_nr</td>
<td align="left">CD56+ NK</td>
<td align="left">L1</td>
<td align="left"><a href="GO:0036230" class="uri">GO:0036230</a>, <a href="GO:0002446" class="uri">GO:0002446</a></td>
<td align="left">granulocyte activation , neutrophil mediated immunity</td>
<td align="left">0.6655074, 0.3354429</td>
</tr>
<tr class="odd">
<td align="left">gobp_nr</td>
<td align="left">CD56+ NK</td>
<td align="left">L2</td>
<td align="left"><a href="GO:0006413" class="uri">GO:0006413</a></td>
<td align="left">translational initiation</td>
<td align="left">0.9954877</td>
</tr>
<tr class="even">
<td align="left">gobp_nr</td>
<td align="left">CD56+ NK</td>
<td align="left">L3</td>
<td align="left"><a href="GO:0042110" class="uri">GO:0042110</a>, <a href="GO:0007159" class="uri">GO:0007159</a></td>
<td align="left">T cell activation , leukocyte cell-cell adhesion</td>
<td align="left">0.985660130, 0.008171918</td>
</tr>
<tr class="odd">
<td align="left">gobp_nr</td>
<td align="left">CD56+ NK</td>
<td align="left">L4</td>
<td align="left"><a href="GO:0009123" class="uri">GO:0009123</a>, <a href="GO:0009141" class="uri">GO:0009141</a>, <a href="GO:0009259" class="uri">GO:0009259</a></td>
<td align="left">nucleoside monophosphate metabolic process, nucleoside triphosphate metabolic process , ribonucleotide metabolic process</td>
<td align="left">0.981592710, 0.015904774, 0.001372449</td>
</tr>
<tr class="even">
<td align="left">gobp_nr</td>
<td align="left">CD56+ NK</td>
<td align="left">L5</td>
<td align="left"><a href="GO:0042113" class="uri">GO:0042113</a></td>
<td align="left">B cell activation</td>
<td align="left">0.9972725</td>
</tr>
<tr class="odd">
<td align="left">gobp_nr</td>
<td align="left">T cell</td>
<td align="left">L1</td>
<td align="left"><a href="GO:0036230" class="uri">GO:0036230</a>, <a href="GO:0002446" class="uri">GO:0002446</a></td>
<td align="left">granulocyte activation , neutrophil mediated immunity</td>
<td align="left">0.7324297, 0.2703360</td>
</tr>
<tr class="even">
<td align="left">gobp_nr</td>
<td align="left">T cell</td>
<td align="left">L2</td>
<td align="left"><a href="GO:0042110" class="uri">GO:0042110</a></td>
<td align="left">T cell activation</td>
<td align="left">0.9999906</td>
</tr>
<tr class="odd">
<td align="left">gobp_nr</td>
<td align="left">T cell</td>
<td align="left">L3</td>
<td align="left"><a href="GO:0070972" class="uri">GO:0070972</a></td>
<td align="left">protein localization to endoplasmic reticulum</td>
<td align="left">0.9998559</td>
</tr>
<tr class="even">
<td align="left">gomf</td>
<td align="left">CD14+ Monocyte</td>
<td align="left">L1</td>
<td align="left"><a href="GO:0003723" class="uri">GO:0003723</a></td>
<td align="left">RNA binding</td>
<td align="left">0.9999998</td>
</tr>
<tr class="odd">
<td align="left">gomf</td>
<td align="left">CD14+ Monocyte</td>
<td align="left">L2</td>
<td align="left"><a href="GO:0000981" class="uri">GO:0000981</a>, <a href="GO:0003700" class="uri">GO:0003700</a></td>
<td align="left">DNA-binding transcription factor activity, RNA polymerase II-specific, DNA-binding transcription factor activity</td>
<td align="left">0.9998274825, 0.0002279895</td>
</tr>
<tr class="even">
<td align="left">gomf</td>
<td align="left">CD14+ Monocyte</td>
<td align="left">L3</td>
<td align="left"><a href="GO:0001012" class="uri">GO:0001012</a>, <a href="GO:0000977" class="uri">GO:0000977</a>, <a href="GO:0000987" class="uri">GO:0000987</a>, <a href="GO:0000978" class="uri">GO:0000978</a>, <a href="GO:0000976" class="uri">GO:0000976</a>, <a href="GO:1990837" class="uri">GO:1990837</a>, <a href="GO:0043565" class="uri">GO:0043565</a>, <a href="GO:0044212" class="uri">GO:0044212</a></td>
<td align="left">RNA polymerase II regulatory region DNA binding , RNA polymerase II regulatory region sequence-specific DNA binding, proximal promoter sequence-specific DNA binding , RNA polymerase II proximal promoter sequence-specific DNA binding, transcription regulatory region sequence-specific DNA binding , sequence-specific double-stranded DNA binding , sequence-specific DNA binding , transcription regulatory region DNA binding</td>
<td align="left">0.6603205658, 0.2924363216, 0.0360620592, 0.0077244907, 0.0030829257, 0.0007025284, 0.0003868975, 0.0001017222</td>
</tr>
<tr class="odd">
<td align="left">gomf</td>
<td align="left">CD14+ Monocyte</td>
<td align="left">L4</td>
<td align="left"><a href="GO:0003735" class="uri">GO:0003735</a></td>
<td align="left">structural constituent of ribosome</td>
<td align="left">1</td>
</tr>
<tr class="even">
<td align="left">gomf</td>
<td align="left">CD14+ Monocyte</td>
<td align="left">L5</td>
<td align="left"><a href="GO:0016655" class="uri">GO:0016655</a>, <a href="GO:0008137" class="uri">GO:0008137</a>, <a href="GO:0050136" class="uri">GO:0050136</a>, <a href="GO:0003954" class="uri">GO:0003954</a>, <a href="GO:0016651" class="uri">GO:0016651</a></td>
<td align="left">oxidoreductase activity, acting on NAD(P)H, quinone or similar compound as acceptor, NADH dehydrogenase (ubiquinone) activity , NADH dehydrogenase (quinone) activity , NADH dehydrogenase activity , oxidoreductase activity, acting on NAD(P)H</td>
<td align="left">0.8270835588, 0.0778621792, 0.0778621792, 0.0178702201, 0.0003469471</td>
</tr>
<tr class="odd">
<td align="left">gomf</td>
<td align="left">CD14+ Monocyte</td>
<td align="left">L6</td>
<td align="left"><a href="GO:0045296" class="uri">GO:0045296</a>, <a href="GO:0050839" class="uri">GO:0050839</a></td>
<td align="left">cadherin binding , cell adhesion molecule binding</td>
<td align="left">0.7890801, 0.1881604</td>
</tr>
<tr class="even">
<td align="left">gomf</td>
<td align="left">CD19+ B</td>
<td align="left">L1</td>
<td align="left"><a href="GO:0003723" class="uri">GO:0003723</a></td>
<td align="left">RNA binding</td>
<td align="left">0.999744</td>
</tr>
<tr class="odd">
<td align="left">gomf</td>
<td align="left">CD19+ B</td>
<td align="left">L2</td>
<td align="left"><a href="GO:0000981" class="uri">GO:0000981</a>, <a href="GO:0003700" class="uri">GO:0003700</a></td>
<td align="left">DNA-binding transcription factor activity, RNA polymerase II-specific, DNA-binding transcription factor activity</td>
<td align="left">0.995003783, 0.005058241</td>
</tr>
<tr class="even">
<td align="left">gomf</td>
<td align="left">CD19+ B</td>
<td align="left">L3</td>
<td align="left"><a href="GO:0000977" class="uri">GO:0000977</a>, <a href="GO:0001012" class="uri">GO:0001012</a>, <a href="GO:0000987" class="uri">GO:0000987</a>, <a href="GO:0000978" class="uri">GO:0000978</a>, <a href="GO:0000976" class="uri">GO:0000976</a>, <a href="GO:0003690" class="uri">GO:0003690</a>, <a href="GO:1990837" class="uri">GO:1990837</a>, <a href="GO:0001067" class="uri">GO:0001067</a>, <a href="GO:0044212" class="uri">GO:0044212</a></td>
<td align="left">RNA polymerase II regulatory region sequence-specific DNA binding, RNA polymerase II regulatory region DNA binding , proximal promoter sequence-specific DNA binding , RNA polymerase II proximal promoter sequence-specific DNA binding, transcription regulatory region sequence-specific DNA binding , double-stranded DNA binding , sequence-specific double-stranded DNA binding , regulatory region nucleic acid binding , transcription regulatory region DNA binding</td>
<td align="left">0.4872513913, 0.4313993168, 0.0365953412, 0.0243679373, 0.0132413813, 0.0059278956, 0.0013679037, 0.0004652721, 0.0003557059</td>
</tr>
<tr class="odd">
<td align="left">gomf</td>
<td align="left">CD19+ B</td>
<td align="left">L6</td>
<td align="left"><a href="GO:0003735" class="uri">GO:0003735</a></td>
<td align="left">structural constituent of ribosome</td>
<td align="left">0.9953441</td>
</tr>
<tr class="even">
<td align="left">gomf</td>
<td align="left">CD19+ B</td>
<td align="left">L7</td>
<td align="left"><a href="GO:0003954" class="uri">GO:0003954</a>, <a href="GO:0008137" class="uri">GO:0008137</a>, <a href="GO:0050136" class="uri">GO:0050136</a>, <a href="GO:0016655" class="uri">GO:0016655</a>, <a href="GO:0016651" class="uri">GO:0016651</a></td>
<td align="left">NADH dehydrogenase activity , NADH dehydrogenase (ubiquinone) activity , NADH dehydrogenase (quinone) activity , oxidoreductase activity, acting on NAD(P)H, quinone or similar compound as acceptor, oxidoreductase activity, acting on NAD(P)H</td>
<td align="left">0.32595362, 0.21858318, 0.21858318, 0.17726552, 0.03879761</td>
</tr>
<tr class="odd">
<td align="left">gomf</td>
<td align="left">CD34+</td>
<td align="left">L1</td>
<td align="left"><a href="GO:0003735" class="uri">GO:0003735</a></td>
<td align="left">structural constituent of ribosome</td>
<td align="left">1</td>
</tr>
<tr class="even">
<td align="left">gomf</td>
<td align="left">CD34+</td>
<td align="left">L2</td>
<td align="left"><a href="GO:0005515" class="uri">GO:0005515</a></td>
<td align="left">protein binding</td>
<td align="left">1</td>
</tr>
<tr class="odd">
<td align="left">gomf</td>
<td align="left">CD56+ NK</td>
<td align="left">L1</td>
<td align="left"><a href="GO:0003735" class="uri">GO:0003735</a></td>
<td align="left">structural constituent of ribosome</td>
<td align="left">1</td>
</tr>
<tr class="even">
<td align="left">gomf</td>
<td align="left">CD56+ NK</td>
<td align="left">L2</td>
<td align="left"><a href="GO:0000981" class="uri">GO:0000981</a>, <a href="GO:0003700" class="uri">GO:0003700</a></td>
<td align="left">DNA-binding transcription factor activity, RNA polymerase II-specific, DNA-binding transcription factor activity</td>
<td align="left">0.996770584, 0.003356549</td>
</tr>
<tr class="odd">
<td align="left">gomf</td>
<td align="left">CD56+ NK</td>
<td align="left">L3</td>
<td align="left"><a href="GO:0000977" class="uri">GO:0000977</a>, <a href="GO:0000976" class="uri">GO:0000976</a>, <a href="GO:0001012" class="uri">GO:0001012</a>, <a href="GO:1990837" class="uri">GO:1990837</a>, <a href="GO:0003690" class="uri">GO:0003690</a>, <a href="GO:0043565" class="uri">GO:0043565</a>, <a href="GO:0000987" class="uri">GO:0000987</a>, <a href="GO:0001067" class="uri">GO:0001067</a>, <a href="GO:0044212" class="uri">GO:0044212</a></td>
<td align="left">RNA polymerase II regulatory region sequence-specific DNA binding, transcription regulatory region sequence-specific DNA binding , RNA polymerase II regulatory region DNA binding , sequence-specific double-stranded DNA binding , double-stranded DNA binding , sequence-specific DNA binding , proximal promoter sequence-specific DNA binding , regulatory region nucleic acid binding , transcription regulatory region DNA binding</td>
<td align="left">0.5246638008, 0.2816905881, 0.1760785081, 0.0110809288, 0.0067171732, 0.0006260858, 0.0003679068, 0.0002517074, 0.0002268130</td>
</tr>
<tr class="even">
<td align="left">gomf</td>
<td align="left">T cell</td>
<td align="left">L1</td>
<td align="left"><a href="GO:0005515" class="uri">GO:0005515</a></td>
<td align="left">protein binding</td>
<td align="left">1</td>
</tr>
<tr class="odd">
<td align="left">kegg</td>
<td align="left">CD14+ Monocyte</td>
<td align="left">L1</td>
<td align="left">hsa03010</td>
<td align="left">Ribosome</td>
<td align="left">1</td>
</tr>
<tr class="even">
<td align="left">kegg</td>
<td align="left">CD14+ Monocyte</td>
<td align="left">L2</td>
<td align="left">hsa05012</td>
<td align="left">Parkinson disease</td>
<td align="left">0.9999021</td>
</tr>
<tr class="odd">
<td align="left">kegg</td>
<td align="left">CD19+ B</td>
<td align="left">L1</td>
<td align="left">hsa00190, hsa05012</td>
<td align="left">Oxidative phosphorylation, Parkinson disease</td>
<td align="left">0.9979902, 0.0125303</td>
</tr>
<tr class="even">
<td align="left">kegg</td>
<td align="left">CD19+ B</td>
<td align="left">L2</td>
<td align="left">hsa04640</td>
<td align="left">Hematopoietic cell lineage</td>
<td align="left">0.9997951</td>
</tr>
<tr class="odd">
<td align="left">kegg</td>
<td align="left">CD19+ B</td>
<td align="left">L3</td>
<td align="left">hsa03010</td>
<td align="left">Ribosome</td>
<td align="left">1</td>
</tr>
<tr class="even">
<td align="left">kegg</td>
<td align="left">CD34+</td>
<td align="left">L1</td>
<td align="left">hsa00190, hsa05012</td>
<td align="left">Oxidative phosphorylation, Parkinson disease</td>
<td align="left">0.9921759, 0.0333608</td>
</tr>
<tr class="odd">
<td align="left">kegg</td>
<td align="left">CD34+</td>
<td align="left">L2</td>
<td align="left">hsa03010</td>
<td align="left">Ribosome</td>
<td align="left">1</td>
</tr>
<tr class="even">
<td align="left">kegg</td>
<td align="left">CD56+ NK</td>
<td align="left">L1</td>
<td align="left">hsa03010</td>
<td align="left">Ribosome</td>
<td align="left">1</td>
</tr>
<tr class="odd">
<td align="left">kegg</td>
<td align="left">CD56+ NK</td>
<td align="left">L2</td>
<td align="left">hsa05012</td>
<td align="left">Parkinson disease</td>
<td align="left">0.9999978</td>
</tr>
<tr class="even">
<td align="left">kegg</td>
<td align="left">CD56+ NK</td>
<td align="left">L3</td>
<td align="left">hsa04640</td>
<td align="left">Hematopoietic cell lineage</td>
<td align="left">0.9978541</td>
</tr>
<tr class="odd">
<td align="left">kegg</td>
<td align="left">T cell</td>
<td align="left">L1</td>
<td align="left">hsa05010</td>
<td align="left">Alzheimer disease</td>
<td align="left">0.9999993</td>
</tr>
<tr class="even">
<td align="left">reactome</td>
<td align="left">CD14+ Monocyte</td>
<td align="left">L1</td>
<td align="left">R-HSA-6798695</td>
<td align="left">Neutrophil degranulation</td>
<td align="left">0.9999954</td>
</tr>
<tr class="odd">
<td align="left">reactome</td>
<td align="left">CD14+ Monocyte</td>
<td align="left">L2</td>
<td align="left">R-HSA-72766</td>
<td align="left">Translation</td>
<td align="left">1</td>
</tr>
<tr class="even">
<td align="left">reactome</td>
<td align="left">CD14+ Monocyte</td>
<td align="left">L3</td>
<td align="left">R-HSA-163200, R-HSA-611105</td>
<td align="left">Respiratory electron transport, ATP synthesis by chemiosmotic coupling, and heat production by uncoupling proteins., Respiratory electron transport</td>
<td align="left">0.9996110591, 0.0008392269</td>
</tr>
<tr class="odd">
<td align="left">reactome</td>
<td align="left">CD14+ Monocyte</td>
<td align="left">L5</td>
<td align="left">R-HSA-72172, R-HSA-72163, R-HSA-72203</td>
<td align="left">mRNA Splicing , mRNA Splicing - Major Pathway , Processing of Capped Intron-Containing Pre-mRNA</td>
<td align="left">0.67495558, 0.30641144, 0.01826453</td>
</tr>
<tr class="even">
<td align="left">reactome</td>
<td align="left">CD14+ Monocyte</td>
<td align="left">L6</td>
<td align="left">R-HSA-198933</td>
<td align="left">Immunoregulatory interactions between a Lymphoid and a non-Lymphoid cell</td>
<td align="left">0.9974877</td>
</tr>
<tr class="odd">
<td align="left">reactome</td>
<td align="left">CD14+ Monocyte</td>
<td align="left">L7</td>
<td align="left">R-HSA-379726</td>
<td align="left">Mitochondrial tRNA aminoacylation</td>
<td align="left">0.9972986</td>
</tr>
<tr class="even">
<td align="left">reactome</td>
<td align="left">CD19+ B</td>
<td align="left">L1</td>
<td align="left">R-HSA-168256</td>
<td align="left">Immune System</td>
<td align="left">1</td>
</tr>
<tr class="odd">
<td align="left">reactome</td>
<td align="left">CD19+ B</td>
<td align="left">L2</td>
<td align="left">R-HSA-156842 , R-HSA-156902 , R-HSA-192823 , R-HSA-2408557, R-HSA-72764 , R-HSA-72689 , R-HSA-1799339, R-HSA-72706 , R-HSA-156827</td>
<td align="left">Eukaryotic Translation Elongation , Peptide chain elongation , Viral mRNA Translation , Selenocysteine synthesis , Eukaryotic Translation Termination , Formation of a pool of free 40S subunits , SRP-dependent cotranslational protein targeting to membrane , GTP hydrolysis and joining of the 60S ribosomal subunit , L13a-mediated translational silencing of Ceruloplasmin expression</td>
<td align="left">0.760651464, 0.124389740, 0.067132923, 0.026006681, 0.017267678, 0.003305376, 0.003131980, 0.001226762, 0.001000830</td>
</tr>
<tr class="even">
<td align="left">reactome</td>
<td align="left">CD19+ B</td>
<td align="left">L3</td>
<td align="left">R-HSA-74160 , R-HSA-212436, R-HSA-73857</td>
<td align="left">Gene expression (Transcription), Generic Transcription Pathway , RNA Polymerase II Transcription</td>
<td align="left">0.88567935, 0.07266257, 0.04183476</td>
</tr>
<tr class="odd">
<td align="left">reactome</td>
<td align="left">CD19+ B</td>
<td align="left">L4</td>
<td align="left">R-HSA-163200 , R-HSA-1428517, R-HSA-611105</td>
<td align="left">Respiratory electron transport, ATP synthesis by chemiosmotic coupling, and heat production by uncoupling proteins., The citric acid (TCA) cycle and respiratory electron transport , Respiratory electron transport</td>
<td align="left">0.845323000, 0.149383366, 0.005965744</td>
</tr>
<tr class="even">
<td align="left">reactome</td>
<td align="left">CD19+ B</td>
<td align="left">L5</td>
<td align="left">R-HSA-983168, R-HSA-983169</td>
<td align="left">Antigen processing: Ubiquitination &amp; Proteasome degradation, Class I MHC mediated antigen processing &amp; presentation</td>
<td align="left">0.997049696, 0.003118792</td>
</tr>
<tr class="odd">
<td align="left">reactome</td>
<td align="left">CD19+ B</td>
<td align="left">L6</td>
<td align="left">R-HSA-180585 , R-HSA-1234176, R-HSA-211733 , R-HSA-69601 , R-HSA-69610 , R-HSA-69613 , R-HSA-69541 , R-HSA-69229 , R-HSA-75815 , R-HSA-180534 , R-HSA-169911 , R-HSA-349425 , R-HSA-9604323, R-HSA-8854050, R-HSA-1236978, R-HSA-174154 , R-HSA-1169091, R-HSA-174084 , R-HSA-1234174, R-HSA-2262749, R-HSA-983705 , R-HSA-174113 , R-HSA-450408 , R-HSA-69563 , R-HSA-69580 , R-HSA-174184 , R-HSA-351202 , R-HSA-68867 , R-HSA-176409 , R-HSA-350562 , R-HSA-69615 , R-HSA-5362768, R-HSA-174178 , R-HSA-179419 , R-HSA-5610780</td>
<td align="left">Vif-mediated degradation of APOBEC3G , Oxygen-dependent proline hydroxylation of Hypoxia-inducible Factor Alpha , Regulation of activated PAK-2p34 by proteasome mediated degradation , Ubiquitin Mediated Degradation of Phosphorylated Cdc25A , p53-Independent DNA Damage Response , p53-Independent G1/S DNA damage checkpoint , Stabilization of p53 , Ubiquitin-dependent degradation of Cyclin D1 , Ubiquitin-dependent degradation of Cyclin D , Vpu mediated degradation of CD4 , Regulation of Apoptosis , Autodegradation of the E3 ubiquitin ligase COP1 , Negative regulation of NOTCH4 signaling , FBXL7 down-regulates AURKA during mitotic entry and in early mitosis , Cross-presentation of soluble exogenous antigens (endosomes) , APC/C:Cdc20 mediated degradation of Securin , Activation of NF-kappaB in B cells , Autodegradation of Cdh1 by Cdh1:APC/C , Regulation of Hypoxia-inducible Factor (HIF) by oxygen , Cellular response to hypoxia , Signaling by the B Cell Receptor (BCR) , SCF-beta-TrCP mediated degradation of Emi1 , AUF1 (hnRNP D0) binds and destabilizes mRNA , p53-Dependent G1 DNA Damage Response , p53-Dependent G1/S DNA damage checkpoint , Cdc20:Phospho-APC/C mediated degradation of Cyclin A , Metabolism of polyamines , Assembly of the pre-replicative complex , APC/C:Cdc20 mediated degradation of mitotic proteins , Regulation of ornithine decarboxylase (ODC) , G1/S DNA Damage Checkpoints , Hh mutants that don’t undergo autocatalytic processing are degraded by ERAD , APC/C:Cdh1 mediated degradation of Cdc20 and other APC/C:Cdh1 targeted proteins in late mitosis/early G1, APC:Cdc20 mediated degradation of cell cycle proteins prior to satisfation of the cell cycle checkpoint , Degradation of GLI1 by the proteasome</td>
<td align="left">0.7490393108, 0.0319137209, 0.0256472437, 0.0242873384, 0.0242873384, 0.0242873384, 0.0225422403, 0.0120133659, 0.0120133659, 0.0119846233, 0.0108046763, 0.0101911528, 0.0071579856, 0.0070377887, 0.0040655484, 0.0038589122, 0.0037679238, 0.0036141487, 0.0030579708, 0.0030579708, 0.0028683640, 0.0024109404, 0.0017126798, 0.0015009738, 0.0015009738, 0.0013319088, 0.0013173159, 0.0012105047, 0.0011649631, 0.0010246799, 0.0009578296, 0.0009514080, 0.0008622334, 0.0007207196, 0.0006610228</td>
</tr>
<tr class="even">
<td align="left">reactome</td>
<td align="left">CD19+ B</td>
<td align="left">L7</td>
<td align="left">R-HSA-72163 , R-HSA-72172 , R-HSA-109688, R-HSA-73856 , R-HSA-72203</td>
<td align="left">mRNA Splicing - Major Pathway , mRNA Splicing , Cleavage of Growing Transcript in the Termination Region, RNA Polymerase II Transcription Termination , Processing of Capped Intron-Containing Pre-mRNA</td>
<td align="left">0.5457887505, 0.4439546364, 0.0021482288, 0.0021482288, 0.0006461411</td>
</tr>
<tr class="odd">
<td align="left">reactome</td>
<td align="left">CD19+ B</td>
<td align="left">L8</td>
<td align="left">R-HSA-76005 , R-HSA-114608, R-HSA-76002 , R-HSA-109582</td>
<td align="left">Response to elevated platelet cytosolic Ca2+ , Platelet degranulation , Platelet activation, signaling and aggregation, Hemostasis</td>
<td align="left">0.7063848487, 0.2314322026, 0.0252623002, 0.0004075344</td>
</tr>
<tr class="even">
<td align="left">reactome</td>
<td align="left">CD34+</td>
<td align="left">L1</td>
<td align="left">R-HSA-72764 , R-HSA-156902, R-HSA-156842</td>
<td align="left">Eukaryotic Translation Termination, Peptide chain elongation , Eukaryotic Translation Elongation</td>
<td align="left">0.993056233, 0.006377217, 0.004060368</td>
</tr>
<tr class="odd">
<td align="left">reactome</td>
<td align="left">CD34+</td>
<td align="left">L2</td>
<td align="left">R-HSA-6798695, R-HSA-168249</td>
<td align="left">Neutrophil degranulation, Innate Immune System</td>
<td align="left">0.98627161, 0.01669373</td>
</tr>
<tr class="even">
<td align="left">reactome</td>
<td align="left">CD34+</td>
<td align="left">L3</td>
<td align="left">R-HSA-163200, R-HSA-611105</td>
<td align="left">Respiratory electron transport, ATP synthesis by chemiosmotic coupling, and heat production by uncoupling proteins., Respiratory electron transport</td>
<td align="left">0.997732525, 0.003758523</td>
</tr>
<tr class="odd">
<td align="left">reactome</td>
<td align="left">CD34+</td>
<td align="left">L4</td>
<td align="left">R-HSA-198933</td>
<td align="left">Immunoregulatory interactions between a Lymphoid and a non-Lymphoid cell</td>
<td align="left">0.9999503</td>
</tr>
<tr class="even">
<td align="left">reactome</td>
<td align="left">CD56+ NK</td>
<td align="left">L1</td>
<td align="left">R-HSA-168256</td>
<td align="left">Immune System</td>
<td align="left">1</td>
</tr>
<tr class="odd">
<td align="left">reactome</td>
<td align="left">CD56+ NK</td>
<td align="left">L2</td>
<td align="left">R-HSA-156842, R-HSA-156902, R-HSA-192823, R-HSA-72764 , R-HSA-72689</td>
<td align="left">Eukaryotic Translation Elongation , Peptide chain elongation , Viral mRNA Translation , Eukaryotic Translation Termination , Formation of a pool of free 40S subunits</td>
<td align="left">0.864742537, 0.106897566, 0.027158260, 0.002413175, 0.001418708</td>
</tr>
<tr class="even">
<td align="left">reactome</td>
<td align="left">CD56+ NK</td>
<td align="left">L3</td>
<td align="left">R-HSA-212436, R-HSA-74160 , R-HSA-73857</td>
<td align="left">Generic Transcription Pathway , Gene expression (Transcription), RNA Polymerase II Transcription</td>
<td align="left">0.68127740, 0.29679295, 0.02222635</td>
</tr>
<tr class="odd">
<td align="left">reactome</td>
<td align="left">CD56+ NK</td>
<td align="left">L4</td>
<td align="left">R-HSA-163200 , R-HSA-1428517, R-HSA-611105</td>
<td align="left">Respiratory electron transport, ATP synthesis by chemiosmotic coupling, and heat production by uncoupling proteins., The citric acid (TCA) cycle and respiratory electron transport , Respiratory electron transport</td>
<td align="left">0.993324533, 0.006034017, 0.001575567</td>
</tr>
<tr class="even">
<td align="left">reactome</td>
<td align="left">CD56+ NK</td>
<td align="left">L5</td>
<td align="left">R-HSA-8878171, R-HSA-157118 , R-HSA-8939236</td>
<td align="left">Transcriptional regulation by RUNX1 , Signaling by NOTCH , RUNX1 regulates transcription of genes involved in differentiation of HSCs</td>
<td align="left">0.97128181, 0.04393861, 0.01998012</td>
</tr>
<tr class="odd">
<td align="left">reactome</td>
<td align="left">CD56+ NK</td>
<td align="left">L6</td>
<td align="left">R-HSA-983168 , R-HSA-983169 , R-HSA-8951664</td>
<td align="left">Antigen processing: Ubiquitination &amp; Proteasome degradation, Class I MHC mediated antigen processing &amp; presentation , Neddylation</td>
<td align="left">0.8808248324, 0.1189949639, 0.0009720837</td>
</tr>
<tr class="even">
<td align="left">reactome</td>
<td align="left">T cell</td>
<td align="left">L1</td>
<td align="left">R-HSA-6798695</td>
<td align="left">Neutrophil degranulation</td>
<td align="left">1</td>
</tr>
<tr class="odd">
<td align="left">reactome</td>
<td align="left">T cell</td>
<td align="left">L2</td>
<td align="left">R-HSA-72706 , R-HSA-156827, R-HSA-72689 , R-HSA-72613 , R-HSA-72737 , R-HSA-192823, R-HSA-156902</td>
<td align="left">GTP hydrolysis and joining of the 60S ribosomal subunit , L13a-mediated translational silencing of Ceruloplasmin expression, Formation of a pool of free 40S subunits , Eukaryotic Translation Initiation , Cap-dependent Translation Initiation , Viral mRNA Translation , Peptide chain elongation</td>
<td align="left">0.586005479, 0.400596683, 0.009180040, 0.004668524, 0.004668524, 0.003496445, 0.003174071</td>
</tr>
<tr class="even">
<td align="left">wikipathway</td>
<td align="left">CD14+ Monocyte</td>
<td align="left">L1</td>
<td align="left">WP477</td>
<td align="left">Cytoplasmic Ribosomal Proteins</td>
<td align="left">1</td>
</tr>
<tr class="odd">
<td align="left">wikipathway</td>
<td align="left">CD14+ Monocyte</td>
<td align="left">L2</td>
<td align="left">WP111</td>
<td align="left">Electron Transport Chain (OXPHOS system in mitochondria)</td>
<td align="left">0.9998312</td>
</tr>
<tr class="even">
<td align="left">wikipathway</td>
<td align="left">CD19+ B</td>
<td align="left">L1</td>
<td align="left">WP477</td>
<td align="left">Cytoplasmic Ribosomal Proteins</td>
<td align="left">1</td>
</tr>
<tr class="odd">
<td align="left">wikipathway</td>
<td align="left">CD19+ B</td>
<td align="left">L2</td>
<td align="left">WP111</td>
<td align="left">Electron Transport Chain (OXPHOS system in mitochondria)</td>
<td align="left">0.9999189</td>
</tr>
<tr class="even">
<td align="left">wikipathway</td>
<td align="left">CD34+</td>
<td align="left">L1</td>
<td align="left">WP477</td>
<td align="left">Cytoplasmic Ribosomal Proteins</td>
<td align="left">1</td>
</tr>
<tr class="odd">
<td align="left">wikipathway</td>
<td align="left">CD34+</td>
<td align="left">L2</td>
<td align="left">WP111</td>
<td align="left">Electron Transport Chain (OXPHOS system in mitochondria)</td>
<td align="left">0.9999971</td>
</tr>
<tr class="even">
<td align="left">wikipathway</td>
<td align="left">CD56+ NK</td>
<td align="left">L1</td>
<td align="left">WP477</td>
<td align="left">Cytoplasmic Ribosomal Proteins</td>
<td align="left">1</td>
</tr>
<tr class="odd">
<td align="left">wikipathway</td>
<td align="left">CD56+ NK</td>
<td align="left">L2</td>
<td align="left">WP111, WP623</td>
<td align="left">Electron Transport Chain (OXPHOS system in mitochondria), Oxidative phosphorylation</td>
<td align="left">0.99086202, 0.01897664</td>
</tr>
<tr class="even">
<td align="left">wikipathway_cancer</td>
<td align="left">CD19+ B</td>
<td align="left">L1</td>
<td align="left">WP619</td>
<td align="left">Type II interferon signaling (IFNG)</td>
<td align="left">0.9998636</td>
</tr>
</tbody>
</table>
<pre class="r"><code>knitr::knit_exit()</code></pre>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
