<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="karltayeb" />

<meta name="date" content="2022-03-23" />

<title>(Univariate) GSEA with gene level summary statistics</title>

<script src="site_libs/header-attrs-2.13/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>





<style type="text/css">
/* for pandoc --citeproc since 2.11 */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">logistic-susie-gsea</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/karltayeb/logistic-susie-gsea">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">(Univariate) GSEA with gene level summary statistics</h1>
<h4 class="author">karltayeb</h4>
<h4 class="date">2022-03-23</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-03-29
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>logistic-susie-gsea/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220105code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20220105)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220105code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220105)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomkarltayeblogisticsusiegseatree122deece85c32abc1d72b089a34a2f24c62d8972targetblank122deeca"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/karltayeb/logistic-susie-gsea/tree/122deece85c32abc1d72b089a34a2f24c62d8972" target="_blank">122deec</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomkarltayeblogisticsusiegseatree122deece85c32abc1d72b089a34a2f24c62d8972targetblank122deeca" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/karltayeb/logistic-susie-gsea/tree/122deece85c32abc1d72b089a34a2f24c62d8972" target="_blank">122deec</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    library/
    Ignored:    renv/library/
    Ignored:    renv/staging/
    Ignored:    staging/

Untracked files:
    Untracked:  _targets.R
    Untracked:  _targets.html
    Untracked:  _targets.md
    Untracked:  _targets/
    Untracked:  _targets_r/
    Untracked:  analysis/fetal_reference_cellid_gsea.Rmd
    Untracked:  analysis/fixed_intercept.Rmd
    Untracked:  analysis/iDEA_examples.Rmd
    Untracked:  analysis/latent_gene_list.Rmd
    Untracked:  analysis/latent_logistic_susie.Rmd
    Untracked:  analysis/libra_setup.Rmd
    Untracked:  analysis/linear_method_failure_modes.Rmd
    Untracked:  analysis/linear_regression_failure_regime.Rmd
    Untracked:  analysis/logistic_susie_veb_boost_vs_vb.Rmd
    Untracked:  analysis/references.bib
    Untracked:  analysis/simulations.Rmd
    Untracked:  analysis/test.Rmd
    Untracked:  analysis/wenhe_baboon_example.Rmd
    Untracked:  build_site.R
    Untracked:  cache/
    Untracked:  code/latent_logistic_susie.R
    Untracked:  code/marginal_sumstat_gsea_collapsed.R
    Untracked:  data/adipose_2yr_topsnp.txt
    Untracked:  data/fetal_reference_cellid_gene_sets.RData
    Untracked:  data/pbmc-purified/
    Untracked:  docs.zip
    Untracked:  index.md
    Untracked:  latent_logistic_susie_cache/
    Untracked:  simulation_targets/
    Untracked:  single_cell_pbmc_cache/
    Untracked:  summary_stat_gsea_exploration_cache/

Unstaged changes:
    Modified:   _simulation_targets.R
    Modified:   _targets.Rmd
    Modified:   analysis/gseabenchmark_tcga.Rmd
    Modified:   code/fit_baselines.R
    Modified:   code/fit_logistic_susie.R
    Modified:   code/fit_mr_ash.R
    Modified:   code/fit_susie.R
    Modified:   code/load_gene_sets.R
    Modified:   code/marginal_sumstat_gsea.R
    Modified:   code/simulate_gene_lists.R
    Modified:   code/utils.R
    Modified:   target_components/factories.R
    Modified:   target_components/methods.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/summary_stat_gsea_exploration.Rmd</code>) and HTML (<code>docs/summary_stat_gsea_exploration.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/karltayeb/logistic-susie-gsea/blob/122deece85c32abc1d72b089a34a2f24c62d8972/analysis/summary_stat_gsea_exploration.Rmd" target="_blank">122deec</a>
</td>
<td>
karltayeb
</td>
<td>
2022-03-29
</td>
<td>
wflow_publish(pages)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/karltayeb/logistic-susie-gsea/e0e196ca94a542e42af13b41cc2122fa33d7796c/docs/summary_stat_gsea_exploration.html" target="_blank">e0e196c</a>
</td>
<td>
karltayeb
</td>
<td>
2022-03-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/karltayeb/logistic-susie-gsea/blob/494dd37ee3f91e9ac7ac95a0b46bc29d8408df7c/analysis/summary_stat_gsea_exploration.Rmd" target="_blank">494dd37</a>
</td>
<td>
karltayeb
</td>
<td>
2022-03-28
</td>
<td>
wflow_publish(“analysis/summary_stat_gsea_exploration.Rmd”)
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>Peter has been thinking about the marginal/single-gene set enrichment problem. (<a href="https://www.overleaf.com/project/623485e71fafe7302e4b34f1" class="uri">https://www.overleaf.com/project/623485e71fafe7302e4b34f1</a>)</p>
<p>We’ve talked about modelling gene level summary statistics with a two-component mixture model (null and non-null component) and then have the prior probability of being in the non-null component dependent on gene-set membership. Here’s the model:</p>
<p><span class="math display">\[
\begin{align}
\hat \beta_i \sim N(\beta_i, s_i^2) \\
\beta_i \sim (1 - \pi_i)\delta_0 + \pi_iN(0,\sigma^2_i) \\
\ln \frac{\pi_i}{1-\pi_i} = \theta_0 + \theta_1x_i
\end{align}
\]</span></p>
<p>The detail that Peter’s expressed intest in is how we model <span class="math inline">\(\sigma_i^2\)</span>. Most treatments of GSEA use just z-scores or p-values. If we were to just model z-scores here with a model like</p>
<p><span class="math display">\[ 
\hat z_i = \frac{\hat\beta}{s_i} \sim N(z_i, 1) \\
z_i \sim (1-\pi_i) \delta_0 + \pi_i N(0, \sigma^2_0)
\]</span></p>
<p>Translating this to the summary stat model we can see it’s equivalent to choosing <span class="math inline">\(\sigma_i = \sigma_0 s_i\)</span>. That is, genes with larger standard error are expected to have larger effect sizes on average. This assumption may not hold in single cell DE expression analysis and other single cell analysis which might be upstream of GSEA.</p>
<p>We can learn the relationship between standard errors and effect sizes. The ASH paper proposed <span class="math inline">\(\sigma_i = s_i^{\alpha} \sigma_0\)</span>, for <span class="math inline">\(\alpha \in [0 ,1]\)</span>. <span class="math inline">\(\alpha=1\)</span> recovers the z-score model, while <span class="math inline">\(\alpha=0\)</span> recovers basic ash prior.</p>
<ul>
<li><p>The basic question are: - can we learn <span class="math inline">\(\alpha\)</span>?</p></li>
<li><p>how does it impact our GSEA results?</p></li>
</ul>
<p>I’m curious how it relates to other GSEA approaches. In particular methods like GORILLA <span class="citation">(Eden et al. 2007)</span> <span class="citation">(<strong>eden2009?</strong>)</span> consider the <strong>minimum</strong> hypergeometic tail probability across all possible cutoffs.</p>
<p>The two component model, with z-scores, won’t give a different gene ranking (3.2.1 of <span class="citation">(<strong>stepehsn2017?</strong>)</span>), but might be thought of loosely as a “soft” thresholding approach.</p>
<p>The beta + standard errors model may reorder genes, and also provides this “soft” thresholding behavior.</p>
</div>
<div id="implimentation" class="section level2">
<h2>Implimentation</h2>
<pre class="r"><code>library(tidyverse)</code></pre>
<pre><code>── Attaching packages ─────────────────────────────────────── tidyverse 1.3.1 ──</code></pre>
<pre><code>✓ ggplot2 3.3.5     ✓ purrr   0.3.4
✓ tibble  3.1.6     ✓ dplyr   1.0.8
✓ tidyr   1.2.0     ✓ stringr 1.4.0
✓ readr   2.1.2     ✓ forcats 0.5.1</code></pre>
<pre><code>── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
x dplyr::filter() masks stats::filter()
x dplyr::lag()    masks stats::lag()</code></pre>
<pre class="r"><code>library(latex2exp)
source(&#39;code/marginal_sumstat_gsea.R&#39;)</code></pre>
<p>I’ve implimented a simple EM optimization procedure in <code>code/marginal_sumstat_gsea.R</code>. Where we iteratively compute responsibilities, update the regression parameters, and update the variance parameters.</p>
<p>The regression parameters <span class="math inline">\(\theta = (\theta_0, \theta_1)\)</span> are optimized jointly. The variance parameters <span class="math inline">\((\alpha, \sigma_0)\)</span> we optimized cooardinate-wise.</p>
<p>To not get bogged down in deriving updates, everything is optimized with calls to <code>optim</code> or <code>optimize</code>. But we should not that the gradient and hessian of the <span class="math inline">\(Q\)</span> function (the EM bound) look just like normal logistic regression. And it shouldn’t be too hard to compute gradients for the variance parameters. So we can probably make things much faster with a bit more thought and effort!</p>
<p>Also, it seems standard to approximation the sampling distribution of the regression coefficients as <span class="math inline">\(\hat\theta \sim \mathcal N(\theta, -\nabla^2l(\hat \theta))\)</span> where <span class="math inline">\(l\)</span> is for likelihood. It’s not obvious to me that this works now that the logistic regression is a layer down in the model, and I’m sure this is motivated by asymptotic results that aren’t too valid especially for small gene sets (something like hessian @ MLE looks like Fisher information + estimate is unbiased and efficient)</p>
</div>
<div id="check-implimentation" class="section level2">
<h2>Check implimentation</h2>
<div id="gene-list-simulation" class="section level3">
<h3>Gene list simulation</h3>
<pre class="r"><code>#&#39; simulate summary statistics from two component enrichment model
#&#39; params is a list with elements theta, sigma0, alpha
#&#39; r.se is a function for sampleing standard errors runif(1e-3, 5) by default
simulate.gene.list = function(params, x = NULL, gene.set.prop=0.01, n.genes=10000, r.se=NULL) {
  theta0 = params$theta[1]
  theta1 = params$theta[2]
  sigma0 = params$sigma0
  alpha = params$alpha
  # simulate x if not provided
  if(is.null(x)){
    x &lt;- rbinom(n.genes, 1, gene.set.prop)  # gene set
  } else{
    n.genes &lt;- length(x)
  }
  
  if(is.null(r.se)){
    se &lt;- runif(n.genes, 1e-3, 5)  # simulate standard errors, reasonable?
  } else {
    se &lt;- r.se(n.genes)
  }
  sigma &lt;- se ^ alpha * sigma0
  gamma &lt;- rbinom(n.genes, 1, sigmoid(theta0 + theta1 * x))  # null/non-null
  beta &lt;- rnorm(n.genes, mean = 0, sd = se)
  beta &lt;- beta + (rnorm(n.genes, mean=0, sd=sigma) * gamma)
  return(list(x=x, beta=beta, se=se, gamma=gamma, params=params))
}</code></pre>
</div>
<div id="quick-check" class="section level3">
<h3>Quick check</h3>
<p>Here’s a quick example to check that optimization is working. And it seems to! We can optimize all of the parameters and the estimates looks close to the true values.</p>
<pre class="r"><code>source(&#39;code/marginal_sumstat_gsea.R&#39;)

params = list(
  theta = c(-2, 4),
  alpha = 0.5,
  sigma0 = 10
)
sim &lt;- simulate.gene.list(params, gene.set.prop = 0.1, n.genes = 10000)
gsea &lt;- summary.stat.gsea(sim$x, sim$beta, sim$se)

# fit just theta
params.init = list(
  theta = c(-4, 10),
  alpha = 0.5,
  sigma0 = 10
)
params.fit = gsea$expectation.maximiztion(
  params.init, n.iter=20, update.alpha = F, update.sigma0 = F)
params.fit$responsibilities &lt;- NULL
params.fit$theta</code></pre>
<pre><code>[1] -1.980673  3.874266</code></pre>
<pre class="r"><code># fit alpha too
params.init = list(
  theta = c(-4, 10),
  alpha = 1.0,
  sigma0 = 10
)
params.fit = gsea$expectation.maximiztion(
  params.init, n.iter=20, update.alpha = T, update.sigma0 = F)
params.fit$theta</code></pre>
<pre><code>[1] -1.979642  3.873384</code></pre>
<pre class="r"><code>params.fit$alpha</code></pre>
<pre><code>[1] 0.4994328</code></pre>
<pre class="r"><code># fit sigma too
params.init = list(
  theta = c(-4, 10),
  alpha = 0.5,
  sigma0 = 1
)
params.fit = gsea$expectation.maximiztion(
  params.init, n.iter=20, update.alpha = F, update.sigma0 = T)
params.fit$theta</code></pre>
<pre><code>[1] -1.987665  3.866905</code></pre>
<pre class="r"><code>params.fit$sigma0</code></pre>
<pre><code>[1] 10.16249</code></pre>
<pre class="r"><code># fit all
params.init = list(
  theta = c(-4, 10),
  alpha = 1.0,
  sigma0 = 1
)
params.fit = gsea$expectation.maximiztion(
  params.init, n.iter=50, update.alpha = T, update.sigma0 = T)
params.fit$theta</code></pre>
<pre><code>[1] -1.986874  3.849943</code></pre>
<pre class="r"><code>params.fit$alpha</code></pre>
<pre><code>[1] 0.4914275</code></pre>
<pre class="r"><code>params.fit$sigma0</code></pre>
<pre><code>[1] 10.21814</code></pre>
<p>And here’s a look at the likelihood surfaces for the enrichment parameters and variance parameters. These maps are a bit coarse, but you can see the strong dependence between the variance parameters. I guess since the target geneset only has a small fraction of the genes, there’s less information to estimate <span class="math inline">\(\theta_1\)</span> compared to <span class="math inline">\(\theta_0\)</span>.</p>
<pre class="r"><code>params = list(
  theta = c(-2, 4),
  alpha = 0.5,
  sigma0 = 10
)
sim &lt;- simulate.gene.list(params, gene.set.prop = 0.1, n.genes = 10000)
gsea &lt;- summary.stat.gsea(sim$x, sim$beta, sim$se)

lik.grid &lt;- xfun::cache_rds({
  purrr::cross_df(list(alpha=seq(0, 1, by=0.05), sigma0=seq(5, 15, by=0.1))) %&gt;%
  mutate(lik = map2_dbl(alpha, sigma0, ~ gsea$likelihood(params, alpha=.x, sigma0=.y)))
})
ggplot(lik.grid, aes(x=alpha, y=sigma0, z=lik)) + geom_contour_filled()</code></pre>
<p><img src="figure/summary_stat_gsea_exploration.Rmd/alpha.sigma0.density.2d-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-alpha.sigma0.density.2d-1">
Past versions of alpha.sigma0.density.2d-1.png
</button>
</p>
<div id="fig-alpha.sigma0.density.2d-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/karltayeb/logistic-susie-gsea/blob/e0e196ca94a542e42af13b41cc2122fa33d7796c/docs/figure/summary_stat_gsea_exploration.Rmd/alpha.sigma0.density.2d-1.png" target="_blank">e0e196c</a>
</td>
<td>
karltayeb
</td>
<td>
2022-03-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>params = list(
  theta = c(-2, 4),
  alpha = 0.5,
  sigma0 = 10
)
sim &lt;- simulate.gene.list(params, gene.set.prop = 0.1, n.genes = 10000)
gsea &lt;- summary.stat.gsea(sim$x, sim$beta, sim$se)

lik.grid &lt;- xfun::cache_rds({
  purrr::cross_df(list(theta0=seq(-4, 0, by=0.1), theta1=seq(2, 6, by=0.1))) %&gt;%
  mutate(lik = map2_dbl(theta0, theta1, ~ gsea$likelihood(params, theta=c(.x, .y))))
})

ggplot(lik.grid, aes(x=theta0, y=theta1, z=lik)) + geom_contour_filled()</code></pre>
<p><img src="figure/summary_stat_gsea_exploration.Rmd/theta.density.2d-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-theta.density.2d-1">
Past versions of theta.density.2d-1.png
</button>
</p>
<div id="fig-theta.density.2d-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/karltayeb/logistic-susie-gsea/blob/e0e196ca94a542e42af13b41cc2122fa33d7796c/docs/figure/summary_stat_gsea_exploration.Rmd/theta.density.2d-1.png" target="_blank">e0e196c</a>
</td>
<td>
karltayeb
</td>
<td>
2022-03-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="simulations" class="section level2">
<h2>Simulations</h2>
<p>The main questions we’re interested in is “does having alpha right matter?” From the likelihood surface plotted above we can see some dependence on <span class="math inline">\(\alpha\)</span> and <span class="math inline">\(\sigma_0\)</span>, and these will jointly influence our estimated enrichment parameters. To what extent does having <span class="math inline">\(\alpha\)</span> set incorrectly - bias our effect estimates - change the way we would prioritize gene sets?</p>
<p>We’ll work with simulated gene sets with varying size and overlap with the target gene set.</p>
<div id="set-up" class="section level3">
<h3>Set-up</h3>
<pre class="r"><code>roll = function(v, n=1){
  if(n==0){
    return(v)
  } else{
    return(c(tail(v, n), head(v, -n)))
  }
}

#&#39; make sequence of simulated gene sets of fixed size
#&#39; and decreasing overlap with first gene set
#&#39; gene.set.size = size of gene set
#&#39; by = how much to overlap incriment
sim.X.base = function(n.genes=1e4, gene.set.size=1e2, from=0, to=NULL, by=5){
  to &lt;- ifelse(is.null(to), gene.set.size, to)
  x = c(rep(1, gene.set.size), rep(0, n.genes-gene.set.size))
  u &lt;- map(seq(from, to, by=by), ~roll(x, n=.x))
  X &lt;- matrix(unlist(u), nrow = n.genes)
  return(X)
}

#&#39; make sequence of gene sets overlapping base gene set
sim.X.other = function(gene.set.size, other.size, by=5){
  sim.X.base(
    gene.set.size=other.size,
    from=max(0, gene.set.size - other.size), to=gene.set.size, by=by
  )
}

#&#39; put it all together
#&#39; returns matrix X
#&#39; first columns is the gene set of interest
#&#39; other columns are gene sets of varying sizes that overlap the first
sim.X = function(gene.set.size=50, set.sizes = NULL, by=5){
  X &lt;- sim.X.base(gene.set.size = gene.set.size, by=by)
  if(!is.null(set.sizes)){
    Xs &lt;- do.call(&#39;cbind&#39;, map(set.sizes, ~sim.X.other(gene.set.size, .x, by=by)))
    X &lt;- cbind(X, Xs)
  }
  return(X)
}</code></pre>
<pre class="r"><code>#&#39; fit the model and return parameter list
#&#39; ... arguments to pass to to EM
fit.sumstat.gsea = function(beta, se, x, params.init=NULL, theta=NULL, alpha=NULL, sigma0=NULL, ...){
  gsea &lt;- summary.stat.gsea(x, beta, se)
  # TODO: figure out how to initialize parameters
  if(is.null(params.init)){
    params.init = list(
      theta = c(-3, 0),
      alpha = 1.0,
      sigma0 = 1
    )
  }
  if(!is.null(theta)){
    params.init$theta = theta
  }
  if(!is.null(alpha)){
    params.init$alpha = alpha
  }
  if(!is.null(sigma0)){
    params.init$sigma0 = sigma0
  }
  params.fit = gsea$expectation.maximiztion(params.init, ...)
  return(params.fit)
}

#&#39; driver function for simulations
#&#39; fit model for all gene sets
#&#39; assume first column of X is the gene is sim$x
sim.driver = function(X, params, params.init=NULL, update.alpha=T, update.sigma0=T){
  sim &lt;- simulate.gene.list(params, x=X[,1])

  # fit model for each gene set
  res &lt;- map(1:dim(X)[2], ~fit.sumstat.gsea(
    sim$beta, sim$se, X[,.x],
    params.init=params.init,
    update.alpha=update.alpha, update.sigma0=update.sigma0
  ))
  
  # clean up data
  for (i in 1:length(res)){
    res[[i]]$responsibilities &lt;- NULL
    names(res[[i]]$theta) &lt;- paste0(&#39;theta&#39;, c(0, 1))
  }
  # dump into table with some useful stats
  res_tbl &lt;- tibble(
      overlap=(X[,1] %*% X)[1,],
      active = 1:dim(X)[2] == 1,
      set.size=colSums(X),
      p.overlap = overlap/set.size,
      res=res,
      theta0.true = params$theta[1],
      theta1.true = params$theta[2],
      alpha.true = params$alpha,
      sigma0.true = params$sigma0
    ) %&gt;% 
    unnest_wider(res) %&gt;%
    unnest_wider(theta) %&gt;%
    mutate(likelihood = map_dbl(lik.history, ~tail(.x, 1)))

  return(res_tbl)
}</code></pre>
</div>
<div id="first-simulation" class="section level3">
<h3>First simulation</h3>
<p>We simulate a target gene set with 50 genes, and gene sets with 5, 10, 15, …, 45 overlapping genes with the target gene set. We set <span class="math inline">\(\theta = (-2, 2)\)</span>, <span class="math inline">\(\alpha = 0.5\)</span> and <span class="math inline">\(\sigma_0 = 4\)</span>. That is background genes are non-null with <span class="math inline">\(\text{log-odds}=-2\)</span> and genes in the target gene set are non-null with <span class="math inline">\(\text{log-odds}=0\)</span>. We simulate summary statistics for 10,000 genes, with standard errors distributed <span class="math inline">\(\text{Unif}[1e-3, 5]\)</span></p>
<p>We estimate the parameters of the model where <span class="math inline">\(\alpha\)</span> is estimated, fixed to <span class="math inline">\(\alpha=0\)</span>, or fixed to <span class="math inline">\(\alpha = 1\)</span>, where the latter two cases represent the ash and z-score models respectively. We want to evaluate how this effects the parameter estimates, and more importantly the enrichment results.</p>
<pre class="r"><code>res &lt;- xfun::cache_rds({
  params = list(
    theta = c(-2, 2),
    alpha = 0.5,
    sigma0 = 4
  )
  X &lt;- sim.X(50, by=10)
  n.rep &lt;- 10
  # fix alpha = 0 
  params.init = list(
    theta = c(0, 0),
    alpha = 0.,
    sigma0 = 1
  )
  res &lt;- list()
  res[[&#39;alpha=0&#39;]] &lt;- rerun(
      n.rep, sim.driver(X, params, params.init=params.init, update.alpha = F)) %&gt;%
      do.call(rbind, .)
  
  # fix alpha = 1
  params.init$alpha = 1.0
  res[[&#39;alpha=1&#39;]] &lt;- rerun(
      n.rep, sim.driver(X, params, params.init=params.init, update.alpha = F)) %&gt;%
      do.call(rbind, .)
  
  # estimate alpha
  res[[&#39;learn.alpha&#39;]] &lt;- rerun(
      n.rep, sim.driver(X, params, params.init=params.init, update.alpha = T)) %&gt;%
      do.call(rbind, .)
  res
})

# add rep column
for (name in names(res)){
  res[[eval(name)]] &lt;- res[[eval(name)]] %&gt;%
    mutate(rep = cumsum(active)) %&gt;% ungroup()
}

res &lt;- do.call(rbind, res)
res &lt;- res %&gt;% mutate(
  fit.alpha = case_when(
    alpha == 0 ~ &#39;alpha=0&#39;,
    alpha == 1 ~ &#39;alpha=1&#39;,
    TRUE ~ &#39;estimate&#39;
  ),
  alpha.rel = (alpha - alpha.true) / alpha.true,
  theta0.rel = (theta0 - theta0.true) / abs(theta0.true),
  theta1.rel = (theta1 - theta1.true) / abs(theta1.true),
  sigma0.rel = (sigma0 - sigma0.true) / abs(sigma0.true),
)</code></pre>
<p>First we can restrict our attention to the parameter estimates of the active gene set. We can see that at <span class="math inline">\(\alpha=1\)</span> we tend to underestimate both the intercept and enrichment parameter. For <span class="math inline">\(\alpha=0\)</span> the regression parameters seem approximately unbiased at these simulation settings.</p>
<p>Unsurprisingly, our estimates of <span class="math inline">\(\sigma_0\)</span> are quite biased for fixed <span class="math inline">\(\alpha\)</span> values.But, at least for these simulation settings, it does not seem to bias our estimates of <span class="math inline">\(\theta_0\)</span>.</p>
<pre class="r"><code>library(latex2exp)
res %&gt;% filter(active) %&gt;% group_by(fit.alpha) %&gt;%
  reshape2::melt(measure.vars=c(&#39;theta0.rel&#39;, &#39;theta1.rel&#39;, &#39;alpha.rel&#39;, &#39;sigma0.rel&#39;)) %&gt;%
  ggplot(aes(x=variable, y=value, color=fit.alpha)) +
  geom_boxplot() +
  labs(title=TeX(&quot;$\\frac{\\hat{\\theta} - \\theta}{\\theta}$&quot;))</code></pre>
<p><img src="figure/summary_stat_gsea_exploration.Rmd/unnamed-chunk-1-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-1-1">
Past versions of unnamed-chunk-1-1.png
</button>
</p>
<div id="fig-unnamed-chunk-1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/karltayeb/logistic-susie-gsea/blob/e0e196ca94a542e42af13b41cc2122fa33d7796c/docs/figure/summary_stat_gsea_exploration.Rmd/unnamed-chunk-1-1.png" target="_blank">e0e196c</a>
</td>
<td>
karltayeb
</td>
<td>
2022-03-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>We can also look at what we estimate for the off-target gene sets, as a function of overlap. We might expect, when we fit the model for gene sets with less and less overlap, that our estimate of <span class="math inline">\(\theta_0\)</span> would increase (as the tested gene set is not relevant, and non-null genes are abosrbed into “background”) and a decrease in <span class="math inline">\(\theta_1\)</span> (gene sets with no or very little overlap are actually depleted relative to “background”)</p>
<p>Indeed, this is what we see. We’d expect the effect to be more extreme as the number of non-null genes in the target gene set increase.</p>
<p>Do we want to consider “depletion” an interesting enrichment? I think we’re often tempted to think of depletion as something like “expression of these genes is tightly regulated/not easily perturbed and maybe that’s because we’re not tolerant to perturbations of this process.”</p>
<p>But if that depletion is an artifact of the intercept term absorbing large enriched gene sets, then that seems less interesting. Depletion may be more interpretable in the joint model.</p>
<pre class="r"><code>res %&gt;% 
  filter(!active &amp; (fit.alpha==&#39;estimate&#39;)) %&gt;%
  group_by(fit.alpha) %&gt;%
  reshape2::melt(measure.vars=c(&#39;theta0&#39;, &#39;theta1&#39;, &#39;alpha&#39;, &#39;sigma0&#39;)) %&gt;%
  ggplot(aes(x=factor(overlap), y=value, color=fit.alpha)) +
  geom_boxplot() + facet_wrap(vars(variable),scales = &#39;free_y&#39;)</code></pre>
<p><img src="figure/summary_stat_gsea_exploration.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-1">
Past versions of unnamed-chunk-2-1.png
</button>
</p>
<div id="fig-unnamed-chunk-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/karltayeb/logistic-susie-gsea/blob/e0e196ca94a542e42af13b41cc2122fa33d7796c/docs/figure/summary_stat_gsea_exploration.Rmd/unnamed-chunk-2-1.png" target="_blank">e0e196c</a>
</td>
<td>
karltayeb
</td>
<td>
2022-03-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>  labs(title=TeX(&quot;$\\frac{\\hat{\\theta} - \\theta}{\\theta}$&quot;))</code></pre>
<pre><code>$title
   LaTeX: $\frac{\hat{\theta} - \theta}{\theta}$ 
plotmath: frac(hat(theta) - theta, theta) 

attr(,&quot;class&quot;)
[1] &quot;labels&quot;</code></pre>
<pre class="r"><code>res %&gt;% 
  filter(!active ) %&gt;%
  group_by(fit.alpha) %&gt;%
  reshape2::melt(measure.vars=c(&#39;theta0&#39;, &#39;theta1&#39;, &#39;alpha&#39;, &#39;sigma0&#39;)) %&gt;%
  ggplot(aes(x=factor(overlap), y=value, color=fit.alpha)) +
  geom_boxplot() + facet_grid(vars(variable), vars(fit.alpha), scales = &#39;free_y&#39;)</code></pre>
<p><img src="figure/summary_stat_gsea_exploration.Rmd/unnamed-chunk-2-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-2">
Past versions of unnamed-chunk-2-2.png
</button>
</p>
<div id="fig-unnamed-chunk-2-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/karltayeb/logistic-susie-gsea/blob/e0e196ca94a542e42af13b41cc2122fa33d7796c/docs/figure/summary_stat_gsea_exploration.Rmd/unnamed-chunk-2-2.png" target="_blank">e0e196c</a>
</td>
<td>
karltayeb
</td>
<td>
2022-03-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>  labs(title=TeX(&quot;$\\frac{\\hat{\\theta} - \\theta}{\\theta}$&quot;))</code></pre>
<pre><code>$title
   LaTeX: $\frac{\hat{\theta} - \theta}{\theta}$ 
plotmath: frac(hat(theta) - theta, theta) 

attr(,&quot;class&quot;)
[1] &quot;labels&quot;</code></pre>
<p>Do the parameter estimates under the fixed <span class="math inline">\(\alpha\)</span> models ever cause us to prioritize other overlapping gene sets over the true gene set? How should we compare/rank enrichment results across gene-sets?</p>
<p>For now I’m just using <span class="math inline">\(z_i = \hat\theta_{1i} / \hat s_{1i}\)</span>. These will tend to be inflated, as I think we tend to underestimate the standard errors. But hopefully they are inflated uniformly across gene sets so their comparison/ranking is meaningful?</p>
<p>Most of the time the target gene set is also the most strongly enriched. I’m sure we can find simulations settings where that is not the case. I should think about what distribution of standard errors would best highlight the weakness of ignoring <span class="math inline">\(\alpha\)</span>.</p>
<pre class="r"><code># add &quot;z-scores&quot;
res &lt;- res %&gt;% mutate(
  theta1.se = map_dbl(theta.se, ~ .x[2]),
  theta1.z = theta1 / theta1.se
  )

library(ggbeeswarm)
res %&gt;% 
  arrange(desc(abs(theta1.z))) %&gt;% 
  group_by(fit.alpha, rep) %&gt;% 
  mutate(rank = order(likelihood, decreasing = TRUE)) %&gt;%
  filter(active) %&gt;%
  ggplot(aes(x=fit.alpha, y=rank)) + geom_beeswarm() + 
  labs(title=&#39;Rank of target gene-set&#39;)</code></pre>
<pre><code>Warning in f(...): The default behavior of beeswarm has changed in version
0.6.0. In versions &lt;0.6.0, this plot would have been dodged on the y-axis. In
versions &gt;=0.6.0, grouponX=FALSE must be explicitly set to group on y-axis.
Please set grouponX=TRUE/FALSE to avoid this warning and ensure proper axis
choice.</code></pre>
<p><img src="figure/summary_stat_gsea_exploration.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/karltayeb/logistic-susie-gsea/blob/e0e196ca94a542e42af13b41cc2122fa33d7796c/docs/figure/summary_stat_gsea_exploration.Rmd/unnamed-chunk-3-1.png" target="_blank">e0e196c</a>
</td>
<td>
karltayeb
</td>
<td>
2022-03-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>res %&gt;% 
  arrange(desc(likelihood)) %&gt;% 
  group_by(fit.alpha, rep) %&gt;% 
  mutate(
    rank = order(likelihood, decreasing = TRUE),
    wrong.rank = rank &lt; rank[which(active)]
  ) %&gt;% 
  ggplot(aes(x=p.overlap, y=theta1, color=wrong.rank)) + 
  geom_jitter(width=0.02, height = 0.02) + facet_wrap(vars(fit.alpha))</code></pre>
<p><img src="figure/summary_stat_gsea_exploration.Rmd/unnamed-chunk-3-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-2">
Past versions of unnamed-chunk-3-2.png
</button>
</p>
<div id="fig-unnamed-chunk-3-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/karltayeb/logistic-susie-gsea/blob/e0e196ca94a542e42af13b41cc2122fa33d7796c/docs/figure/summary_stat_gsea_exploration.Rmd/unnamed-chunk-3-2.png" target="_blank">e0e196c</a>
</td>
<td>
karltayeb
</td>
<td>
2022-03-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>res %&gt;% 
  arrange(desc(likelihood)) %&gt;% 
  group_by(fit.alpha, rep) %&gt;% 
  mutate(
    rank = order(likelihood, decreasing = TRUE),
    wrong.rank = rank &lt; rank[which(active)]
  ) %&gt;% 
  ggplot(aes(x=p.overlap, y=theta1, color=factor(rank))) + 
  geom_jitter(width=0.02, height = 0.02) + facet_wrap(vars(fit.alpha))</code></pre>
<p><img src="figure/summary_stat_gsea_exploration.Rmd/unnamed-chunk-3-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-3">
Past versions of unnamed-chunk-3-3.png
</button>
</p>
<div id="fig-unnamed-chunk-3-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/karltayeb/logistic-susie-gsea/blob/e0e196ca94a542e42af13b41cc2122fa33d7796c/docs/figure/summary_stat_gsea_exploration.Rmd/unnamed-chunk-3-3.png" target="_blank">e0e196c</a>
</td>
<td>
karltayeb
</td>
<td>
2022-03-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>  labs(title=&#39;a&#39;)</code></pre>
<pre><code>$title
[1] &quot;a&quot;

attr(,&quot;class&quot;)
[1] &quot;labels&quot;</code></pre>
</div>
</div>
<div id="questions" class="section level2">
<h2>Questions</h2>
<ul>
<li>What’s a reasonable distribution of standard errors for single cell DE?</li>
<li>How do we evaluate significance of enrichment parameter <span class="math inline">\(\theta_1\)</span>/compare enrichment results across gene sets? Right now we’re just optimizing the marginal likelihood to get point estimates of <span class="math inline">\(\theta_1\)</span>, and hoping that the the usual approximate standard error for logistic regression is good here also. Putting a prior on the parameters and estimating the posterior of <span class="math inline">\(\theta_1\)</span> may get us closer to where we need to be.</li>
</ul>
</div>
<div id="other-things-to-try" class="section level2">
<h2>Other things to try</h2>
<ul>
<li>how does the marginal model behave for gene sets with hierarchical structure (simulation where the target gene set is a super set and subset of off-target gene sets).</li>
</ul>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-eden2007" class="csl-entry">
Eden, Eran, Doron Lipson, Sivan Yogev, and Zohar Yakhini. 2007. <span>“Discovering Motifs in Ranked Lists of DNA Sequences.”</span> <em>PLOS Computational Biology</em> 3 (3): e39. <a href="https://doi.org/10.1371/journal.pcbi.0030039">https://doi.org/10.1371/journal.pcbi.0030039</a>.
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.2 (2021-11-01)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices datasets  utils     methods   base     

other attached packages:
 [1] ggbeeswarm_0.6.0 latex2exp_0.9.4  forcats_0.5.1    stringr_1.4.0   
 [5] dplyr_1.0.8      purrr_0.3.4      readr_2.1.2      tidyr_1.2.0     
 [9] tibble_3.1.6     ggplot2_3.3.5    tidyverse_1.3.1 

loaded via a namespace (and not attached):
 [1] httr_1.4.2          sass_0.4.0          jsonlite_1.8.0     
 [4] viridisLite_0.4.0   modelr_0.1.8        bslib_0.3.1        
 [7] assertthat_0.2.1    BiocManager_1.30.16 highr_0.9          
[10] vipor_0.4.5         renv_0.15.0         cellranger_1.1.0   
[13] yaml_2.3.5          progress_1.2.2      pillar_1.7.0       
[16] backports_1.4.1     glue_1.6.2          digest_0.6.29      
[19] promises_1.2.0.1    rvest_1.0.2         colorspace_2.0-3   
[22] htmltools_0.5.2     httpuv_1.6.5        plyr_1.8.6         
[25] pkgconfig_2.0.3     broom_0.7.12        haven_2.4.3        
[28] scales_1.1.1        whisker_0.4         later_1.3.0        
[31] tzdb_0.2.0          git2r_0.29.0        generics_0.1.2     
[34] farver_2.1.0        ellipsis_0.3.2      withr_2.5.0        
[37] cli_3.2.0           magrittr_2.0.2      crayon_1.5.0       
[40] readxl_1.3.1        evaluate_0.15       fs_1.5.2           
[43] fansi_1.0.2         xml2_1.3.3          beeswarm_0.4.0     
[46] tools_4.1.2         prettyunits_1.1.1   hms_1.1.1          
[49] lifecycle_1.0.1     matrixStats_0.61.0  munsell_0.5.0      
[52] reprex_2.0.1        isoband_0.2.5       compiler_4.1.2     
[55] jquerylib_0.1.4     rlang_1.0.2         grid_4.1.2         
[58] rstudioapi_0.13     labeling_0.4.2      rmarkdown_2.13     
[61] gtable_0.3.0        DBI_1.1.2           reshape2_1.4.4     
[64] R6_2.5.1            lubridate_1.8.0     knitr_1.37         
[67] fastmap_1.1.0       utf8_1.2.2          workflowr_1.7.0    
[70] rprojroot_2.0.2     stringi_1.7.6       Rcpp_1.0.8.2       
[73] vctrs_0.3.8         dbplyr_2.1.1        tidyselect_1.1.2   
[76] xfun_0.30          </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
