---
title: "Single Cell PBMC Differential Expression"
author: "karltayeb"
date: "2022-04-11"
output:
  workflowr::wflow_html:
    code_folding: hide
editor_options:
  chunk_output_type: console
---

## Introduction

PBMC data from Zheng et. al 2017.

```{r setup, include=FALSE}
knitr::opts_chunk$set(
  comment = '#>',
  message=FALSE, warning=FALSE
)
```

```{r load.results}
library(gseasusie)
library(ggplot2)
library(tidyverse)
library(targets)

tar_load(sc_pbmc_deseq_enrichment_summary)
results <- sc_pbmc_deseq_enrichment_summary
```

## Enrichment results

```{r plot.function, hide=TRUE}
color_sign <- gseasusie:::color_sign

interactive_table2 = function(res){
  dt <- res %>% 
    dplyr::filter(overlap > 0) %>%
    dplyr::mutate(
      logOddsRatio = log(oddsRatio),
      nlog10pFishersExact = -log10(pFishersExact)
    ) %>%
    dplyr::arrange(dplyr::desc(nlog10pFishersExact)) %>%
    dplyr::mutate(
      fisherRank = dplyr::row_number(),
      in_active_cs = dplyr::if_else(is.na(in_cs), FALSE, in_cs & active_cs)) %>%
    dplyr::select(geneSet, description, in_active_cs, beta, pip, overlap, geneSetSize, logOddsRatio, nlog10pFishersExact, component, fisherRank) %>%
    dplyr::mutate(dplyr::across(!where(is.numeric), as.factor))

  dt %>%
    reactable::reactable(
      filterable=TRUE,
      minRows=20,
      columns = list(
        pip = reactable::colDef(format = reactable::colFormat(digits = 3)),
        logOddsRatio = reactable::colDef(style= function(value){color_sign(value)},
                                         format = reactable::colFormat(digits = 3)),
        beta = reactable::colDef(style= function(value){color_sign(value)},
                                 format = reactable::colFormat(digits = 3)),
        nlog10pFishersExact = reactable::colDef(format = reactable::colFormat(digits = 3))
      ),
      rowStyle = function(index){
        if(dt$in_active_cs[index] == TRUE){
          list(background = "#e5f5e0")
        }
      },
      defaultSorted = list(nlog10pFishersExact='desc')
    )
}

#' @export
residual_enrichment_histogram2 = function(res){
  res %>%
    dplyr::select(geneSet, pval_marginal, pval_residual) %>%
    tidyr::pivot_longer(dplyr::starts_with('pval'), values_to = 'pval') %>%
    ggplot2::ggplot(aes(x=pval)) +
      ggplot2::geom_histogram() + ggplot2::facet_wrap(vars(name))
}

pack_group <- gseasusie:::pack_group

static_table = function(res){
  require(kableExtra)
  tbl_filtered <-
    res %>%
    arrange(pFishersExact) %>%
    mutate(fisherRank = row_number()) %>%
    filter(in_cs, active_cs) %>%
    group_by(component) %>%
    arrange(component, desc(alpha)) %>%
    dplyr::ungroup() %>%
    dplyr::mutate(logOddsRatio = log10(oddsRatio))

  tbl_filtered %>%
    dplyr::select(
      component, geneSet, description, geneSetSize, overlap,
      logOddsRatio, conditional_beta, conditional_beta_se,
      alpha, pip, pFishersExact, fisherRank) %>%
    dplyr::mutate_if(is.numeric, funs(as.character(signif(., 3)))) %>%
    pack_group %>%
    column_spec(c(4), color=ifelse(tbl_filtered$beta > 0, 'green', 'red')) %>%
    kableExtra::kable_styling()
}
```

## Enrichment results

```{r experiment.level.results, results='asis'}
results <- results %>% arrange(experiment, ptop, db)
N <- dim(results)[1]
for (i in 1:N){
  this_experiment <- results$experiment[[i]]
  this_db <- results$db[[i]]
  this_ptop <- results$ptop[[i]]
  res <- results$enrichment_summary[[i]]

  cat("\n") 
  cat("###", this_experiment, ' ', this_ptop, ' ', this_db, "\n") # Create second level headings with the names.

  volcano <- res %>% gseasusie::enrichment_volcano2()
  hist <- res %>% filter(overlap > 0) %>% residual_enrichment_histogram2() 
  print(volcano)
  print(hist)

  cat("\n\n")
  print(static_table(res))
}
```

### Glossary

* `alpha` is the posterior probability of SuSiE including this gene set **in this component** which is different from PIP (probability of SuSiE including this gene set in ANY component)
* `beta` posterior mean/standard error of posterior mean for effect size. Standard errors are likely too small. 
* `oddsRatio, pHypergeometric, pFishersExact` construct a contingency table (gene list membersip) x (gene set membership), estimate the `oddsRatio` gives the odds of being in the gene list conditional on being in the gene set / odds of being in the gene list conditional on NOT being in the gene set. `pHypergeometric` and `pFishersExact` are pvalues from 1 and 2 sided test respectively. 
